<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-12-13" />

<title>Data_ind_Diffabtrial</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Omar-Johnson/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Data_ind_Diffabtrial</h1>
<h4 class="date">2023-12-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230911code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230911)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230911code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230911)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreed7917090991e40eceab98de8887fb7675a795cc2targetblankd791709a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Omar-Johnson/myproject/tree/d7917090991e40eceab98de8887fb7675a795cc2" target="_blank">d791709</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreed7917090991e40eceab98de8887fb7675a795cc2targetblankd791709a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Omar-Johnson/myproject/tree/d7917090991e40eceab98de8887fb7675a795cc2" target="_blank">d791709</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  #14_venn_diagramm.png
    Untracked:  #14_venn_diagramm.png.2024-01-11_15-41-45.log
    Untracked:  #14_venn_diagramm.png.2024-01-11_15-41-51.log
    Untracked:  VennDiagram.2024-01-11_15-43-33.log
    Untracked:  VennDiagram.2024-01-11_15-43-45.log
    Untracked:  VennDiagram.2024-01-11_15-48-56.log
    Untracked:  VennDiagram.2024-01-11_15-49-14.log
    Untracked:  VennDiagram.2024-01-11_15-49-51.log
    Untracked:  VennDiagram.2024-01-11_15-52-24.log
    Untracked:  VennDiagram.2024-01-11_15-53-14.log
    Untracked:  VennDiagram.2024-01-11_15-53-47.log
    Untracked:  VennDiagram.2024-01-11_15-54-07.log
    Untracked:  VennDiagram.2024-01-11_18-08-58.log
    Untracked:  VennDiagram.2024-01-11_18-14-06.log
    Untracked:  VennDiagram.2024-01-11_18-17-11.log
    Untracked:  VennDiagram.2024-01-11_18-18-29.log
    Untracked:  VennDiagram.2024-01-11_18-19-23.log
    Untracked:  VennDiagram.2024-01-11_18-19-52.log
    Untracked:  VennDiagram.2024-01-11_18-22-22.log
    Untracked:  VennDiagram.2024-01-11_18-22-23.log
    Untracked:  VennDiagram.2024-01-12_17-13-03.log
    Untracked:  VennDiagram.2024-01-12_17-13-04.log
    Untracked:  analysis/Data_ind_imp_RUVg_WPCNA.Rmd
    Untracked:  analysis/Random_bed.R

Unstaged changes:
    Modified:   Data_ind_Diffabtrial.Rmd
    Modified:   analysis/Diffabtrial.Rmd
    Modified:   analysis/Differentially_abundant_proteins_impute.Rmd
    Modified:   analysis/Differentially_abunddant_proteins.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Data_ind_Diffabtrial.Rmd</code>)
and HTML (<code>docs/Data_ind_Diffabtrial.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/d7917090991e40eceab98de8887fb7675a795cc2/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">d791709</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-12
</td>
<td>
Updated after removal of non-human peptides
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/Data_ind_Diffabtrial.html" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/66a78d57e171358c0f7288144bb79fffdb8337c5/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">66a78d5</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/Data_ind_Diffabtrial.html" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/9f6792be7a3787359cf554dc8d7686fa9b082633/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">9f6792b</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/4c91815ce583a586f165366016fe3e45cfd58976/docs/Data_ind_Diffabtrial.html" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/d019bfff2b69dc1b3bfd2179e43a42ca48acbfb4/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">d019bff</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/2929416dfeadd1a4f9ac6a99c90135eced046d50/docs/Data_ind_Diffabtrial.html" target="_blank">2929416</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/fd2f3e14b23d5e43e671319258845d74f8c7253a/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">fd2f3e1</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
<td>
wflow_publish("analysis/Data_ind_Diffabtrial.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/397b327d39f332fa31b98c8b82ac499fe31a9f81/docs/Data_ind_Diffabtrial.html" target="_blank">397b327</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/2c1648060c8ef0674909f51c1604cb5c535ee359/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">2c16480</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/Data_ind_Diffabtrial.html" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/fefc20b7587b5cea029769167c178589027a69f8/analysis/Data_ind_Diffabtrial.Rmd" target="_blank">fefc20b</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="added-funtions" class="section level1">
<h1>Added funtions</h1>
<pre class="r"><code># 1
remove_rows_with_nas &lt;- function(data_frame) {
  data_frame[rowSums(is.na(data_frame)) &lt; 5, ]
}

# 2
remove_rows_with_nas_var &lt;- function(data_frame, NA_remove) {
  data_frame[rowSums(is.na(data_frame)) &lt; NA_remove, ]
}

# 3
plot_nrow_vs_naremove &lt;- function(data_frame) {
  results &lt;- data.frame(NA_remove = integer(), Rows_Remaining = integer())

  for (i in 1:ncol(data_frame)) {
    filtered_df &lt;- remove_rows_with_nas_var(data_frame, NA_remove = i)
    results &lt;- rbind(results, data.frame(NA_remove = i, Rows_Remaining = nrow(filtered_df)))
  }

  ggplot(results, aes(x = NA_remove, y = Rows_Remaining)) +
    geom_point() +
    geom_line() +
    xlab(&quot;Na tolerance threshold&quot;) +
    ylab(&quot;Number of proteins remaining&quot;) +
    theme_minimal()
}</code></pre>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
<pre class="r"><code># Load your data
Protein_DF &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/RNA_Protein_DF/Data_ind_prot_DOX_24hr.csv&quot;, header = TRUE)

# Include metadata about samples 
Meta &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Meta_DIA.csv&quot;, header = TRUE)</code></pre>
</div>
<div id="wrangle-data-frame" class="section level1">
<h1>Wrangle Data Frame</h1>
<pre class="r"><code># Save the original 
Full_DF &lt;- Protein_DF 
Full_DF %&gt;% head()</code></pre>
<pre><code>  Protein.Group Protein.Ids Protein.Names   Genes First.Protein.Description
1    A0A0B4J2A2  A0A0B4J2A2            NA PPIAL4C                        NA
2    A0A0B4J2D5  A0A0B4J2D5            NA  GATD3B                        NA
3    A0A494C071  A0A494C071            NA   PWWP4                        NA
4        A0AVT1      A0AVT1            NA    UBA6                        NA
5        A0FGR8      A0FGR8            NA   ESYT2                        NA
6        A0JLT2      A0JLT2            NA   MED19                        NA
          S1       S3       S5       S7       S9         S2       S4       S6
1   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000  1460000
2 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000 13600000
3   381352.0   429735   287237   412285   380499   433456.0   540008   376304
4    62810.7   135522   159045   123045   123823    49388.1   144860   127488
5  1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000  1080000
6   217977.0   181404   182657   145408   208660   224881.0   202427   160150
        S8      S10
1  1390000  1620000
2 13200000 12600000
3   460580   355701
4    84684   102306
5  1120000  1060000
6   126127   209726</code></pre>
<pre class="r"><code>Full_DF %&gt;% dim()</code></pre>
<pre><code>[1] 4261   15</code></pre>
<pre class="r"><code># Subset the columns referring to abundance
Protein_DF_Abundance &lt;- Full_DF[,colnames(Full_DF) %in% c(&quot;Protein.Ids&quot;, &quot;S1&quot; , &quot;S3&quot; , &quot;S5&quot; , &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot; , &quot;S8&quot;, &quot;S10&quot;)] 
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>  Protein.Ids         S1       S3       S5       S7       S9         S2
1  A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0
2  A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0
3  A0A494C071   381352.0   429735   287237   412285   380499   433456.0
4      A0AVT1    62810.7   135522   159045   123045   123823    49388.1
5      A0FGR8  1240000.0  1120000   969130  1040000  1020000  1120000.0
6      A0JLT2   217977.0   181404   182657   145408   208660   224881.0
        S4       S6       S8      S10
1  2020000  1460000  1390000  1620000
2 13500000 13600000 13200000 12600000
3   540008   376304   460580   355701
4   144860   127488    84684   102306
5  1120000  1080000  1120000  1060000
6   202427   160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   11</code></pre>
<pre class="r"><code>colnames(Protein_DF_Abundance) &lt;- c(&quot;Accession&quot;, &quot;S1&quot; , &quot;S3&quot; , &quot;S5&quot; , &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot; , &quot;S8&quot;, &quot;S10&quot;)
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>   Accession         S1       S3       S5       S7       S9         S2       S4
1 A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
2 A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
3 A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
4     A0AVT1    62810.7   135522   159045   123045   123823    49388.1   144860
5     A0FGR8  1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
6     A0JLT2   217977.0   181404   182657   145408   208660   224881.0   202427
        S6       S8      S10
1  1460000  1390000  1620000
2 13600000 13200000 12600000
3   376304   460580   355701
4   127488    84684   102306
5  1080000  1120000  1060000
6   160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   11</code></pre>
<pre class="r"><code># Check for duplicated rows 
# Identify duplicated values
Protein_DF_Abundance$duplicated_name &lt;- duplicated(Protein_DF_Abundance$Accession)

# This will return a logical vector where TRUE indicates the position of duplicates in the column.
# To see only rows with duplicated values, you can subset the dataframe like this:
duplicated_rows &lt;- Protein_DF_Abundance[Protein_DF_Abundance$duplicated_name == TRUE, ]
print(duplicated_rows) # We have 0 rows with duplicate protein names </code></pre>
<pre><code> [1] Accession       S1              S3              S5             
 [5] S7              S9              S2              S4             
 [9] S6              S8              S10             duplicated_name
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># Make rownames of the data frame accession IDs
rownames(Protein_DF_Abundance) &lt;- Protein_DF_Abundance$Accession

# Change colnames 
colnames(Protein_DF_Abundance)</code></pre>
<pre><code> [1] &quot;Accession&quot;       &quot;S1&quot;              &quot;S3&quot;              &quot;S5&quot;             
 [5] &quot;S7&quot;              &quot;S9&quot;              &quot;S2&quot;              &quot;S4&quot;             
 [9] &quot;S6&quot;              &quot;S8&quot;              &quot;S10&quot;             &quot;duplicated_name&quot;</code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[, -c(1, 12)]

# Remove non-human peptides 
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   10</code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[!rownames(Protein_DF_Abundance) %in% c(&quot;Q6ZSR9&quot;, &quot;P15252&quot;,&quot;P25691&quot;,&quot;P00761&quot;), ]
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4257   10</code></pre>
<pre class="r"><code># Assuming column names of Protein_DF match with the rows in Meta
Meta</code></pre>
<pre><code>   Samples  Ind  Rep    Cond     Cond_Ind
1       S1 B_77  Bio     Dox     B_77_Dox
2       S3 C_87  Bio     Dox     C_87_Dox
3       S5 A_48 Tech     Dox     A_48_Dox
4       S7 A_48 Tech     Dox     A_48_Dox
5       S9 A_48 Tech     Dox     A_48_Dox
6       S2 B_77  Bio Control B_77_Control
7       S4 C_87  Bio Control C_87_Control
8       S6 A_48 Tech Control A_48_Control
9       S8 A_48 Tech Control A_48_Control
10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>rownames(Meta) &lt;- Meta$Samples
Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>colnames(Protein_DF_Abundance) &lt;- Meta$Samples
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
</div>
<div id="remove-na" class="section level1">
<h1>Remove NA</h1>
<pre class="r"><code>#1.  Find rows with any NA values
rows_with_na &lt;- apply(Protein_DF_Abundance, 1, function(x) any(is.na(x))) %&gt;% which()
 

rows_with_na &lt;- as.numeric(rows_with_na)


# 2. Remove rows with NA values 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[-rows_with_na, ]

# 3. Sanity check 
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 3932   10</code></pre>
</div>
<div id="log2-transform" class="section level1">
<h1>Log2 transform</h1>
<pre class="r"><code>log2_norm_counts &lt;- log2(Protein_DF_Abundance)
log2_norm_counts %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%2090-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-90-1">
Past versions of "Wrangle Data Frame 90-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-90-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 90-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="quantile-normalization" class="section level1">
<h1>Quantile Normalization</h1>
<pre class="r"><code>normalizedData &lt;- limma::normalizeBetweenArrays(log2_norm_counts[, c(1,2,4,6,7,9)], method = &quot;quantile&quot;)
normalizedData %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%2055-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-55-1">
Past versions of "Wrangle Data Frame 55-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-55-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 55-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="linear-model---na-proteins" class="section level1">
<h1>Linear model - NA proteins</h1>
<pre class="r"><code>#  counts need to be integer values and in a numeric matrix 
counts &lt;- as.matrix(normalizedData) 
counts %&gt;% dim()</code></pre>
<pre><code>[1] 3932    6</code></pre>
<pre class="r"><code># Create a DataFrame for the phenoData
phenoData &lt;- DataFrame(Meta)
phenoData_sub &lt;- phenoData[c(1,2,4,6,7,9), ]


# Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~ 0 + Cond , phenoData_sub)
design</code></pre>
<pre><code>   CondControl CondDox
S1           0       1
S3           0       1
S7           0       1
S2           1       0
S4           1       0
S8           1       0
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;)


# Fit model
dupcor &lt;- duplicateCorrelation(counts , design, block = phenoData_sub$Ind)
fit &lt;- lmFit(object = counts, design = design, block = phenoData_sub$Ind, correlation = dupcor$consensus)
fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2 )


# Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        570
NotSig     2833
Up          529</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(normalizedData)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)
toptable_summary$P.Value %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-1">
Past versions of "upcor on 6 samples 2-1.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of log2-quantile normalized-RUVs values:
prcomp_res &lt;- prcomp(t(normalizedData %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res, data = as.data.frame(phenoData_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-2">
Past versions of "upcor on 6 samples 2-2.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-2.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary &lt;- toptable_summary %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)

# 2. Plot 
ggplot(toptable_summary)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 7.5)+
  xlim(-5, 5)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-3">
Past versions of "upcor on 6 samples 2-3.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-3.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Other fits 
fit3 &lt;- eBayes(fit2, robust = TRUE, trend = TRUE)</code></pre>
<pre><code>Warning: One very small variance detected, has been offset away from zero</code></pre>
<pre class="r"><code># Summarize
results_summary &lt;- decideTests(fit3, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        571
NotSig     2808
Up          553</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit3, coef = &quot;DoxvNorm&quot;,number = (nrow(normalizedData)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)
toptable_summary$P.Value %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-4">
Past versions of "upcor on 6 samples 2-4.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-4.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of log2-quantile normalized-RUVs values:
prcomp_res &lt;- prcomp(t(normalizedData %&gt;% as.matrix()), center = TRUE)


ggplot2::autoplot(prcomp_res, data = as.data.frame(phenoData_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-5">
Past versions of "upcor on 6 samples 2-5.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-5.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary &lt;- toptable_summary %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)


# 2. Plot 
ggplot(toptable_summary)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 7.5)+
  xlim(-5, 5)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/upcor%20on%206%20samples%202-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-6-samples-2-6">
Past versions of "upcor on 6 samples 2-6.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-6-samples-2-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/upcor on 6 samples 2-6.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="wrangle-data-frame-for-imputed-data" class="section level1">
<h1>Wrangle Data Frame for imputed data</h1>
<pre class="r"><code># Save the original
Full_DF &lt;- Protein_DF
Full_DF %&gt;% head()</code></pre>
<pre><code>  Protein.Group Protein.Ids Protein.Names   Genes First.Protein.Description
1    A0A0B4J2A2  A0A0B4J2A2            NA PPIAL4C                        NA
2    A0A0B4J2D5  A0A0B4J2D5            NA  GATD3B                        NA
3    A0A494C071  A0A494C071            NA   PWWP4                        NA
4        A0AVT1      A0AVT1            NA    UBA6                        NA
5        A0FGR8      A0FGR8            NA   ESYT2                        NA
6        A0JLT2      A0JLT2            NA   MED19                        NA
          S1       S3       S5       S7       S9         S2       S4       S6
1   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000  1460000
2 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000 13600000
3   381352.0   429735   287237   412285   380499   433456.0   540008   376304
4    62810.7   135522   159045   123045   123823    49388.1   144860   127488
5  1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000  1080000
6   217977.0   181404   182657   145408   208660   224881.0   202427   160150
        S8      S10
1  1390000  1620000
2 13200000 12600000
3   460580   355701
4    84684   102306
5  1120000  1060000
6   126127   209726</code></pre>
<pre class="r"><code>Full_DF %&gt;% dim()</code></pre>
<pre><code>[1] 4261   15</code></pre>
<pre class="r"><code># # Subset the columns referring to abundance
Protein_DF_Abundance &lt;- Full_DF[,colnames(Full_DF) %in% c(&quot;Protein.Ids&quot;, &quot;S1&quot; , &quot;S3&quot; , &quot;S5&quot; , &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot; , &quot;S8&quot;, &quot;S10&quot;)]
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>  Protein.Ids         S1       S3       S5       S7       S9         S2
1  A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0
2  A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0
3  A0A494C071   381352.0   429735   287237   412285   380499   433456.0
4      A0AVT1    62810.7   135522   159045   123045   123823    49388.1
5      A0FGR8  1240000.0  1120000   969130  1040000  1020000  1120000.0
6      A0JLT2   217977.0   181404   182657   145408   208660   224881.0
        S4       S6       S8      S10
1  2020000  1460000  1390000  1620000
2 13500000 13600000 13200000 12600000
3   540008   376304   460580   355701
4   144860   127488    84684   102306
5  1120000  1080000  1120000  1060000
6   202427   160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   11</code></pre>
<pre class="r"><code>colnames(Protein_DF_Abundance) &lt;- c(&quot;Accession&quot;, &quot;S1&quot; , &quot;S3&quot; , &quot;S5&quot; , &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot; , &quot;S8&quot;, &quot;S10&quot;)
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>   Accession         S1       S3       S5       S7       S9         S2       S4
1 A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
2 A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
3 A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
4     A0AVT1    62810.7   135522   159045   123045   123823    49388.1   144860
5     A0FGR8  1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
6     A0JLT2   217977.0   181404   182657   145408   208660   224881.0   202427
        S6       S8      S10
1  1460000  1390000  1620000
2 13600000 13200000 12600000
3   376304   460580   355701
4   127488    84684   102306
5  1080000  1120000  1060000
6   160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   11</code></pre>
<pre class="r"><code># # Check for duplicated rows 
# # Identify duplicated values
Protein_DF_Abundance$duplicated_name &lt;- duplicated(Protein_DF_Abundance$Accession)

# # This will return a logical vector where TRUE indicates the position of duplicates in the column.
# # To see only rows with duplicated values, you can subset the dataframe like this:
duplicated_rows &lt;- Protein_DF_Abundance[Protein_DF_Abundance$duplicated_name == TRUE, ]
print(duplicated_rows) # We have 0 rows with duplicate protein names</code></pre>
<pre><code> [1] Accession       S1              S3              S5             
 [5] S7              S9              S2              S4             
 [9] S6              S8              S10             duplicated_name
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># # Make rownames of the data frame accession IDs
rownames(Protein_DF_Abundance) &lt;- Protein_DF_Abundance$Accession

# # Change colnames 
colnames(Protein_DF_Abundance)</code></pre>
<pre><code> [1] &quot;Accession&quot;       &quot;S1&quot;              &quot;S3&quot;              &quot;S5&quot;             
 [5] &quot;S7&quot;              &quot;S9&quot;              &quot;S2&quot;              &quot;S4&quot;             
 [9] &quot;S6&quot;              &quot;S8&quot;              &quot;S10&quot;             &quot;duplicated_name&quot;</code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[, -c(1, 12)]
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   10</code></pre>
<pre class="r"><code># Remove non-human peptides 
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4261   10</code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[!rownames(Protein_DF_Abundance) %in% c(&quot;Q6ZSR9&quot;, &quot;P15252&quot;,&quot;P25691&quot;,&quot;P00761&quot;), ]
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4257   10</code></pre>
<pre class="r"><code># # Assuming column names of Protein_DF match with the rows in Meta
Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>rownames(Meta) &lt;- Meta$Samples
Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>colnames(Protein_DF_Abundance) &lt;- Meta$Samples
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
</div>
<div id="viewing-how-na-threshold-vs-remaining-proteins"
class="section level1">
<h1>Viewing how NA threshold vs remaining proteins</h1>
<pre class="r"><code># View how NA thresholds affect the amount of proteins remaining to model
# All proteins 
Protein_DF_Abundance %&gt;% nrow()</code></pre>
<pre><code>[1] 4257</code></pre>
<pre class="r"><code># Removing proteins with 1 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 1 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 3932</code></pre>
<pre class="r"><code># Removing proteins with 2 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 2 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4048</code></pre>
<pre class="r"><code># Removing proteins with 3 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 3 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4114</code></pre>
<pre class="r"><code># Removing proteins with 4 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 4 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4149</code></pre>
<pre class="r"><code># Removing proteins with 5 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 5 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4178</code></pre>
<pre class="r"><code># Removing proteins with 6 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 6 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4208</code></pre>
<pre class="r"><code># Removing proteins with 7 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 7 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4225</code></pre>
<pre class="r"><code># Removing proteins with 8 or more NA
remove_rows_with_nas_var(data_frame = Protein_DF_Abundance , NA_remove = 8 ) %&gt;% nrow()</code></pre>
<pre><code>[1] 4237</code></pre>
<pre class="r"><code># View in graph 
plot_nrow_vs_naremove(Protein_DF_Abundance)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/viewing%20NA%20and%20imp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-viewing-NA-and-imp-1">
Past versions of "viewing NA and imp-1.png"
</button>
</p>
<div id="fig-no-spaces-viewing-NA-and-imp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/viewing NA and imp-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="na-removal-and-imputing-with-knn" class="section level1">
<h1>NA removal and imputing with knn</h1>
<pre class="r"><code># View original abundances 
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-1">
Past versions of "knn impute NA-1.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-2">
Past versions of "knn impute NA-2.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-2.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-3">
Past versions of "knn impute NA-3.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-3.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-4">
Past versions of "knn impute NA-4.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-4.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-5">
Past versions of "knn impute NA-5.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-5.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-6">
Past versions of "knn impute NA-6.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-6.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-7">
Past versions of "knn impute NA-7.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-7.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-8">
Past versions of "knn impute NA-8.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-8.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-9">
Past versions of "knn impute NA-9.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-9.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-10">
Past versions of "knn impute NA-10.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-10.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Remove non - human peptides 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[!rownames(Protein_DF_Abundance) %in% c(&quot;Q6ZSR9&quot;, &quot;P15252&quot;,&quot;P25691&quot;,&quot;P00761&quot;), ]
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4257   10</code></pre>
<pre class="r"><code># View abundances after removing proteins with 5 or more (50%) NA 
Protein_DF_Abundance_imp &lt;- remove_rows_with_nas(Protein_DF_Abundance)
Protein_DF_Abundance_imp %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>Protein_DF_Abundance_imp[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-11">
Past versions of "knn impute NA-11.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-11.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-12">
Past versions of "knn impute NA-12.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-12.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-13">
Past versions of "knn impute NA-13.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-13.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-14">
Past versions of "knn impute NA-14.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-14.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-15">
Past versions of "knn impute NA-15.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-15.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-16">
Past versions of "knn impute NA-16.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-16.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-17">
Past versions of "knn impute NA-17.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-17.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-18">
Past versions of "knn impute NA-18.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-18.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-19">
Past versions of "knn impute NA-19.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-19.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-20">
Past versions of "knn impute NA-20.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-20.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>Protein_DF_Abundance_imp_2 &lt;- impute.knn(as.matrix(Protein_DF_Abundance_imp), k = 10, rowmax = 0.4)$data</code></pre>
<pre><code>Cluster size 4178 broken into 4119 59 
Cluster size 4119 broken into 3801 318 
Cluster size 3801 broken into 820 2981 
Done cluster 820 
Cluster size 2981 broken into 2082 899 
Cluster size 2082 broken into 1208 874 
Done cluster 1208 
Done cluster 874 
Done cluster 2082 
Done cluster 899 
Done cluster 2981 
Done cluster 3801 
Done cluster 318 
Done cluster 4119 
Done cluster 59 </code></pre>
<pre class="r"><code>Protein_DF_Abundance_imp_2 %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-21">
Past versions of "knn impute NA-21.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-21.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-22">
Past versions of "knn impute NA-22.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-22.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-23">
Past versions of "knn impute NA-23.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-23.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-24">
Past versions of "knn impute NA-24.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-24.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-25.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-25">
Past versions of "knn impute NA-25.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-25.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-26.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-26">
Past versions of "knn impute NA-26.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-26.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-27.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-27">
Past versions of "knn impute NA-27.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-27.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-28.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-28">
Past versions of "knn impute NA-28.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-28.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-29.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-29">
Past versions of "knn impute NA-29.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-29.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/knn%20impute%20NA-30.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-knn-impute-NA-30">
Past versions of "knn impute NA-30.png"
</button>
</p>
<div id="fig-no-spaces-knn-impute-NA-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/knn impute NA-30.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Protein_DF_Abundance_imp_2 %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
</div>
<div id="log2-transform-for-imputed-data" class="section level1">
<h1>Log2 transform for imputed data</h1>
<pre class="r"><code>log2_norm_counts &lt;- log2(Protein_DF_Abundance_imp_2)
log2_norm_counts %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-1">
Past versions of "Wrangle Data Frame 3-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,1] %&gt;% hist(breaks = 100) </code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-2">
Past versions of "Wrangle Data Frame 3-2.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-2.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-3">
Past versions of "Wrangle Data Frame 3-3.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-3.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-4">
Past versions of "Wrangle Data Frame 3-4.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-4.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-5">
Past versions of "Wrangle Data Frame 3-5.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-5.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-6">
Past versions of "Wrangle Data Frame 3-6.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-6.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-7">
Past versions of "Wrangle Data Frame 3-7.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-7.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-8">
Past versions of "Wrangle Data Frame 3-8.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-8.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-9">
Past versions of "Wrangle Data Frame 3-9.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-9.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-10">
Past versions of "Wrangle Data Frame 3-10.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-10.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-11">
Past versions of "Wrangle Data Frame 3-11.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 3-11.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>log2_norm_counts %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
</div>
<div id="quantile-normalization-for-imputed-data"
class="section level1">
<h1>Quantile Normalization for imputed data</h1>
<pre class="r"><code>normalizedData &lt;- limma::normalizeBetweenArrays(log2_norm_counts[, c(1,2,4,6,7,9)], method = &quot;quantile&quot;)
normalizedData %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-1">
Past versions of "Wrangle Data Frame 4-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,1] %&gt;% hist(breaks = 100) </code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-2">
Past versions of "Wrangle Data Frame 4-2.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-2.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-3">
Past versions of "Wrangle Data Frame 4-3.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-3.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-4">
Past versions of "Wrangle Data Frame 4-4.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-4.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-5">
Past versions of "Wrangle Data Frame 4-5.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-5.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-6">
Past versions of "Wrangle Data Frame 4-6.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-6.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-7">
Past versions of "Wrangle Data Frame 4-7.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Wrangle Data Frame 4-7.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>normalizedData %&gt;% dim()</code></pre>
<pre><code>[1] 4178    6</code></pre>
</div>
<div id="view-distribution-of-imputed-na-values-for-all-samples"
class="section level1">
<h1>View distribution of imputed NA values for all samples</h1>
<pre class="r"><code>na_positions &lt;- is.na(Protein_DF_Abundance_imp)

imputed_dataframe &lt;- normalizedData

imputed_values &lt;- imputed_dataframe[na_positions] %&gt;% na.omit()

hist(imputed_values, breaks = 50, main = &quot;Histogram of Imputed Values&quot;, xlab = &quot;Value&quot;)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Imputed%20value%20distribution-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Imputed-value-distribution-1">
Past versions of "Imputed value distribution-1.png"
</button>
</p>
<div id="fig-no-spaces-Imputed-value-distribution-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/397b327d39f332fa31b98c8b82ac499fe31a9f81/docs/figure/Data_ind_Diffabtrial.Rmd/Imputed value distribution-1.png" target="_blank">397b327</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Imputed value distribution-1.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Histogram of original data (with NA values removed)
hist(log2(Protein_DF_Abundance_imp) %&gt;% rowMeans(na.rm = TRUE), xlim = range(log2(Protein_DF_Abundance_imp) %&gt;% rowMeans(), na.rm = TRUE), breaks = 50, main = &quot;Histogram with Imputed &amp; Non-imputed Values&quot;, xlab = &quot;Value&quot;)

# Histogram of imputed values
hist(imputed_values, breaks = 50, add = TRUE, col = &quot;red&quot;)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/Imputed%20value%20distribution-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Imputed-value-distribution-2">
Past versions of "Imputed value distribution-2.png"
</button>
</p>
<div id="fig-no-spaces-Imputed-value-distribution-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/397b327d39f332fa31b98c8b82ac499fe31a9f81/docs/figure/Data_ind_Diffabtrial.Rmd/Imputed value distribution-2.png" target="_blank">397b327</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5f32731ad2e318c62f6290cf100ae515e63b38f1/docs/figure/Data_ind_Diffabtrial.Rmd/Imputed value distribution-2.png" target="_blank">5f32731</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ruv-iii" class="section level1">
<h1>RUV III</h1>
<pre class="r"><code>Protein_DF_Abundance_imp_2 %&gt;% head()</code></pre>
<pre><code>                   S1       S3       S5       S7       S9         S2       S4
A0A0B4J2A2   979954.0  1470000  1460000  1560000  1700000   940535.0  2020000
A0A0B4J2D5 16000000.0 17000000 15800000 14400000 15500000 15300000.0 13500000
A0A494C071   381352.0   429735   287237   412285   380499   433456.0   540008
A0AVT1        62810.7   135522   159045   123045   123823    49388.1   144860
A0FGR8      1240000.0  1120000   969130  1040000  1020000  1120000.0  1120000
A0JLT2       217977.0   181404   182657   145408   208660   224881.0   202427
                 S6       S8      S10
A0A0B4J2A2  1460000  1390000  1620000
A0A0B4J2D5 13600000 13200000 12600000
A0A494C071   376304   460580   355701
A0AVT1       127488    84684   102306
A0FGR8      1080000  1120000  1060000
A0JLT2       160150   126127   209726</code></pre>
<pre class="r"><code>Protein_DF_Abundance_imp_2 %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code># Convert to DGEList object 
dge &lt;- DGEList(counts = as.matrix(Protein_DF_Abundance_imp_2))
dge %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code># PCA imputed data frame abundance BEFORE RUVIII
prcomp_res &lt;- prcomp(t(Protein_DF_Abundance_imp_2 %&gt;% as.matrix()), center = TRUE, scale. = TRUE)
ggplot2::autoplot(prcomp_res, data = as.data.frame(Meta), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-1">
Past versions of RUVIII-1.png
</button>
</p>
<div id="fig-RUVIII-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-1.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># correlation matrix 
Prot_cor_matrix &lt;- cor(Protein_DF_Abundance_imp_2 %&gt;% as.matrix())

# Heatmap 1. 
pheatmap(Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-2">
Past versions of RUVIII-2.png
</button>
</p>
<div id="fig-RUVIII-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-2.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Adjust column names 
rownames(Prot_cor_matrix) &lt;- Meta$Cond_Ind
colnames(Prot_cor_matrix) &lt;- Meta$Cond_Ind

# Heatmap 2. 
pheatmap(mat = Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE
            )</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-3">
Past versions of RUVIII-3.png
</button>
</p>
<div id="fig-RUVIII-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-3.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Moving on to find the low variance genes across samples 
# for the RUVIII normalization.

# Calculate variability 
cv &lt;- rowSds(dge$counts, useNames = TRUE)/rowMeans(dge$counts)

# Identify the 10% least variable proteins
num_controls &lt;- ceiling(0.05 * nrow(dge$counts))
control_proteins &lt;- names(sort(cv, decreasing = FALSE))[1:num_controls]
control_proteins %&gt;% length()</code></pre>
<pre><code>[1] 209</code></pre>
<pre class="r"><code># Create an index vector
control_proteins_index &lt;- match(control_proteins, rownames(Protein_DF_Abundance_imp_2))
control_proteins_index %&gt;% length()</code></pre>
<pre><code>[1] 209</code></pre>
<pre class="r"><code># Design matrix from metadata
Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>Model_mat &lt;- replicate.matrix(a = Meta[,c(4)])
Model_mat</code></pre>
<pre><code>   Control Dox
1        0   1
2        0   1
3        0   1
4        0   1
5        0   1
6        1   0
7        1   0
8        1   0
9        1   0
10       1   0
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$a
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code>Test &lt;- RUVIII(Y = t(Protein_DF_Abundance_imp_2 %&gt;% log2()), M = Model_mat, ctl =control_proteins_index , k = 1, return.info = TRUE)


New_Y &lt;- Test$newY %&gt;% t() 
New_Y[(New_Y %&gt;% rowMeans()) &lt; 0, ]</code></pre>
<pre><code>     S1 S3 S5 S7 S9 S2 S4 S6 S8 S10</code></pre>
<pre class="r"><code># PCA of RUVIII normalized abundance values:
prcomp_res &lt;- prcomp(t(New_Y %&gt;% as.matrix()), center = TRUE, scale. = TRUE)
ggplot2::autoplot(prcomp_res, data = as.data.frame(Meta), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-4">
Past versions of RUVIII-4.png
</button>
</p>
<div id="fig-RUVIII-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-4.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># correlation matrix 
Prot_cor_matrix &lt;- cor(New_Y %&gt;% as.matrix())

# Heatmap 1. 
pheatmap(Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-5">
Past versions of RUVIII-5.png
</button>
</p>
<div id="fig-RUVIII-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-5.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Adjust column names 
rownames(Prot_cor_matrix) &lt;- Meta$Cond_Ind
colnames(Prot_cor_matrix) &lt;- Meta$Cond_Ind

# Heatmap 2. 
pheatmap(mat = Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE
            )</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-6">
Past versions of RUVIII-6.png
</button>
</p>
<div id="fig-RUVIII-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-6.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y %&gt;% head()</code></pre>
<pre><code>                 S1       S3       S5       S7       S9       S2       S4
A0A0B4J2A2 18.12610 18.08307 18.57100 18.68989 18.78086 18.26364 18.52828
A0A0B4J2D5 23.98654 24.09344 23.97242 23.83785 23.94507 23.91591 23.76128
A0A494C071 17.87490 17.81179 17.41718 17.94732 17.81920 18.13343 18.13633
A0AVT1     13.20610 13.34933 14.34603 14.01164 13.96995 13.16197 13.42496
A0FGR8     20.28742 20.15667 19.93518 20.03640 20.00923 20.13554 20.15701
A0JLT2     18.00880 17.84106 17.77393 17.44129 17.96745 18.02332 18.00132
                 S6       S8      S10
A0A0B4J2A2 18.54053 18.44973 18.60141
A0A0B4J2D5 23.75705 23.71460 23.64962
A0A494C071 17.79542 18.07951 17.68077
A0AVT1     13.98008 13.35925 13.52547
A0FGR8     20.09223 20.14521 20.06755
A0JLT2     17.58893 17.24747 17.99182</code></pre>
<pre class="r"><code>New_Y %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>New_Y %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-7">
Past versions of RUVIII-7.png
</button>
</p>
<div id="fig-RUVIII-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-7.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y %&gt;% na.omit() %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>New_Y[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-8">
Past versions of RUVIII-8.png
</button>
</p>
<div id="fig-RUVIII-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-8.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-9">
Past versions of RUVIII-9.png
</button>
</p>
<div id="fig-RUVIII-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-9.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-10">
Past versions of RUVIII-10.png
</button>
</p>
<div id="fig-RUVIII-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-10.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-11">
Past versions of RUVIII-11.png
</button>
</p>
<div id="fig-RUVIII-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-11.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-12">
Past versions of RUVIII-12.png
</button>
</p>
<div id="fig-RUVIII-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-12.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-13">
Past versions of RUVIII-13.png
</button>
</p>
<div id="fig-RUVIII-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-13.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-14">
Past versions of RUVIII-14.png
</button>
</p>
<div id="fig-RUVIII-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-14.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-15">
Past versions of RUVIII-15.png
</button>
</p>
<div id="fig-RUVIII-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-15.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-16">
Past versions of RUVIII-16.png
</button>
</p>
<div id="fig-RUVIII-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-16.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>New_Y[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-17">
Past versions of RUVIII-17.png
</button>
</p>
<div id="fig-RUVIII-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-17.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Meta_sub&lt;-Meta[c(1,2,4,6,7,9),]
counts_RUV_log2_sub &lt;- New_Y[,c(1,2,4,6,7,9)]
counts_RUV_log2_sub %&gt;% head()</code></pre>
<pre><code>                 S1       S3       S7       S2       S4       S8
A0A0B4J2A2 18.12610 18.08307 18.68989 18.26364 18.52828 18.44973
A0A0B4J2D5 23.98654 24.09344 23.83785 23.91591 23.76128 23.71460
A0A494C071 17.87490 17.81179 17.94732 18.13343 18.13633 18.07951
A0AVT1     13.20610 13.34933 14.01164 13.16197 13.42496 13.35925
A0FGR8     20.28742 20.15667 20.03640 20.13554 20.15701 20.14521
A0JLT2     18.00880 17.84106 17.44129 18.02332 18.00132 17.24747</code></pre>
<pre class="r"><code>counts_RUV_log2_sub %&gt;% dim()</code></pre>
<pre><code>[1] 4178    6</code></pre>
<pre class="r"><code>counts_RUV_log2_sub[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-18">
Past versions of RUVIII-18.png
</button>
</p>
<div id="fig-RUVIII-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-18.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-19">
Past versions of RUVIII-19.png
</button>
</p>
<div id="fig-RUVIII-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-19.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-20">
Past versions of RUVIII-20.png
</button>
</p>
<div id="fig-RUVIII-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-20.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-20.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-21">
Past versions of RUVIII-21.png
</button>
</p>
<div id="fig-RUVIII-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-21.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-21.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-22">
Past versions of RUVIII-22.png
</button>
</p>
<div id="fig-RUVIII-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-22.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-22.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-23">
Past versions of RUVIII-23.png
</button>
</p>
<div id="fig-RUVIII-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-23.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-23.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of RUVIII normalized abundance values:
prcomp_res &lt;- prcomp(t(counts_RUV_log2_sub %&gt;% as.matrix()), center = TRUE, scale. = TRUE)
ggplot2::autoplot(prcomp_res, data = as.data.frame(Meta_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-24">
Past versions of RUVIII-24.png
</button>
</p>
<div id="fig-RUVIII-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-24.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-24.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># correlation matrix 
Prot_cor_matrix &lt;- cor(counts_RUV_log2_sub %&gt;% as.matrix())

# Heatmap 1. 
pheatmap(Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-25.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-25">
Past versions of RUVIII-25.png
</button>
</p>
<div id="fig-RUVIII-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-25.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-25.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Adjust column names 
rownames(Prot_cor_matrix) &lt;- Meta_sub$Cond_Ind
colnames(Prot_cor_matrix) &lt;- Meta_sub$Cond_Ind

# Heatmap 2. 
pheatmap(mat = Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE
            )</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-26.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-26">
Past versions of RUVIII-26.png
</button>
</p>
<div id="fig-RUVIII-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-26.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-26.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm &lt;- limma::normalizeBetweenArrays(counts_RUV_log2_sub, method = &quot;quantile&quot;)
counts_RUV_log2_sub_quantnorm %&gt;% head()</code></pre>
<pre><code>                 S1       S3       S7       S2       S4       S8
A0A0B4J2A2 18.17516 18.19764 18.75051 18.22063 18.44611 18.37436
A0A0B4J2D5 23.89376 24.08493 23.87250 23.97993 23.82959 23.63583
A0A494C071 17.90859 17.92729 18.00677 18.10054 18.02723 18.02111
A0AVT1     13.35950 13.30205 14.10541 13.22825 13.32444 13.34552
A0FGR8     20.29701 20.24292 20.05735 20.11984 20.08767 20.09303
A0JLT2     18.04660 17.95815 17.46046 17.97729 17.88938 17.17767</code></pre>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm %&gt;% dim()</code></pre>
<pre><code>[1] 4178    6</code></pre>
<pre class="r"><code>Meta_sub</code></pre>
<pre><code>   Samples  Ind  Rep    Cond     Cond_Ind
S1      S1 B_77  Bio     Dox     B_77_Dox
S3      S3 C_87  Bio     Dox     C_87_Dox
S7      S7 A_48 Tech     Dox     A_48_Dox
S2      S2 B_77  Bio Control B_77_Control
S4      S4 C_87  Bio Control C_87_Control
S8      S8 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code># 1.
counts_RUV_log2_sub_quantnorm[,1] %&gt;% hist(main = &quot;B_77_Dox&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-27.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-27">
Past versions of RUVIII-27.png
</button>
</p>
<div id="fig-RUVIII-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-27.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-27.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 2.
counts_RUV_log2_sub_quantnorm[,2] %&gt;% hist(main = &quot;C_87_Dox&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-28.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-28">
Past versions of RUVIII-28.png
</button>
</p>
<div id="fig-RUVIII-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5b66212e764d792d40b5d8ea3d2229165283d9ca/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-28.png" target="_blank">5b66212</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-28.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 3.
counts_RUV_log2_sub_quantnorm[,3] %&gt;% hist(main = &quot;A_48_Dox&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-29.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-29">
Past versions of RUVIII-29.png
</button>
</p>
<div id="fig-RUVIII-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-29.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 4.
counts_RUV_log2_sub_quantnorm[,4] %&gt;% hist(main = &quot;B_77_Control&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-30.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-30">
Past versions of RUVIII-30.png
</button>
</p>
<div id="fig-RUVIII-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-30.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 5.
counts_RUV_log2_sub_quantnorm[,5] %&gt;% hist(main = &quot;C_87_Control&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-31.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-31">
Past versions of RUVIII-31.png
</button>
</p>
<div id="fig-RUVIII-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-31.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 6.
counts_RUV_log2_sub_quantnorm[,6] %&gt;% hist(main = &quot;A_48_Control&quot;, breaks = 50)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-32.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-32">
Past versions of RUVIII-32.png
</button>
</p>
<div id="fig-RUVIII-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-32.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of RUVIII normalized abundance values:
prcomp_res &lt;- prcomp(t(counts_RUV_log2_sub_quantnorm %&gt;% as.matrix()), center = TRUE, scale. = TRUE)
ggplot2::autoplot(prcomp_res, data = as.data.frame(Meta_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-33.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-33">
Past versions of RUVIII-33.png
</button>
</p>
<div id="fig-RUVIII-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/4c91815ce583a586f165366016fe3e45cfd58976/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-33.png" target="_blank">4c91815</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># correlation matrix 
Prot_cor_matrix &lt;- cor(counts_RUV_log2_sub_quantnorm %&gt;% as.matrix())

# Heatmap 1. 
pheatmap(Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE)</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-34.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-34">
Past versions of RUVIII-34.png
</button>
</p>
<div id="fig-RUVIII-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-34.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Adjust column names 
rownames(Prot_cor_matrix) &lt;- Meta_sub$Cond_Ind
colnames(Prot_cor_matrix) &lt;- Meta_sub$Cond_Ind

# Heatmap 2. 
pheatmap(mat = Prot_cor_matrix, 
         color = colorRampPalette(c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;))(200),
         display_numbers = TRUE
            )</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-35.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-35">
Past versions of RUVIII-35.png
</button>
</p>
<div id="fig-RUVIII-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-35.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Moving on to the actual linear model 
# Create a DataFrame for the phenoData
phenoData_RUVIII &lt;- DataFrame(Meta)
phenoData_RUVIII &lt;- Meta[c(1,2,4,6,7,9), ]



# View changes 
phenoData_RUVIII</code></pre>
<pre><code>   Samples  Ind  Rep    Cond     Cond_Ind
S1      S1 B_77  Bio     Dox     B_77_Dox
S3      S3 C_87  Bio     Dox     C_87_Dox
S7      S7 A_48 Tech     Dox     A_48_Dox
S2      S2 B_77  Bio Control B_77_Control
S4      S4 C_87  Bio Control C_87_Control
S8      S8 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code># Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~ 0 + Cond, data = phenoData_RUVIII)
design</code></pre>
<pre><code>   CondControl CondDox
S1           0       1
S3           0       1
S7           0       1
S2           1       0
S4           1       0
S8           1       0
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;)


# Get model counts 
Model_counts &lt;- counts_RUV_log2_sub_quantnorm


# Fit model
dupcor &lt;- duplicateCorrelation(Model_counts, design = design, block = phenoData_RUVIII$Ind)

fit &lt;- lmFit(object = as.matrix(Model_counts), block = phenoData_RUVIII$Ind, design = design, correlation = dupcor$consensus.correlation)

fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2, robust = TRUE, trend = TRUE)


# Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        554
NotSig     3011
Up          613</code></pre>
<pre class="r"><code># Toptable summary organized   
toptable_summary_DoxvNorm &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(Model_counts)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary_DoxvNorm$Protein &lt;- rownames(toptable_summary_DoxvNorm)
toptable_summary_DoxvNorm$P.Value %&gt;% hist(breaks = 100)
abline(v = c(0.01,0.025, 0.05, 0.1), col = &quot;red&quot; )</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-36.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-36">
Past versions of RUVIII-36.png
</button>
</p>
<div id="fig-RUVIII-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-36.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>toptable_summary_DoxvNorm %&gt;% head()</code></pre>
<pre><code>           logFC  AveExpr         t      P.Value    adj.P.Val        B Protein
Q15061  2.436903 18.60721  26.00939 1.499311e-07 1.499311e-07 7.414993  Q15061
O76021 -2.224860 21.98800 -18.86036 1.191879e-06 1.191879e-06 6.029757  O76021
P43694 -1.821642 18.81578 -15.98230 3.129665e-06 3.129665e-06 5.276514  P43694
P01130 -1.819689 20.01194 -15.77595 3.483331e-06 3.483331e-06 5.186033  P01130
Q96P11 -1.843458 17.28018 -14.33937 5.735316e-06 5.735316e-06 4.764600  Q96P11
Q6EMK4  1.838032 19.30419  14.22737 6.455771e-06 6.455771e-06 4.660093  Q6EMK4</code></pre>
<pre class="r"><code>toptable_summary_DoxvNorm %&gt;% dim()</code></pre>
<pre><code>[1] 4178    7</code></pre>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary_DoxvNorm &lt;- toptable_summary_DoxvNorm %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)


# 2. Plot 
ggplot(toptable_summary_DoxvNorm)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 7)+
  xlim(-6, 6)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-37.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-RUVIII-37">
Past versions of RUVIII-37.png
</button>
</p>
<div id="fig-RUVIII-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Data_ind_Diffabtrial.Rmd/RUVIII-37.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Make top_proteins your toptable summary 
top_proteins &lt;- toptable_summary_DoxvNorm[1:5, ] %&gt;% rownames()

# Loop through each of the top proteins within the Model_counts
for (i in top_proteins) {
  # Create a boxplot for each protein
  boxplot(Model_counts[i, ] ~ Meta_sub$Cond, 
          main = i, 
          xlab = &quot;Condition&quot;, 
          ylab = &quot;log2-RUVIII-Quantnorm Abundance&quot;, 
          col = c(&quot;lightblue&quot;, &quot;red2&quot;),
          las = 2)  # las = 2 makes the axis labels perpendicular to the axis


}</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-38.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-39.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-40.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-41.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-42.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Comparing to a different cutoff
top_proteins &lt;- toptable_summary_DoxvNorm[(toptable_summary_DoxvNorm$P.Value &lt; 0.05) &amp; (toptable_summary_DoxvNorm$P.Value &gt; 0.0495),] %&gt;% rownames()

# Loop through each of the top proteins within the Model_counts
for (i in top_proteins) {
  # Create a boxplot for each protein
  boxplot(Model_counts[i, ] ~ Meta_sub$Cond, 
          main = i, 
          xlab = &quot;Condition&quot;, 
          ylab = &quot;log2-RUVIII-Quantnorm Abundance&quot;, 
          col = c(&quot;lightblue&quot;, &quot;red2&quot;),
          las = 2)  # las = 2 makes the axis labels perpendicular to the axis


}</code></pre>
<p><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-43.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-44.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-45.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-46.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-47.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-48.png" width="672" style="display: block; margin: auto;" /><img src="figure/Data_ind_Diffabtrial.Rmd/RUVIII-49.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ruv_0.9.7.1                            
 [2] DESeq2_1.36.0                          
 [3] impute_1.70.0                          
 [4] WGCNA_1.72-1                           
 [5] fastcluster_1.2.3                      
 [6] dynamicTreeCut_1.63-1                  
 [7] BioNERO_1.4.2                          
 [8] reshape2_1.4.4                         
 [9] ggridges_0.5.4                         
[10] biomaRt_2.52.0                         
[11] ggvenn_0.1.10                          
[12] UpSetR_1.4.0                           
[13] DOSE_3.22.1                            
[14] variancePartition_1.26.0               
[15] clusterProfiler_4.4.4                  
[16] gplots_3.1.3                           
[17] corrplot_0.92                          
[18] pheatmap_1.0.12                        
[19] qvalue_2.28.0                          
[20] Homo.sapiens_1.3.1                     
[21] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[22] org.Hs.eg.db_3.15.0                    
[23] GO.db_3.15.0                           
[24] OrganismDbi_1.38.1                     
[25] GenomicFeatures_1.48.4                 
[26] AnnotationDbi_1.58.0                   
[27] cluster_2.1.4                          
[28] ggfortify_0.4.16                       
[29] lubridate_1.9.2                        
[30] forcats_1.0.0                          
[31] stringr_1.5.0                          
[32] dplyr_1.1.2                            
[33] purrr_1.0.2                            
[34] readr_2.1.4                            
[35] tidyr_1.3.0                            
[36] tibble_3.2.1                           
[37] ggplot2_3.4.3                          
[38] tidyverse_2.0.0                        
[39] RColorBrewer_1.1-3                     
[40] RUVSeq_1.30.0                          
[41] edgeR_3.38.4                           
[42] limma_3.52.4                           
[43] EDASeq_2.30.0                          
[44] ShortRead_1.54.0                       
[45] GenomicAlignments_1.32.1               
[46] SummarizedExperiment_1.26.1            
[47] MatrixGenerics_1.8.1                   
[48] matrixStats_1.0.0                      
[49] Rsamtools_2.12.0                       
[50] GenomicRanges_1.48.0                   
[51] Biostrings_2.64.1                      
[52] GenomeInfoDb_1.32.4                    
[53] XVector_0.36.0                         
[54] IRanges_2.30.1                         
[55] S4Vectors_0.34.0                       
[56] BiocParallel_1.30.4                    
[57] Biobase_2.56.0                         
[58] BiocGenerics_0.42.0                    
[59] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           geneplotter_1.74.0    
 [40] htmlwidgets_1.6.2      igraph_1.5.1           DBI_1.1.3             
 [43] ggnewscale_0.4.9       backports_1.4.1        annotate_1.74.0       
 [46] aod_1.3.2              deldir_1.0-9           vctrs_0.6.3           
 [49] abind_1.4-5            cachem_1.0.8           withr_2.5.0           
 [52] ggforce_0.4.1          checkmate_2.2.0        treeio_1.20.2         
 [55] prettyunits_1.1.1      ape_5.7-1              lazyeval_0.2.2        
 [58] crayon_1.5.2           genefilter_1.78.0      labeling_0.4.2        
 [61] pkgconfig_2.0.3        tweenr_2.0.2           nlme_3.1-163          
 [64] nnet_7.3-19            rlang_1.1.1            lifecycle_1.0.3       
 [67] downloader_0.4         filelock_1.0.2         BiocFileCache_2.4.0   
 [70] rprojroot_2.0.3        polyclip_1.10-4        graph_1.74.0          
 [73] Matrix_1.5-4.1         aplot_0.2.0            NetRep_1.2.7          
 [76] boot_1.3-28.1          base64enc_0.1-3        GlobalOptions_0.1.2   
 [79] whisker_0.4.1          processx_3.8.2         png_0.1-8             
 [82] viridisLite_0.4.2      rjson_0.2.21           bitops_1.0-7          
 [85] getPass_0.2-2          R.oo_1.25.0            ggnetwork_0.5.12      
 [88] KernSmooth_2.23-22     blob_1.2.4             shape_1.4.6           
 [91] jpeg_0.1-10            gridGraphics_0.5-1     scales_1.2.1          
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.8            
 [97] zlibbioc_1.42.0        compiler_4.2.0         scatterpie_0.2.1      
[100] BiocIO_1.6.0           clue_0.3-64            intergraph_2.0-3      
[103] lme4_1.1-34            cli_3.6.1              patchwork_1.1.3       
[106] ps_1.7.5               htmlTable_2.4.1        Formula_1.2-5         
[109] mgcv_1.9-0             MASS_7.3-60            tidyselect_1.2.0      
[112] stringi_1.7.12         highr_0.10             yaml_2.3.7            
[115] GOSemSim_2.22.0        locfit_1.5-9.8         latticeExtra_0.6-30   
[118] ggrepel_0.9.3          sass_0.4.7             fastmatch_1.1-4       
[121] tools_4.2.0            timechange_0.2.0       parallel_4.2.0        
[124] circlize_0.4.15        rstudioapi_0.15.0      foreign_0.8-84        
[127] foreach_1.5.2          git2r_0.32.0           gridExtra_2.3         
[130] farver_2.1.1           ggraph_2.1.0           digest_0.6.33         
[133] BiocManager_1.30.22    networkD3_0.4          Rcpp_1.0.11           
[136] broom_1.0.5            later_1.3.1            httr_1.4.7            
[139] ComplexHeatmap_2.12.1  GENIE3_1.18.0          Rdpack_2.5            
[142] colorspace_2.1-0       XML_3.99-0.14          fs_1.6.3              
[145] splines_4.2.0          statmod_1.5.0          yulab.utils_0.0.8     
[148] RBGL_1.72.0            tidytree_0.4.5         graphlayouts_1.0.0    
[151] ggplotify_0.1.2        xtable_1.8-4           jsonlite_1.8.7        
[154] nloptr_2.0.3           ggtree_3.4.4           tidygraph_1.2.3       
[157] ggfun_0.1.2            R6_2.5.1               Hmisc_5.1-0           
[160] pillar_1.9.0           htmltools_0.5.6        glue_1.6.2            
[163] fastmap_1.1.1          minqa_1.2.5            codetools_0.2-19      
[166] fgsea_1.22.0           utf8_1.2.3             sva_3.44.0            
[169] lattice_0.21-8         bslib_0.5.1            network_1.18.1        
[172] pbkrtest_0.5.2         curl_5.0.2             gtools_3.9.4          
[175] interp_1.1-4           survival_3.5-7         statnet.common_4.9.0  
[178] rmarkdown_2.24         munsell_0.5.0          GetoptLong_1.0.5      
[181] DO.db_2.9              GenomeInfoDbData_1.2.8 iterators_1.0.14      
[184] gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
