<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-12-08" />

<title>Diff_abundance_231207</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Omar-Johnson/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Diff_abundance_231207</h1>
<h4 class="date">2023-12-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230911code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230911)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230911code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230911)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttree2350e41ea2c909844b9ee0bc9ca7d4789be9b388targetblank2350e41a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Omar-Johnson/myproject/tree/2350e41ea2c909844b9ee0bc9ca7d4789be9b388" target="_blank">2350e41</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttree2350e41ea2c909844b9ee0bc9ca7d4789be9b388targetblank2350e41a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Omar-Johnson/myproject/tree/2350e41ea2c909844b9ee0bc9ca7d4789be9b388" target="_blank">2350e41</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  analysis/Random_bed.R

Unstaged changes:
    Modified:   analysis/Differentially_abundant_proteins_impute.Rmd
    Modified:   analysis/Differentially_abunddant_proteins.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Diffabtrial.Rmd</code>) and HTML
(<code>docs/Diffabtrial.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/2350e41ea2c909844b9ee0bc9ca7d4789be9b388/analysis/Diffabtrial.Rmd" target="_blank">2350e41</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
<td>
Updaate
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/Diffabtrial.html" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/849930ccab38f9734cd18fbf9be31e25a37e6f9b/analysis/Diffabtrial.Rmd" target="_blank">849930c</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
<pre class="r"><code># Load your data
Protein_DF &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Abundance.csv&quot;, header = TRUE)

# Include metadata about samples 
Meta &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Meta.csv&quot;, header = TRUE)</code></pre>
</div>
<div id="wrangle-data-frame" class="section level1">
<h1>Wrangle Data Frame</h1>
<pre class="r"><code># Save the original 
Full_DF &lt;- Protein_DF 


# Subset the columns referring to abundance
Protein_DF_Abundance &lt;- Protein_DF[,c(4, 5:ncol(Protein_DF))] 

# Check for duplicated rows 
# Identify duplicated values
Protein_DF_Abundance$duplicated_name &lt;- duplicated(Protein_DF_Abundance$Accession)

# This will return a logical vector where TRUE indicates the position of duplicates in the column.
# To see only rows with duplicated values, you can subset the dataframe like this:
duplicated_rows &lt;- Protein_DF_Abundance[Protein_DF_Abundance$duplicated_name == TRUE, ]
print(duplicated_rows)</code></pre>
<pre><code>    Accession Abundance..F1..Sample..Dox..n.a Abundance..F3..Sample..Dox..n.a
61     P04264                        3.87e+08                        1.45e+08
276    P35908                              NA                              NA
997    P04040                              NA                              NA
    Abundance..F5..Sample..Dox..Dox Abundance..F7..Sample..Dox..Dox
61                         2.58e+08                        7.07e+08
276                              NA                              NA
997                              NA                              NA
    Abundance..F9..Sample..Dox..Dox Abundance..F2..Control..Veh..n.a
61                         3.17e+08                         1.19e+10
276                              NA                         5.31e+06
997                              NA                               NA
    Abundance..F4..Control..Veh..n.a Abundance..F6..Control..Veh..Veh
61                          1.11e+09                         8.03e+08
276                         1.14e+06                               NA
997                               NA                               NA
    Abundance..F8..Control..Veh..Veh Abundance..F10..Control..Veh..Veh
61                          4.61e+08                          5.45e+08
276                         5.00e+05                          4.90e+05
997                               NA                                NA
    duplicated_name
61             TRUE
276            TRUE
997            TRUE</code></pre>
<pre class="r"><code># We have 3 rows with duplicates, two of which have NA values. We will remove them now since we will go ahead 
# and remove the duplicated rows later on anyways. 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[- c(61, 276, 997), ] 

# Make rownames of the data frame accession IDs
rownames(Protein_DF_Abundance) &lt;- Protein_DF_Abundance$Accession

# Change colnames 
colnames(Protein_DF_Abundance)</code></pre>
<pre><code> [1] &quot;Accession&quot;                         &quot;Abundance..F1..Sample..Dox..n.a&quot;  
 [3] &quot;Abundance..F3..Sample..Dox..n.a&quot;   &quot;Abundance..F5..Sample..Dox..Dox&quot;  
 [5] &quot;Abundance..F7..Sample..Dox..Dox&quot;   &quot;Abundance..F9..Sample..Dox..Dox&quot;  
 [7] &quot;Abundance..F2..Control..Veh..n.a&quot;  &quot;Abundance..F4..Control..Veh..n.a&quot; 
 [9] &quot;Abundance..F6..Control..Veh..Veh&quot;  &quot;Abundance..F8..Control..Veh..Veh&quot; 
[11] &quot;Abundance..F10..Control..Veh..Veh&quot; &quot;duplicated_name&quot;                  </code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[, -c(1, 12)]

# Assuming column names of Protein_DF match with the rows in Meta
rownames(Meta) &lt;- Meta$Samples

colnames(Protein_DF_Abundance) &lt;- Meta$Samples</code></pre>
</div>
<div id="remove-na" class="section level1">
<h1>Remove NA</h1>
<pre class="r"><code>#1.  Find rows with any NA values
rows_with_na &lt;- apply(Protein_DF_Abundance, 1, function(x) any(is.na(x))) %&gt;% which()
 

rows_with_na &lt;- as.numeric(rows_with_na)


# 2. Remove rows with NA values 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[-rows_with_na, ]

# 3. Sanity check 
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
</div>
<div id="log2-transform" class="section level1">
<h1>Log2 transform</h1>
<pre class="r"><code>log2_norm_counts &lt;- log2(Protein_DF_Abundance)
log2_norm_counts %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/Wrangle%20Data%20Frame%203-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-3-1">
Past versions of "Wrangle Data Frame 3-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/Wrangle Data Frame 3-1.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="quantile-normalization" class="section level1">
<h1>Quantile Normalization</h1>
<pre class="r"><code>normalizedData &lt;- limma::normalizeBetweenArrays(log2_norm_counts[, c(1,2,4,6,7,9)], method = &quot;quantile&quot;)
normalizedData %&gt;% rowMeans() %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/Wrangle%20Data%20Frame%204-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Wrangle-Data-Frame-4-1">
Past versions of "Wrangle Data Frame 4-1.png"
</button>
</p>
<div id="fig-no-spaces-Wrangle-Data-Frame-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/Wrangle Data Frame 4-1.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="linear-model---ruvs" class="section level1">
<h1>Linear model - RUVs</h1>
<pre class="r"><code>#  counts need to be integer values and in a numeric matrix 
counts &lt;- as.matrix(normalizedData) 
counts %&gt;% dim()</code></pre>
<pre><code>[1] 3384    6</code></pre>
<pre class="r"><code># Create a DataFrame for the phenoData
phenoData &lt;- DataFrame(Meta)
phenoData_sub &lt;- phenoData[c(1,2,4,6,7,9), ]


# Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~ 0 + Cond , phenoData_sub)
design</code></pre>
<pre><code>   CondControl CondDox
S1           0       1
S3           0       1
S7           0       1
S2           1       0
S4           1       0
S8           1       0
attr(,&quot;assign&quot;)
[1] 1 1
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;)


# Fit model
dupcor &lt;- duplicateCorrelation(counts , design, block = phenoData_sub$Ind)
fit &lt;- lmFit(object = counts, design = design, block = phenoData_sub$Ind, correlation = dupcor$consensus)
fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2 )


# Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        325
NotSig     2788
Up          271</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(normalizedData)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)
toptable_summary$P.Value %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-1">
Past versions of "upcor on 10 samples-1.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-1.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of log2-quantile normalized-RUVs values:
prcomp_res &lt;- prcomp(t(normalizedData %&gt;% as.matrix()), center = TRUE)

ggplot2::autoplot(prcomp_res, data = as.data.frame(phenoData_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-2">
Past versions of "upcor on 10 samples-2.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-2.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary &lt;- toptable_summary %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)

# 2. Plot 
ggplot(toptable_summary)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 10)+
  xlim(-7, 7)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-3">
Past versions of "upcor on 10 samples-3.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-3.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Other fits 
fit3 &lt;- eBayes(fit2, robust = TRUE, trend = TRUE)</code></pre>
<pre><code>Warning: 3 very small variances detected, have been offset away from zero</code></pre>
<pre class="r"><code># Summarize
results_summary &lt;- decideTests(fit3, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        335
NotSig     2778
Up          271</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit3, coef = &quot;DoxvNorm&quot;,number = (nrow(normalizedData)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)
toptable_summary$P.Value %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-4">
Past versions of "upcor on 10 samples-4.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-4.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of log2-quantile normalized-RUVs values:
prcomp_res &lt;- prcomp(t(normalizedData %&gt;% as.matrix()), center = TRUE)


ggplot2::autoplot(prcomp_res, data = as.data.frame(phenoData_sub), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-5">
Past versions of "upcor on 10 samples-5.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-5.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary &lt;- toptable_summary %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)


# 2. Plot 
ggplot(toptable_summary)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 10)+
  xlim(-7, 7)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Diffabtrial.Rmd/upcor%20on%2010%20samples-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upcor-on-10-samples-6">
Past versions of "upcor on 10 samples-6.png"
</button>
</p>
<div id="fig-no-spaces-upcor-on-10-samples-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/upcor on 10 samples-6.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="linear-model-ruvs" class="section level1">
<h1>Linear model + RUVs</h1>
<pre class="r"><code># 1. Get non log abundance values 
Protein_DF_Abundance %&gt;% head()</code></pre>
<pre><code>             S1       S3       S5       S7       S9       S2       S4       S6
P12883 5.25e+10 7.41e+10 5.28e+10 6.82e+10 6.43e+10 4.33e+10 7.41e+10 7.00e+10
P13533 1.00e+09 1.47e+09 5.11e+09 4.25e+09 2.62e+09 1.50e+09 1.17e+09 2.42e+09
Q8WZ42 1.39e+10 2.08e+10 2.19e+10 2.49e+10 2.34e+10 1.29e+10 2.25e+10 2.52e+10
P68032 3.92e+10 4.11e+10 3.76e+10 3.54e+10 3.69e+10 3.54e+10 2.96e+10 3.56e+10
P68133 1.72e+07 5.81e+07 3.85e+07 5.07e+07 6.19e+07 1.64e+07 3.44e+07 5.42e+07
P60709 4.02e+09 8.43e+08 2.04e+09 1.62e+09 1.61e+09 4.16e+09 6.45e+08 1.92e+09
             S8      S10
P12883 5.64e+10 5.26e+10
P13533 3.67e+09 3.27e+09
Q8WZ42 2.46e+10 2.21e+10
P68032 2.98e+10 2.91e+10
P68133 3.36e+07 4.05e+07
P60709 1.81e+09 1.56e+09</code></pre>
<pre class="r"><code># 2. Generate RUVs related objects (differences_all and set_all)
Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>genes_all &lt;- rownames(Protein_DF_Abundance)
x_all &lt;- c(1,2,3,3,3,4,5,6,6,6)
differences_all &lt;- makeGroups(x_all)


set_all &lt;- newSeqExpressionSet(as.matrix(Protein_DF_Abundance),phenoData = data.frame(x_all, row.names=colnames(Protein_DF_Abundance)))




# 3. Get RUV weights using all 10 samples
set_all_RUV &lt;- RUVSeq::RUVs(set_all, genes_all, k=1, differences_all, isLog = FALSE)


# 4. Subset counts by the 6 columns of interest, Metadata by 6 rows of interest, RUV weights vector to the 6 values of interest
#  counts need to be integer values and in a numeric matrix 
counts_RUVs &lt;- as.matrix(Protein_DF_Abundance) 

# Create a DataFrame for the phenoData
phenoData_RUVs &lt;- DataFrame(Meta)
phenoData_RUVs &lt;- phenoData_RUVs[c(1,2,4,6,7,9), ]

counts_RUVs &lt;- counts_RUVs[,c(1,2,4,6,7,9)]

counts_RUVs %&gt;% dim()</code></pre>
<pre><code>[1] 3384    6</code></pre>
<pre class="r"><code>phenoData_RUVs %&gt;% dim()</code></pre>
<pre><code>[1] 6 5</code></pre>
<pre class="r"><code>RUV_1 &lt;- set_all_RUV$W_1
phenoData_RUVs$RUV_1 &lt;- RUV_1[c(1,2,4,6,7,9)]


# View changes 
phenoData_RUVs %&gt;% head()</code></pre>
<pre><code>DataFrame with 6 rows and 6 columns
       Samples         Ind         Rep        Cond     Cond_Ind     RUV_1
   &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;  &lt;character&gt; &lt;numeric&gt;
S1          S1        B_77         Bio         Dox     B_77_Dox  -5.51097
S3          S3        C_87         Bio         Dox     C_87_Dox  -5.69038
S7          S7        A_48        Tech         Dox     A_48_Dox  -5.08594
S2          S2        B_77         Bio     Control B_77_Control  -5.51861
S4          S4        C_87         Bio     Control C_87_Control  -5.67646
S8          S8        A_48        Tech     Control A_48_Control  -5.91862</code></pre>
<pre class="r"><code>counts_RUVs %&gt;% head()</code></pre>
<pre><code>             S1       S3       S7       S2       S4       S8
P12883 5.25e+10 7.41e+10 6.82e+10 4.33e+10 7.41e+10 5.64e+10
P13533 1.00e+09 1.47e+09 4.25e+09 1.50e+09 1.17e+09 3.67e+09
Q8WZ42 1.39e+10 2.08e+10 2.49e+10 1.29e+10 2.25e+10 2.46e+10
P68032 3.92e+10 4.11e+10 3.54e+10 3.54e+10 2.96e+10 2.98e+10
P68133 1.72e+07 5.81e+07 5.07e+07 1.64e+07 3.44e+07 3.36e+07
P60709 4.02e+09 8.43e+08 1.62e+09 4.16e+09 6.45e+08 1.81e+09</code></pre>
<pre class="r"><code># 5. Take the info from 4 and assign it to a new object
counts_RUV_nolog &lt;- assayData(set_all_RUV)$counts
counts_RUV_nolog %&gt;% head()</code></pre>
<pre><code>             S1       S3       S5       S7       S9       S2       S4       S6
P12883 5.25e+10 7.41e+10 5.28e+10 6.82e+10 6.43e+10 4.33e+10 7.41e+10 7.00e+10
P13533 1.00e+09 1.47e+09 5.11e+09 4.25e+09 2.62e+09 1.50e+09 1.17e+09 2.42e+09
Q8WZ42 1.39e+10 2.08e+10 2.19e+10 2.49e+10 2.34e+10 1.29e+10 2.25e+10 2.52e+10
P68032 3.92e+10 4.11e+10 3.76e+10 3.54e+10 3.69e+10 3.54e+10 2.96e+10 3.56e+10
P68133 1.72e+07 5.81e+07 3.85e+07 5.07e+07 6.19e+07 1.64e+07 3.44e+07 5.42e+07
P60709 4.02e+09 8.43e+08 2.04e+09 1.62e+09 1.61e+09 4.16e+09 6.45e+08 1.92e+09
             S8      S10
P12883 5.64e+10 5.26e+10
P13533 3.67e+09 3.27e+09
Q8WZ42 2.46e+10 2.21e+10
P68032 2.98e+10 2.91e+10
P68133 3.36e+07 4.05e+07
P60709 1.81e+09 1.56e+09</code></pre>
<pre class="r"><code>counts_RUV_nolog %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
<pre class="r"><code>counts_RUV_nolog[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-1">
Past versions of "MW suggested workflow-1.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-1.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-2">
Past versions of "MW suggested workflow-2.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-2.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-3">
Past versions of "MW suggested workflow-3.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-3.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-4">
Past versions of "MW suggested workflow-4.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-4.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-5">
Past versions of "MW suggested workflow-5.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-5.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-6">
Past versions of "MW suggested workflow-6.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-6.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-7">
Past versions of "MW suggested workflow-7.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-7.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-8">
Past versions of "MW suggested workflow-8.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-8.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-9">
Past versions of "MW suggested workflow-9.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-9.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_nolog[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-10">
Past versions of "MW suggested workflow-10.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-10.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized &lt;- assayData(set_all_RUV)$normalizedCounts
counts_RUV_normalized %&gt;% head()</code></pre>
<pre><code>                S1          S3          S5          S7          S9          S2
P12883 45337542962 63685880830 45126164890 59565663253 54642064882 37385080434
P13533  3565077374  5462082782 19930829758 13736619415 10737042018  5357042250
Q8WZ42 13032583533 19461130062 20440011127 23462448749 21785328324 12093907902
P68032 23106295847 23812947580 21350141224 21734611833 20526180407 20851124975
P68133    11291597    37622973    24533766    34382003    38804593    10760131
P60709  1869420388   382369912   898668323   799173901   688427329  1932473604
                S4          S6          S8         S10
P12883 63709478113 59654834496 48179881020 45822356691
P13533  4333436814  9676674167 14373819178 10807981320
Q8WZ42 21055129503 23490366118 22955184919 20800523884
P68032 17172867696 20006427755 16891959606 17701326282
P68133    22299598    34255093    21381900    27262452
P60709   293126904   833223906   795358069   759285096</code></pre>
<pre class="r"><code>counts_RUV_normalized %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
<pre class="r"><code>counts_RUV_normalized[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-11">
Past versions of "MW suggested workflow-11.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-11.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-12">
Past versions of "MW suggested workflow-12.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-12.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-13">
Past versions of "MW suggested workflow-13.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-13.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-14">
Past versions of "MW suggested workflow-14.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-14.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-15">
Past versions of "MW suggested workflow-15.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-15.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-16">
Past versions of "MW suggested workflow-16.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-16.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-17">
Past versions of "MW suggested workflow-17.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-17.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-18">
Past versions of "MW suggested workflow-18.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-18.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-19">
Past versions of "MW suggested workflow-19.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-19.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_normalized[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-20">
Past versions of "MW suggested workflow-20.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-20.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2 &lt;- assayData(set_all_RUV)$normalizedCounts %&gt;% log2()
counts_RUV_log2 %&gt;% head()</code></pre>
<pre><code>             S1       S3       S5       S7       S9       S2       S4       S6
P12883 35.39999 35.89025 35.39325 35.79376 35.66929 35.12174 35.89079 35.79592
P13533 31.73129 32.34680 34.21428 33.67731 33.32188 32.31879 32.01286 33.17186
Q8WZ42 33.60140 34.17988 34.25068 34.44963 34.34264 33.49356 34.29345 34.45135
P68032 34.42757 34.47103 34.31353 34.33928 34.25675 34.27941 33.99941 34.21974
P68133 23.42875 25.16511 24.54827 25.03515 25.20972 23.35919 24.41051 25.02982
P60709 30.79994 28.51039 29.74321 29.57393 29.35873 30.84780 28.12695 29.63413
             S8      S10
P12883 35.48771 35.41533
P13533 33.74272 33.33138
Q8WZ42 34.41810 34.27590
P68032 33.97562 34.04314
P68133 24.34989 24.70041
P60709 29.56703 29.50007</code></pre>
<pre class="r"><code>counts_RUV_log2 %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
<pre class="r"><code>counts_RUV_log2[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-21">
Past versions of "MW suggested workflow-21.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-21.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-22">
Past versions of "MW suggested workflow-22.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-22.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-23">
Past versions of "MW suggested workflow-23.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-23.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-24">
Past versions of "MW suggested workflow-24.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-24.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-25.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-25">
Past versions of "MW suggested workflow-25.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-25.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-26.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-26">
Past versions of "MW suggested workflow-26.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-26.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,7] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-27.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-27">
Past versions of "MW suggested workflow-27.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-27.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,8] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-28.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-28">
Past versions of "MW suggested workflow-28.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-28.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,9] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-29.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-29">
Past versions of "MW suggested workflow-29.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-29.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2[,10] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-30.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-30">
Past versions of "MW suggested workflow-30.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-30.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub &lt;- counts_RUV_log2[,c(1,2,4,6,7,9)]
counts_RUV_log2_sub %&gt;% head()</code></pre>
<pre><code>             S1       S3       S7       S2       S4       S8
P12883 35.39999 35.89025 35.79376 35.12174 35.89079 35.48771
P13533 31.73129 32.34680 33.67731 32.31879 32.01286 33.74272
Q8WZ42 33.60140 34.17988 34.44963 33.49356 34.29345 34.41810
P68032 34.42757 34.47103 34.33928 34.27941 33.99941 33.97562
P68133 23.42875 25.16511 25.03515 23.35919 24.41051 24.34989
P60709 30.79994 28.51039 29.57393 30.84780 28.12695 29.56703</code></pre>
<pre class="r"><code>counts_RUV_log2_sub %&gt;% dim()</code></pre>
<pre><code>[1] 3384    6</code></pre>
<pre class="r"><code>counts_RUV_log2_sub[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-31.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-31">
Past versions of "MW suggested workflow-31.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-31.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-32.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-32">
Past versions of "MW suggested workflow-32.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-32.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-33.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-33">
Past versions of "MW suggested workflow-33.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-33.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-34.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-34">
Past versions of "MW suggested workflow-34.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-34.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,5] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-35.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-35">
Past versions of "MW suggested workflow-35.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-35.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub[,6] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-36.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-36">
Past versions of "MW suggested workflow-36.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-36.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm &lt;- limma::normalizeBetweenArrays(counts_RUV_log2_sub, method = &quot;quantile&quot;)
counts_RUV_log2_sub_quantnorm %&gt;% head()</code></pre>
<pre><code>             S1       S3       S7       S2       S4       S8
P12883 35.19711 35.99097 35.99097 35.19711 35.19711 35.99097
P13533 31.76669 32.49737 33.42168 32.07787 32.07787 33.42168
Q8WZ42 33.96387 34.80375 34.38638 33.42168 33.96387 34.80375
P68032 34.38638 35.19711 33.96387 33.96387 33.71671 33.96387
P68133 23.41004 25.33452 25.06394 23.26259 24.51328 24.13988
P60709 30.80305 28.66952 29.75892 30.63208 28.24612 29.53613</code></pre>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm %&gt;% dim()</code></pre>
<pre><code>[1] 3384    6</code></pre>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm[,1] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-37.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-37">
Past versions of "MW suggested workflow-37.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-37.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm[,2] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-38.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-38">
Past versions of "MW suggested workflow-38.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-38.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm[,3] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-39.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-39">
Past versions of "MW suggested workflow-39.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-39.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm[,4] %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-40.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-40">
Past versions of "MW suggested workflow-40.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-40.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>counts_RUV_log2_sub_quantnorm[,5] %&gt;% hist(breaks = 100)
counts_RUV_log2_sub_quantnorm[,6] %&gt;% hist(breaks = 100)




# Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~ 0 + Cond + RUV_1, data = phenoData_RUVs)
design</code></pre>
<pre><code>   CondControl CondDox     RUV_1
S1           0       1 -5.510972
S3           0       1 -5.690375
S7           0       1 -5.085941
S2           1       0 -5.518607
S4           1       0 -5.676456
S8           1       0 -5.918624
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;, &quot;RUV_1&quot;)


# Get model counts 
Model_counts &lt;- counts_RUV_log2_sub_quantnorm


# Fit model
dupcor &lt;- duplicateCorrelation(as.matrix(Model_counts), design, block = phenoData_RUVs$Ind)
fit &lt;- lmFit(object = as.matrix(Model_counts), design = design, block = phenoData_RUVs$Ind, correlation = dupcor$consensus)
fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2)


# Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        338
NotSig     2784
Up          260</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(Model_counts)), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)
toptable_summary$P.Value %&gt;% hist(breaks = 100)</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-41.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-41">
Past versions of "MW suggested workflow-41.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-41.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PCA of log2-quantile normalized-RUVs values:
Model_counts_PCA &lt;- na.omit(Model_counts)
sum(is.na(Model_counts_PCA))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>Model_counts_PCA %&gt;% dim()</code></pre>
<pre><code>[1] 3382    6</code></pre>
<pre class="r"><code>prcomp_res &lt;- prcomp(t(as.matrix(Model_counts_PCA)))
ggplot2::autoplot(prcomp_res, data = as.data.frame(phenoData_RUVs), colour = &quot;Cond&quot;, shape = &quot;Ind&quot;, size =4)+
  theme_bw()</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-42.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-42">
Past versions of "MW suggested workflow-42.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-42" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-42.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Volcano plots 
# 1. Create a column to threshold P-values 
toptable_summary &lt;- toptable_summary %&gt;%  mutate(threshold_P = P.Value &lt; 0.05)


# 2. Plot 
ggplot(toptable_summary)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 6)+
  xlim(-7, 7)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/Diffabtrial.Rmd/MW%20suggested%20workflow-43.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MW-suggested-workflow-43">
Past versions of "MW suggested workflow-43.png"
</button>
</p>
<div id="fig-no-spaces-MW-suggested-workflow-43" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/bf63ae40c7f06a012b36b5e63822bdb1f1463990/docs/figure/Diffabtrial.Rmd/MW suggested workflow-43.png" target="_blank">bf63ae4</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] impute_1.70.0                          
 [2] WGCNA_1.72-1                           
 [3] fastcluster_1.2.3                      
 [4] dynamicTreeCut_1.63-1                  
 [5] BioNERO_1.4.2                          
 [6] reshape2_1.4.4                         
 [7] ggridges_0.5.4                         
 [8] biomaRt_2.52.0                         
 [9] ggvenn_0.1.10                          
[10] UpSetR_1.4.0                           
[11] DOSE_3.22.1                            
[12] variancePartition_1.26.0               
[13] clusterProfiler_4.4.4                  
[14] pheatmap_1.0.12                        
[15] qvalue_2.28.0                          
[16] Homo.sapiens_1.3.1                     
[17] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[18] org.Hs.eg.db_3.15.0                    
[19] GO.db_3.15.0                           
[20] OrganismDbi_1.38.1                     
[21] GenomicFeatures_1.48.4                 
[22] AnnotationDbi_1.58.0                   
[23] cluster_2.1.4                          
[24] ggfortify_0.4.16                       
[25] lubridate_1.9.2                        
[26] forcats_1.0.0                          
[27] stringr_1.5.0                          
[28] dplyr_1.1.2                            
[29] purrr_1.0.2                            
[30] readr_2.1.4                            
[31] tidyr_1.3.0                            
[32] tibble_3.2.1                           
[33] ggplot2_3.4.3                          
[34] tidyverse_2.0.0                        
[35] RColorBrewer_1.1-3                     
[36] RUVSeq_1.30.0                          
[37] edgeR_3.38.4                           
[38] limma_3.52.4                           
[39] EDASeq_2.30.0                          
[40] ShortRead_1.54.0                       
[41] GenomicAlignments_1.32.1               
[42] SummarizedExperiment_1.26.1            
[43] MatrixGenerics_1.8.1                   
[44] matrixStats_1.0.0                      
[45] Rsamtools_2.12.0                       
[46] GenomicRanges_1.48.0                   
[47] Biostrings_2.64.1                      
[48] GenomeInfoDb_1.32.4                    
[49] XVector_0.36.0                         
[50] IRanges_2.30.1                         
[51] S4Vectors_0.34.0                       
[52] BiocParallel_1.30.4                    
[53] Biobase_2.56.0                         
[54] BiocGenerics_0.42.0                    
[55] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           htmlwidgets_1.6.2     
 [40] igraph_1.5.1           DBI_1.1.3              ggnewscale_0.4.9      
 [43] backports_1.4.1        annotate_1.74.0        aod_1.3.2             
 [46] deldir_1.0-9           vctrs_0.6.3            abind_1.4-5           
 [49] cachem_1.0.8           withr_2.5.0            ggforce_0.4.1         
 [52] checkmate_2.2.0        treeio_1.20.2          prettyunits_1.1.1     
 [55] ape_5.7-1              lazyeval_0.2.2         crayon_1.5.2          
 [58] genefilter_1.78.0      labeling_0.4.2         pkgconfig_2.0.3       
 [61] tweenr_2.0.2           nlme_3.1-163           nnet_7.3-19           
 [64] rlang_1.1.1            lifecycle_1.0.3        downloader_0.4        
 [67] filelock_1.0.2         BiocFileCache_2.4.0    rprojroot_2.0.3       
 [70] polyclip_1.10-4        graph_1.74.0           Matrix_1.5-4.1        
 [73] aplot_0.2.0            NetRep_1.2.7           boot_1.3-28.1         
 [76] base64enc_0.1-3        GlobalOptions_0.1.2    whisker_0.4.1         
 [79] processx_3.8.2         png_0.1-8              viridisLite_0.4.2     
 [82] rjson_0.2.21           bitops_1.0-7           getPass_0.2-2         
 [85] R.oo_1.25.0            ggnetwork_0.5.12       KernSmooth_2.23-22    
 [88] blob_1.2.4             shape_1.4.6            jpeg_0.1-10           
 [91] gridGraphics_0.5-1     scales_1.2.1           memoise_2.0.1         
 [94] magrittr_2.0.3         plyr_1.8.8             gplots_3.1.3          
 [97] zlibbioc_1.42.0        compiler_4.2.0         scatterpie_0.2.1      
[100] BiocIO_1.6.0           clue_0.3-64            intergraph_2.0-3      
[103] lme4_1.1-34            cli_3.6.1              patchwork_1.1.3       
[106] ps_1.7.5               htmlTable_2.4.1        Formula_1.2-5         
[109] mgcv_1.9-0             MASS_7.3-60            tidyselect_1.2.0      
[112] stringi_1.7.12         highr_0.10             yaml_2.3.7            
[115] GOSemSim_2.22.0        locfit_1.5-9.8         latticeExtra_0.6-30   
[118] ggrepel_0.9.3          sass_0.4.7             fastmatch_1.1-4       
[121] tools_4.2.0            timechange_0.2.0       parallel_4.2.0        
[124] circlize_0.4.15        rstudioapi_0.15.0      foreign_0.8-84        
[127] foreach_1.5.2          git2r_0.32.0           gridExtra_2.3         
[130] farver_2.1.1           ggraph_2.1.0           digest_0.6.33         
[133] BiocManager_1.30.22    networkD3_0.4          Rcpp_1.0.11           
[136] broom_1.0.5            later_1.3.1            httr_1.4.7            
[139] ComplexHeatmap_2.12.1  GENIE3_1.18.0          Rdpack_2.5            
[142] colorspace_2.1-0       XML_3.99-0.14          fs_1.6.3              
[145] splines_4.2.0          statmod_1.5.0          yulab.utils_0.0.8     
[148] RBGL_1.72.0            tidytree_0.4.5         graphlayouts_1.0.0    
[151] ggplotify_0.1.2        xtable_1.8-4           jsonlite_1.8.7        
[154] nloptr_2.0.3           ggtree_3.4.4           tidygraph_1.2.3       
[157] ggfun_0.1.2            R6_2.5.1               Hmisc_5.1-0           
[160] pillar_1.9.0           htmltools_0.5.6        glue_1.6.2            
[163] fastmap_1.1.1          minqa_1.2.5            codetools_0.2-19      
[166] fgsea_1.22.0           utf8_1.2.3             sva_3.44.0            
[169] lattice_0.21-8         bslib_0.5.1            network_1.18.1        
[172] pbkrtest_0.5.2         curl_5.0.2             gtools_3.9.4          
[175] interp_1.1-4           survival_3.5-7         statnet.common_4.9.0  
[178] rmarkdown_2.24         munsell_0.5.0          GetoptLong_1.0.5      
[181] DO.db_2.9              GenomeInfoDbData_1.2.8 iterators_1.0.14      
[184] gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
