<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-12-06" />

<title>GO_Reactome_enrich_Imputed_proteins</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Omar-Johnson/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GO_Reactome_enrich_Imputed_proteins</h1>
<h4 class="date">2023-12-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230911code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230911)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230911code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230911)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttree5a5048dbd78eff535f9cba368139c512c9554161targetblank5a5048da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Omar-Johnson/myproject/tree/5a5048dbd78eff535f9cba368139c512c9554161" target="_blank">5a5048d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttree5a5048dbd78eff535f9cba368139c512c9554161targetblank5a5048da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Omar-Johnson/myproject/tree/5a5048dbd78eff535f9cba368139c512c9554161" target="_blank">5a5048d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/GO_Reactome_enrich_Imputed_proteins.Rmd</code>) and HTML
(<code>docs/GO_Reactome_enrich_Imputed_proteins.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/5a5048dbd78eff535f9cba368139c512c9554161/analysis/GO_Reactome_enrich_Imputed_proteins.Rmd" target="_blank">5a5048d</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-06
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="added-funtions" class="section level1">
<h1>Added funtions</h1>
<pre class="r"><code>remove_rows_with_nas &lt;- function(data_frame) {
  data_frame[rowSums(is.na(data_frame)) &lt; 5, ]
}</code></pre>
</div>
<div id="steps" class="section level1">
<h1>Steps</h1>
<pre class="r"><code>#* 1. Take Abundance values/counts 
#* 2. Remove NA values from Abundance values 
#* 3. Apply log2 transform to Abundance values 
#* 4. Quantile normalize abundance values 
#* 5. RUVs applied to account for technical variation 
#* 6. Use Limma to model differential protein abundaance </code></pre>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
<pre class="r"><code># Load your data
Protein_DF &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Abundance.csv&quot;, header = TRUE)

# Include metadata about samples 
Meta &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Meta.csv&quot;, header = TRUE)</code></pre>
</div>
<div id="wrangle-data-frame" class="section level1">
<h1>Wrangle Data Frame</h1>
<pre class="r"><code># Save the original 
Full_DF &lt;- Protein_DF 


# Subset the columns referring to abundance
Protein_DF_Abundance &lt;- Protein_DF[,c(4, 5:ncol(Protein_DF))] 

# Check for duplicated rows 
# Identify duplicated values
Protein_DF_Abundance$duplicated_name &lt;- duplicated(Protein_DF_Abundance$Accession)

# This will return a logical vector where TRUE indicates the position of duplicates in the column.
# To see only rows with duplicated values, you can subset the dataframe like this:
duplicated_rows &lt;- Protein_DF_Abundance[Protein_DF_Abundance$duplicated_name == TRUE, ]
print(duplicated_rows)</code></pre>
<pre><code>    Accession Abundance..F1..Sample..Dox..n.a Abundance..F3..Sample..Dox..n.a
61     P04264                        3.87e+08                        1.45e+08
276    P35908                              NA                              NA
997    P04040                              NA                              NA
    Abundance..F5..Sample..Dox..Dox Abundance..F7..Sample..Dox..Dox
61                         2.58e+08                        7.07e+08
276                              NA                              NA
997                              NA                              NA
    Abundance..F9..Sample..Dox..Dox Abundance..F2..Control..Veh..n.a
61                         3.17e+08                         1.19e+10
276                              NA                         5.31e+06
997                              NA                               NA
    Abundance..F4..Control..Veh..n.a Abundance..F6..Control..Veh..Veh
61                          1.11e+09                         8.03e+08
276                         1.14e+06                               NA
997                               NA                               NA
    Abundance..F8..Control..Veh..Veh Abundance..F10..Control..Veh..Veh
61                          4.61e+08                          5.45e+08
276                         5.00e+05                          4.90e+05
997                               NA                                NA
    duplicated_name
61             TRUE
276            TRUE
997            TRUE</code></pre>
<pre class="r"><code># We have 3 rows with duplicates, two of which have NA values. We will remove them now since we will go ahead 
# and remove the duplicated rows later on anyways. 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[- c(61, 276, 997), ] 

# Make rownames of the data frame accession IDs
rownames(Protein_DF_Abundance) &lt;- Protein_DF_Abundance$Accession

# Change colnames 
colnames(Protein_DF_Abundance)</code></pre>
<pre><code> [1] &quot;Accession&quot;                         &quot;Abundance..F1..Sample..Dox..n.a&quot;  
 [3] &quot;Abundance..F3..Sample..Dox..n.a&quot;   &quot;Abundance..F5..Sample..Dox..Dox&quot;  
 [5] &quot;Abundance..F7..Sample..Dox..Dox&quot;   &quot;Abundance..F9..Sample..Dox..Dox&quot;  
 [7] &quot;Abundance..F2..Control..Veh..n.a&quot;  &quot;Abundance..F4..Control..Veh..n.a&quot; 
 [9] &quot;Abundance..F6..Control..Veh..Veh&quot;  &quot;Abundance..F8..Control..Veh..Veh&quot; 
[11] &quot;Abundance..F10..Control..Veh..Veh&quot; &quot;duplicated_name&quot;                  </code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[, -c(1, 12)]

# Assuming column names of Protein_DF match with the rows in Meta
rownames(Meta) &lt;- Meta$Samples

colnames(Protein_DF_Abundance) &lt;- Meta$Samples</code></pre>
</div>
<div id="remove-na-values" class="section level1">
<h1>Remove NA values</h1>
<pre class="r"><code>#1.  Find rows with any NA values
rows_with_na &lt;- apply(Protein_DF_Abundance, 1, function(x) any(is.na(x))) %&gt;% which()
 

rows_with_na &lt;- as.numeric(rows_with_na)
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4501   10</code></pre>
<pre class="r"><code># 2. Remove rows with NA values 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[-rows_with_na, ]

# 3. Sanity check 
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
</div>
<div id="log2-transform" class="section level1">
<h1>Log2 transform</h1>
<pre class="r"><code>log2_norm_counts &lt;- log2(Protein_DF_Abundance)</code></pre>
</div>
<div id="quantile-normalization" class="section level1">
<h1>Quantile Normalization</h1>
<pre class="r"><code>normalizedData &lt;- limma::normalizeBetweenArrays(log2_norm_counts, method = &quot;quantile&quot;)</code></pre>
</div>
<div id="prepare-data-object-for-ruvs" class="section level1">
<h1>Prepare Data Object for RUVs</h1>
<pre class="r"><code>#  counts need to be integer values and in a numeric matrix 
counts &lt;- as.matrix(normalizedData) 

# Create a DataFrame for the phenoData
phenoData &lt;- DataFrame(Meta)

# Now create the RangedSummarizedExperiment necessary for RUVs input 
set &lt;- SummarizedExperiment(assays =  normalizedData, metadata =phenoData)</code></pre>
</div>
<div id="ruvs-matrix" class="section level1">
<h1>RUVs Matrix</h1>
<pre class="r"><code># Generate a background matrix where each set of replicates are in a row
# and null spaces where there are replicate mismatches get the value -1. 
# This will be used in the RUVs function, which is the typeII method to
# remove unwanted variation in the RUVs package. This will be used for all 
# RUVs permutations where the number of variation factors will be changed. 
scIdx &lt;- matrix(-1, nrow = 6, ncol = 3)
scIdx[1, 1:3] &lt;- c(which(colnames(counts) %in% c(&quot;S6&quot;, &quot;S8&quot;, &quot;S10&quot;))) # Control group- Technical reps Ind. 48-1
scIdx[2, 1:3] &lt;- c(which(colnames(counts) %in% c(&quot;S5&quot;, &quot;S7&quot;, &quot;S9&quot;))) # Dox group- Technical reps Ind. 48-1
scIdx[3, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S2&quot;))) # Control group- Bio reps Ind. 77-1
scIdx[4, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S1&quot;))) # Dox group- 77-1
scIdx[5, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S4&quot;))) # Control group- Bio reps Ind. 87-1
scIdx[6, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S3&quot;))) # Dox group- 87-1
scIdx</code></pre>
<pre><code>     [,1] [,2] [,3]
[1,]    8    9   10
[2,]    3    4    5
[3,]    6   -1   -1
[4,]    1   -1   -1
[5,]    7   -1   -1
[6,]    2   -1   -1</code></pre>
</div>
<div id="linear-modelling" class="section level1">
<h1>Linear Modelling</h1>
<pre class="r"><code># Apply RUVs function from RUVSeq and add to set object
set &lt;- RUVSeq::RUVs(x = normalizedData, k = 1, scIdx = scIdx, isLog = TRUE)
RUV_1 &lt;- set$W[,1]

set$W</code></pre>
<pre><code>            W_1
S1   0.14326547
S3  -0.08146518
S5  -0.25579041
S7   0.54847999
S9  -0.50097929
S2   0.15639944
S4  -0.04767015
S6  -0.35626503
S8  -0.19216750
S10  0.48449144</code></pre>
<pre class="r"><code>phenoData$RUV_1 &lt;- RUV_1


# Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~0 + Cond + RUV_1, phenoData)
design</code></pre>
<pre><code>    CondControl CondDox       RUV_1
S1            0       1  0.14326547
S3            0       1 -0.08146518
S5            0       1 -0.25579041
S7            0       1  0.54847999
S9            0       1 -0.50097929
S2            1       0  0.15639944
S4            1       0 -0.04767015
S6            1       0 -0.35626503
S8            1       0 -0.19216750
S10           1       0  0.48449144
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;, &quot;RUV_1&quot;)


# Fit model
fit &lt;- lmFit(set$normalizedCounts, design)
fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2, robust = TRUE, trend = TRUE)</code></pre>
<pre><code>Warning: 3 very small variances detected, have been offset away from zero</code></pre>
<pre class="r"><code># Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        300
NotSig     2882
Up          202</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(set$normalizedCounts )), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)</code></pre>
</div>
<div id="wrangling-for-go" class="section level1">
<h1>Wrangling for GO</h1>
<pre class="r"><code># 1. Read in the comprehensive DF that contains all sample info
Full_DF %&gt;% head()</code></pre>
<pre><code>  Entrez.Gene.ID Ensembl.Gene.ID Gene.Symbol Accession
1           4625 ENSG00000092054        MYH7    P12883
2           4624 ENSG00000197616        MYH6    P13533
3           7273 ENSG00000155657         TTN    Q8WZ42
4             70 ENSG00000159251       ACTC1    P68032
5             58 ENSG00000143632       ACTA1    P68133
6             60 ENSG00000075624        ACTB    P60709
  Abundance..F1..Sample..Dox..n.a Abundance..F3..Sample..Dox..n.a
1                        5.25e+10                        7.41e+10
2                        1.00e+09                        1.47e+09
3                        1.39e+10                        2.08e+10
4                        3.92e+10                        4.11e+10
5                        1.72e+07                        5.81e+07
6                        4.02e+09                        8.43e+08
  Abundance..F5..Sample..Dox..Dox Abundance..F7..Sample..Dox..Dox
1                        5.28e+10                        6.82e+10
2                        5.11e+09                        4.25e+09
3                        2.19e+10                        2.49e+10
4                        3.76e+10                        3.54e+10
5                        3.85e+07                        5.07e+07
6                        2.04e+09                        1.62e+09
  Abundance..F9..Sample..Dox..Dox Abundance..F2..Control..Veh..n.a
1                        6.43e+10                         4.33e+10
2                        2.62e+09                         1.50e+09
3                        2.34e+10                         1.29e+10
4                        3.69e+10                         3.54e+10
5                        6.19e+07                         1.64e+07
6                        1.61e+09                         4.16e+09
  Abundance..F4..Control..Veh..n.a Abundance..F6..Control..Veh..Veh
1                         7.41e+10                         7.00e+10
2                         1.17e+09                         2.42e+09
3                         2.25e+10                         2.52e+10
4                         2.96e+10                         3.56e+10
5                         3.44e+07                         5.42e+07
6                         6.45e+08                         1.92e+09
  Abundance..F8..Control..Veh..Veh Abundance..F10..Control..Veh..Veh
1                         5.64e+10                          5.26e+10
2                         3.67e+09                          3.27e+09
3                         2.46e+10                          2.21e+10
4                         2.98e+10                          2.91e+10
5                         3.36e+07                          4.05e+07
6                         1.81e+09                          1.56e+09</code></pre>
<pre class="r"><code># 2.
# Get set of DE proteins as per BH adjusted P-value 
DE_proteins_Padj &lt;- toptable_summary %&gt;%
  filter(P.Value &lt; 0.05) %&gt;% rownames()

# Get SYMBOL ID for the proteins in DE_proteins_Padj
# Find the rows where &#39;var&#39; is in &#39;values&#39;
rows &lt;- which(Full_DF$Accession %in% DE_proteins_Padj)

# Subset the data frame
subset_Full_data &lt;- Full_DF[rows, ]

# Here are the SYMBOL IDs for the DE proteins as per BH adj P value &lt; 0.05
DE_SYMBOL_Padj &lt;- subset_Full_data$Gene.Symbol %&gt;% unique() %&gt;% na.omit()



# Get background Symbol IDs 
All_proteins_tested &lt;- toptable_summary %&gt;%
  filter(P.Value &lt; 1) %&gt;% rownames()

# Get SYMBOL ID for the proteins in DE_proteins_Padj
# Find the rows where &#39;var&#39; is in &#39;values&#39;
rows &lt;- which(Full_DF$Accession %in% All_proteins_tested)

# Subset the data frame
subset_back_data &lt;- Full_DF[rows, ]
Background_SYMBOL &lt;- subset_back_data$Gene.Symbol %&gt;% unique() %&gt;% na.omit() </code></pre>
</div>
<div id="go-enrichment-bp" class="section level1">
<h1>GO Enrichment BP</h1>
<pre class="r"><code># 1.
go_enrichment &lt;- enrichGO(gene = DE_SYMBOL_Padj,
                          OrgDb = org.Hs.eg.db,
                          keyType = &quot;SYMBOL&quot;,
                          universe = Background_SYMBOL,
                          ont = &quot;BP&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

# Term heirarchy plot 
goplot(go_enrichment, showCategory = 10)</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/BP-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># View the top enriched GO terms as a tibble 
GO_Tibble &lt;- go_enrichment %&gt;% as_tibble()
GO_Tibble</code></pre>
<pre><code># A tibble: 57 × 9
   ID      Description GeneRatio BgRatio   pvalue p.adjust   qvalue geneID Count
   &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;  &lt;int&gt;
 1 GO:000… cytoplasmi… 70/473    104/31… 2.49e-35 6.60e-32 6.44e-32 HNRNP…    70
 2 GO:000… translation 108/473   317/31… 5.58e-20 7.39e-17 7.22e-17 VIM/H…   108
 3 GO:004… peptide bi… 109/473   324/31… 1.11e-19 9.78e-17 9.55e-17 VIM/H…   109
 4 GO:004… ribosome b… 58/473    118/31… 1.84e-19 1.22e-16 1.19e-16 NPM1/…    58
 5 GO:000… RNA proces… 114/473   356/31… 1.13e-18 5.99e-16 5.85e-16 HNRNP…   114
 6 GO:000… rRNA proce… 46/473    85/3192 9.77e-18 4.31e-15 4.21e-15 DDX17…    46
 7 GO:001… rRNA metab… 50/473    99/3192 1.63e-17 6.18e-15 6.03e-15 NCL/D…    50
 8 GO:004… amide bios… 113/473   374/31… 2.56e-16 8.49e-14 8.29e-14 VIM/H…   113
 9 GO:000… peptide me… 114/473   383/31… 6.37e-16 1.87e-13 1.83e-13 VIM/H…   114
10 GO:003… cellular m… 118/473   407/31… 1.55e-15 4.11e-13 4.02e-13 VIM/H…   118
# ℹ 47 more rows</code></pre>
<pre class="r"><code># Arrange by adjusted p value 
GO_Tibble &lt;- GO_Tibble %&gt;%
  arrange(desc(p.adjust))

# Factor by P.adjusted value 
GO_Tibble$Description &lt;- factor(GO_Tibble$Description, levels = GO_Tibble$Description[order(GO_Tibble$p.adjust)])

# Get P value for consistent gradient fill
Gradient_value &lt;- GO_Tibble$qvalue %&gt;% median()

# Visualize the total enriched GO terms using a vertical bar plot 
ggplot(data = GO_Tibble[(nrow(GO_Tibble) - 55):nrow(GO_Tibble) ,], aes(x = -log(p.adjust), y = reorder(Description, -log(p.adjust)))) +
  geom_bar(stat = &quot;identity&quot;) + 
  geom_bar(stat = &quot;identity&quot;, fill = &quot;#4B9CD3&quot;)+
  labs(x = &quot;-log(p-adj)&quot;,
       y = &quot;Biologcal Processes&quot;,
       fill = &quot;P.adjusted&quot;,
       title = &quot;GO enrichment analysis&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  xlim(c(0,100))</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/BP-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>edo &lt;- enrichplot::pairwise_termsim(go_enrichment)
emapplot(edo, showCategory = 20, cex_category= 2 )</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/BP-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="go-enrichment-mf" class="section level1">
<h1>GO Enrichment MF</h1>
<pre class="r"><code># 1.
go_enrichment &lt;- enrichGO(gene = DE_SYMBOL_Padj,
                          OrgDb = org.Hs.eg.db,
                          keyType = &quot;SYMBOL&quot;,
                          universe = Background_SYMBOL,
                          ont = &quot;MF&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

# Go plot 
goplot(go_enrichment, showCategory = 20)</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/MF-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># View the top enriched GO terms as a tibble 
GO_Tibble &lt;- go_enrichment %&gt;% as_tibble()
GO_Tibble</code></pre>
<pre><code># A tibble: 13 × 9
   ID      Description GeneRatio BgRatio   pvalue p.adjust   qvalue geneID Count
   &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;  &lt;int&gt;
 1 GO:000… structural… 77/480    129/32… 2.07e-33 8.93e-31 8.62e-31 RPS27…    77
 2 GO:000… structural… 99/480    302/32… 7.02e-17 1.51e-14 1.46e-14 VIM/T…    99
 3 GO:000… mRNA bindi… 53/480    155/32… 4.79e-10 6.88e- 8 6.64e- 8 HNRNP…    53
 4 GO:000… DNA binding 75/480    298/32… 5.67e- 7 6.11e- 5 5.90e- 5 HNRNP…    75
 5 GO:000… chromatin … 39/480    126/32… 2.08e- 6 1.79e- 4 1.73e- 4 HNRNP…    39
 6 GO:000… poly(A) bi… 9/480     13/3239 1.32e- 5 9.47e- 4 9.13e- 4 HNRNP…     9
 7 GO:004… mRNA 5&#39;-UT… 11/480    20/3239 3.20e- 5 1.97e- 3 1.90e- 3 NCL/S…    11
 8 GO:001… rRNA bindi… 17/480    42/3239 4.22e- 5 2.28e- 3 2.19e- 3 RPS3/…    17
 9 GO:007… poly-purin… 9/480     16/3239 1.39e- 4 6.64e- 3 6.41e- 3 HNRNP…     9
10 GO:000… mRNA 3&#39;-UT… 17/480    47/3239 2.21e- 4 9.51e- 3 9.18e- 3 HNRNP…    17
11 GO:000… RNA helica… 13/480    33/3239 4.70e- 4 1.69e- 2 1.63e- 2 DDX17…    13
12 GO:000… ATP-depend… 13/480    33/3239 4.70e- 4 1.69e- 2 1.63e- 2 DDX17…    13
13 GO:003… nucleosome… 12/480    32/3239 1.30e- 3 4.31e- 2 4.16e- 2 H1-4/…    12</code></pre>
<pre class="r"><code># Arrange by adjusted p value 
GO_Tibble &lt;- GO_Tibble %&gt;%
  arrange(desc(p.adjust))

# Factor by P.adjusted value 
GO_Tibble$Description &lt;- factor(GO_Tibble$Description, levels = GO_Tibble$Description[order(GO_Tibble$p.adjust)])

# Get P value for consistent gradient fill
Gradient_value &lt;- GO_Tibble$qvalue %&gt;% median()

# Visualize the total enriched GO terms using a vertical bar plot 
ggplot(data = GO_Tibble, aes(x = -log(p.adjust), y = reorder(Description, -log(p.adjust)))) +
  geom_bar(stat = &quot;identity&quot;) + 
  geom_bar(stat = &quot;identity&quot;, fill = &quot;#4B9CD3&quot;)+
  labs(x = &quot;-log(p-adj)&quot;,
       y = &quot;Biologcal Processes&quot;,
       fill = &quot;P.adjusted&quot;,
       title = &quot;GO enrichment analysis&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  xlim(c(0,100))</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/MF-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>edo &lt;- enrichplot::pairwise_termsim(go_enrichment)
emapplot(edo, showCategory = 15, cex_category=1.5)</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/MF-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="go-enrichment-cc" class="section level1">
<h1>GO Enrichment CC</h1>
<pre class="r"><code># 1.
go_enrichment &lt;- enrichGO(gene = DE_SYMBOL_Padj,
                          OrgDb = org.Hs.eg.db,
                          keyType = &quot;SYMBOL&quot;,
                          universe = Background_SYMBOL,
                          ont = &quot;CC&quot;,
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.05)

goplot(go_enrichment, showCategory = 20)</code></pre>
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/CC-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># View the top enriched GO terms as a tibble 
GO_Tibble &lt;- go_enrichment %&gt;% as_tibble()
GO_Tibble</code></pre>
<pre><code># A tibble: 25 × 9
   ID      Description GeneRatio BgRatio   pvalue p.adjust   qvalue geneID Count
   &lt;chr&gt;   &lt;chr&gt;       &lt;chr&gt;     &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;  &lt;int&gt;
 1 GO:002… cytosolic … 64/487    78/3283 3.26e-41 1.41e-38 1.32e-38 RPS27…    64
 2 GO:004… ribosomal … 78/487    141/32… 1.36e-30 2.94e-28 2.74e-28 RPS27…    78
 3 GO:000… ribosome    80/487    155/32… 1.41e-28 2.03e-26 1.90e-26 RPS27…    80
 4 GO:199… ribonucleo… 133/487   367/32… 6.35e-28 6.85e-26 6.40e-26 VIM/H…   133
 5 GO:002… cytosolic … 35/487    42/3283 3.30e-23 2.85e-21 2.66e-21 RPL4/…    35
 6 GO:002… cytosolic … 29/487    36/3283 1.34e-18 9.66e-17 9.02e-17 RPS27…    29
 7 GO:001… large ribo… 44/487    84/3283 2.95e-16 1.82e-14 1.70e-14 RPL4/…    44
 8 GO:001… small ribo… 34/487    58/3283 9.41e-15 5.08e-13 4.74e-13 RPS27…    34
 9 GO:004… polysomal … 17/487    23/3283 2.69e-10 1.29e- 8 1.21e- 8 RPL7A…    17
10 GO:000… polysome    23/487    41/3283 7.99e-10 3.45e- 8 3.22e- 8 VIM/R…    23
# ℹ 15 more rows</code></pre>
<pre class="r"><code># Arrange by adjusted p value 
GO_Tibble &lt;- GO_Tibble %&gt;%
  arrange(desc(p.adjust))

# Factor by P.adjusted value 
GO_Tibble$Description &lt;- factor(GO_Tibble$Description, levels = GO_Tibble$Description[order(GO_Tibble$p.adjust)])

# Get P value for consistent gradient fill
Gradient_value &lt;- GO_Tibble$qvalue %&gt;% median()

# Visualize the total enriched GO terms using a vertical bar plot 
ggplot(data = GO_Tibble[(nrow(GO_Tibble) - 20):nrow(GO_Tibble) ,], aes(x = -log(p.adjust), y = reorder(Description, -log(p.adjust)))) +
  geom_bar(stat = &quot;identity&quot;) + 
  geom_bar(stat = &quot;identity&quot;, fill = &quot;#4B9CD3&quot;)+
  labs(x = &quot;-log(p-adj)&quot;,
       y = &quot;Biologcal Processes&quot;,
       fill = &quot;P.adjusted&quot;,
       title = &quot;GO enrichment analysis&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  xlim(c(0,100))</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/CC-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>edo &lt;- enrichplot::pairwise_termsim(go_enrichment)
emapplot(edo, showCategory = 15, cex_category=1.5)</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/CC-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="reactome" class="section level1">
<h1>Reactome</h1>
<pre class="r"><code># Wrangle ENTREZ IDs for Pval 
# 1. Read in the comprehensive DF that contains all sample info
Full_DF %&gt;% head()</code></pre>
<pre><code>  Entrez.Gene.ID Ensembl.Gene.ID Gene.Symbol Accession
1           4625 ENSG00000092054        MYH7    P12883
2           4624 ENSG00000197616        MYH6    P13533
3           7273 ENSG00000155657         TTN    Q8WZ42
4             70 ENSG00000159251       ACTC1    P68032
5             58 ENSG00000143632       ACTA1    P68133
6             60 ENSG00000075624        ACTB    P60709
  Abundance..F1..Sample..Dox..n.a Abundance..F3..Sample..Dox..n.a
1                        5.25e+10                        7.41e+10
2                        1.00e+09                        1.47e+09
3                        1.39e+10                        2.08e+10
4                        3.92e+10                        4.11e+10
5                        1.72e+07                        5.81e+07
6                        4.02e+09                        8.43e+08
  Abundance..F5..Sample..Dox..Dox Abundance..F7..Sample..Dox..Dox
1                        5.28e+10                        6.82e+10
2                        5.11e+09                        4.25e+09
3                        2.19e+10                        2.49e+10
4                        3.76e+10                        3.54e+10
5                        3.85e+07                        5.07e+07
6                        2.04e+09                        1.62e+09
  Abundance..F9..Sample..Dox..Dox Abundance..F2..Control..Veh..n.a
1                        6.43e+10                         4.33e+10
2                        2.62e+09                         1.50e+09
3                        2.34e+10                         1.29e+10
4                        3.69e+10                         3.54e+10
5                        6.19e+07                         1.64e+07
6                        1.61e+09                         4.16e+09
  Abundance..F4..Control..Veh..n.a Abundance..F6..Control..Veh..Veh
1                         7.41e+10                         7.00e+10
2                         1.17e+09                         2.42e+09
3                         2.25e+10                         2.52e+10
4                         2.96e+10                         3.56e+10
5                         3.44e+07                         5.42e+07
6                         6.45e+08                         1.92e+09
  Abundance..F8..Control..Veh..Veh Abundance..F10..Control..Veh..Veh
1                         5.64e+10                          5.26e+10
2                         3.67e+09                          3.27e+09
3                         2.46e+10                          2.21e+10
4                         2.98e+10                          2.91e+10
5                         3.36e+07                          4.05e+07
6                         1.81e+09                          1.56e+09</code></pre>
<pre class="r"><code># 2.
# Get set of DE proteins as per BH adjusted P-value 
DE_ENTREZ_Padj &lt;- toptable_summary %&gt;%
  filter(P.Value &lt; 0.05) %&gt;% rownames()

# Get SYMBOL ID for the proteins in DE_proteins_Padj
# Find the rows where &#39;var&#39; is in &#39;values&#39;
rows &lt;- which(Full_DF$Accession %in% DE_ENTREZ_Padj)


# Subset the data frame
subset_Full_data &lt;- Full_DF[rows, ]

# Here are the SYMBOL IDs for the DE proteins as per BH adj P value &lt; 0.05
DE_ENTREZ_Padj &lt;- subset_Full_data$Entrez.Gene.ID %&gt;% unique() %&gt;% na.omit()


# Get background Symbol IDs 
All_proteins_tested &lt;- toptable_summary %&gt;%
  filter(P.Value &lt; 1) %&gt;% rownames()

# Get SYMBOL ID for the proteins in DE_proteins_Padj
# Find the rows where &#39;var&#39; is in &#39;values&#39;
rows &lt;- which(Full_DF$Accession %in% All_proteins_tested)

# Subset the data frame
subset_back_data &lt;- Full_DF[rows, ]
Background_ENTREZ &lt;- subset_back_data$Entrez.Gene.ID %&gt;% unique() %&gt;% na.omit() 


# Fusion proteins in ENTREZ ids. Therefore, need to split them up 
# vector to split 
my_vector_test &lt;- DE_ENTREZ_Padj


# Split the elements of the vector on semicolons
split_vector &lt;- strsplit(my_vector_test, split = &quot;;&quot;)

# Unlist the split vector
new_vector_ENTREZ &lt;- unlist(split_vector)


# Do the same for background 
back_vector_test &lt;- Background_ENTREZ

# Split the elements of the vector on semicolons
split_vector &lt;- strsplit(back_vector_test, split = &quot;;&quot;)

# Unlist the split vector
back_new_vector_ENTREZ &lt;- unlist(split_vector)




# Test 
EnrichPathobj &lt;- ReactomePA::enrichPathway(
  new_vector_ENTREZ,
  organism = &quot;human&quot;,
  pvalueCutoff = 0.05,
  pAdjustMethod = &quot;BH&quot;,
  qvalueCutoff = 0.05,
  universe = back_new_vector_ENTREZ,
  minGSSize = 2,
  maxGSSize = 500,
  readable = FALSE
)



# summary of the result that most people use 
plot(barplot(EnrichPathobj, showCategory = 10 ))</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/Reactome-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># View the top enriched GO terms as a tibble 
GO_Tibble &lt;- EnrichPathobj %&gt;% as_tibble()


# Arrange by adjusted p value 
GO_Tibble &lt;- GO_Tibble %&gt;%
  arrange(desc(p.adjust))

# Factor by P.adjusted value 
GO_Tibble$Description &lt;- factor(GO_Tibble$Description, levels = GO_Tibble$Description[order(GO_Tibble$p.adjust)])

# Get P value for consistent gradient fill
Gradient_value &lt;- GO_Tibble$qvalue %&gt;% median()

# Visualize the total enriched GO terms using a vertical bar plot 
ggplot(data = GO_Tibble[(nrow(GO_Tibble) - 30):nrow(GO_Tibble) ,], aes(x = -log(p.adjust), y = reorder(Description, -log(p.adjust)))) +
  geom_bar(stat = &quot;identity&quot;) + 
  geom_bar(stat = &quot;identity&quot;, fill = &quot;#4B9CD3&quot;)+
  labs(x = &quot;-log(p-adj)&quot;,
       y = &quot;Molecular Functions&quot;,
       fill = &quot;P.adjusted&quot;,
       title = &quot;GO enrichment analysis&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  xlim(c(0,130))</code></pre>
<p><img src="figure/GO_Reactome_enrich_Imputed_proteins.Rmd/Reactome-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] enrichplot_1.16.2                      
 [2] impute_1.70.0                          
 [3] WGCNA_1.72-1                           
 [4] fastcluster_1.2.3                      
 [5] dynamicTreeCut_1.63-1                  
 [6] BioNERO_1.4.2                          
 [7] reshape2_1.4.4                         
 [8] ggridges_0.5.4                         
 [9] biomaRt_2.52.0                         
[10] ggvenn_0.1.10                          
[11] UpSetR_1.4.0                           
[12] DOSE_3.22.1                            
[13] variancePartition_1.26.0               
[14] clusterProfiler_4.4.4                  
[15] pheatmap_1.0.12                        
[16] qvalue_2.28.0                          
[17] Homo.sapiens_1.3.1                     
[18] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[19] org.Hs.eg.db_3.15.0                    
[20] GO.db_3.15.0                           
[21] OrganismDbi_1.38.1                     
[22] GenomicFeatures_1.48.4                 
[23] AnnotationDbi_1.58.0                   
[24] cluster_2.1.4                          
[25] ggfortify_0.4.16                       
[26] lubridate_1.9.2                        
[27] forcats_1.0.0                          
[28] stringr_1.5.0                          
[29] dplyr_1.1.2                            
[30] purrr_1.0.2                            
[31] readr_2.1.4                            
[32] tidyr_1.3.0                            
[33] tibble_3.2.1                           
[34] ggplot2_3.4.3                          
[35] tidyverse_2.0.0                        
[36] RColorBrewer_1.1-3                     
[37] RUVSeq_1.30.0                          
[38] edgeR_3.38.4                           
[39] limma_3.52.4                           
[40] EDASeq_2.30.0                          
[41] ShortRead_1.54.0                       
[42] GenomicAlignments_1.32.1               
[43] SummarizedExperiment_1.26.1            
[44] MatrixGenerics_1.8.1                   
[45] matrixStats_1.0.0                      
[46] Rsamtools_2.12.0                       
[47] GenomicRanges_1.48.0                   
[48] Biostrings_2.64.1                      
[49] GenomeInfoDb_1.32.4                    
[50] XVector_0.36.0                         
[51] IRanges_2.30.1                         
[52] S4Vectors_0.34.0                       
[53] BiocParallel_1.30.4                    
[54] Biobase_2.56.0                         
[55] BiocGenerics_0.42.0                    
[56] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] xml2_1.3.5             httpuv_1.6.11          viridis_0.6.4         
 [28] xfun_0.40              hms_1.1.3              jquerylib_0.1.4       
 [31] evaluate_0.21          promises_1.2.1         fansi_1.0.4           
 [34] restfulr_0.0.15        progress_1.2.2         caTools_1.18.2        
 [37] dbplyr_2.3.3           htmlwidgets_1.6.2      igraph_1.5.1          
 [40] DBI_1.1.3              ggnewscale_0.4.9       backports_1.4.1       
 [43] annotate_1.74.0        aod_1.3.2              deldir_1.0-9          
 [46] vctrs_0.6.3            abind_1.4-5            cachem_1.0.8          
 [49] withr_2.5.0            ggforce_0.4.1          checkmate_2.2.0       
 [52] treeio_1.20.2          prettyunits_1.1.1      ape_5.7-1             
 [55] lazyeval_0.2.2         crayon_1.5.2           genefilter_1.78.0     
 [58] labeling_0.4.2         pkgconfig_2.0.3        tweenr_2.0.2          
 [61] nlme_3.1-163           nnet_7.3-19            rlang_1.1.1           
 [64] lifecycle_1.0.3        downloader_0.4         filelock_1.0.2        
 [67] BiocFileCache_2.4.0    rprojroot_2.0.3        polyclip_1.10-4       
 [70] graph_1.74.0           Matrix_1.5-4.1         aplot_0.2.0           
 [73] NetRep_1.2.7           boot_1.3-28.1          base64enc_0.1-3       
 [76] GlobalOptions_0.1.2    whisker_0.4.1          processx_3.8.2        
 [79] png_0.1-8              viridisLite_0.4.2      rjson_0.2.21          
 [82] bitops_1.0-7           getPass_0.2-2          R.oo_1.25.0           
 [85] ggnetwork_0.5.12       KernSmooth_2.23-22     blob_1.2.4            
 [88] shape_1.4.6            jpeg_0.1-10            gridGraphics_0.5-1    
 [91] reactome.db_1.81.0     scales_1.2.1           graphite_1.42.0       
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.8            
 [97] gplots_3.1.3           zlibbioc_1.42.0        compiler_4.2.0        
[100] scatterpie_0.2.1       BiocIO_1.6.0           clue_0.3-64           
[103] intergraph_2.0-3       lme4_1.1-34            cli_3.6.1             
[106] patchwork_1.1.3        ps_1.7.5               htmlTable_2.4.1       
[109] Formula_1.2-5          mgcv_1.9-0             MASS_7.3-60           
[112] tidyselect_1.2.0       stringi_1.7.12         highr_0.10            
[115] yaml_2.3.7             GOSemSim_2.22.0        locfit_1.5-9.8        
[118] latticeExtra_0.6-30    ggrepel_0.9.3          sass_0.4.7            
[121] fastmatch_1.1-4        tools_4.2.0            timechange_0.2.0      
[124] parallel_4.2.0         circlize_0.4.15        rstudioapi_0.15.0     
[127] foreign_0.8-84         foreach_1.5.2          git2r_0.32.0          
[130] gridExtra_2.3          farver_2.1.1           ggraph_2.1.0          
[133] digest_0.6.33          BiocManager_1.30.22    networkD3_0.4         
[136] Rcpp_1.0.11            broom_1.0.5            later_1.3.1           
[139] httr_1.4.7             ComplexHeatmap_2.12.1  GENIE3_1.18.0         
[142] Rdpack_2.5             colorspace_2.1-0       XML_3.99-0.14         
[145] fs_1.6.3               splines_4.2.0          statmod_1.5.0         
[148] yulab.utils_0.0.8      RBGL_1.72.0            tidytree_0.4.5        
[151] graphlayouts_1.0.0     ggplotify_0.1.2        xtable_1.8-4          
[154] jsonlite_1.8.7         nloptr_2.0.3           ggtree_3.4.4          
[157] tidygraph_1.2.3        ggfun_0.1.2            R6_2.5.1              
[160] Hmisc_5.1-0            pillar_1.9.0           htmltools_0.5.6       
[163] glue_1.6.2             fastmap_1.1.1          minqa_1.2.5           
[166] codetools_0.2-19       fgsea_1.22.0           utf8_1.2.3            
[169] sva_3.44.0             lattice_0.21-8         bslib_0.5.1           
[172] network_1.18.1         pbkrtest_0.5.2         curl_5.0.2            
[175] ReactomePA_1.40.0      gtools_3.9.4           interp_1.1-4          
[178] survival_3.5-7         statnet.common_4.9.0   rmarkdown_2.24        
[181] munsell_0.5.0          GetoptLong_1.0.5       DO.db_2.9             
[184] GenomeInfoDbData_1.2.8 iterators_1.0.14       gtable_0.3.4          
[187] rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
