<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Omar Johnson" />

<meta name="date" content="2023-12-27" />

<title>Sign_and_genetic_summary_stat_testing</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Omar-Johnson/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Sign_and_genetic_summary_stat_testing</h1>
<h4 class="author">Omar Johnson</h4>
<h4 class="date">2023-12-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230911code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230911)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230911code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230911)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreed7917090991e40eceab98de8887fb7675a795cc2targetblankd791709a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Omar-Johnson/myproject/tree/d7917090991e40eceab98de8887fb7675a795cc2" target="_blank">d791709</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreed7917090991e40eceab98de8887fb7675a795cc2targetblankd791709a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Omar-Johnson/myproject/tree/d7917090991e40eceab98de8887fb7675a795cc2" target="_blank">d791709</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  #14_venn_diagramm.png
    Untracked:  #14_venn_diagramm.png.2024-01-11_15-41-45.log
    Untracked:  #14_venn_diagramm.png.2024-01-11_15-41-51.log
    Untracked:  VennDiagram.2024-01-11_15-43-33.log
    Untracked:  VennDiagram.2024-01-11_15-43-45.log
    Untracked:  VennDiagram.2024-01-11_15-48-56.log
    Untracked:  VennDiagram.2024-01-11_15-49-14.log
    Untracked:  VennDiagram.2024-01-11_15-49-51.log
    Untracked:  VennDiagram.2024-01-11_15-52-24.log
    Untracked:  VennDiagram.2024-01-11_15-53-14.log
    Untracked:  VennDiagram.2024-01-11_15-53-47.log
    Untracked:  VennDiagram.2024-01-11_15-54-07.log
    Untracked:  VennDiagram.2024-01-11_18-08-58.log
    Untracked:  VennDiagram.2024-01-11_18-14-06.log
    Untracked:  VennDiagram.2024-01-11_18-17-11.log
    Untracked:  VennDiagram.2024-01-11_18-18-29.log
    Untracked:  VennDiagram.2024-01-11_18-19-23.log
    Untracked:  VennDiagram.2024-01-11_18-19-52.log
    Untracked:  VennDiagram.2024-01-11_18-22-22.log
    Untracked:  VennDiagram.2024-01-11_18-22-23.log
    Untracked:  VennDiagram.2024-01-12_17-13-03.log
    Untracked:  VennDiagram.2024-01-12_17-13-04.log
    Untracked:  analysis/Data_ind_imp_RUVg_WPCNA.Rmd
    Untracked:  analysis/Random_bed.R

Unstaged changes:
    Modified:   Data_ind_Diffabtrial.Rmd
    Modified:   analysis/Diffabtrial.Rmd
    Modified:   analysis/Differentially_abundant_proteins_impute.Rmd
    Modified:   analysis/Differentially_abunddant_proteins.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/Sign_and_genetic_summary_stat_testing.Rmd</code>) and
HTML (<code>docs/Sign_and_genetic_summary_stat_testing.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/d7917090991e40eceab98de8887fb7675a795cc2/analysis/Sign_and_genetic_summary_stat_testing.Rmd" target="_blank">d791709</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-12
</td>
<td>
Updated after removal of non-human peptides
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Omar-Johnson/myproject/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/Sign_and_genetic_summary_stat_testing.html" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/66a78d57e171358c0f7288144bb79fffdb8337c5/analysis/Sign_and_genetic_summary_stat_testing.Rmd" target="_blank">66a78d5</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Get data 
RNA_Seq_Top_Table &lt;- read_csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/RNA_Seq/Doxtoplist.csv&quot;, col_names = TRUE, row.names(1))</code></pre>
<pre><code>New names:
Rows: 28168 Columns: 9
── Column specification
──────────────────────────────────────────────────────── Delimiter: &quot;,&quot; chr
(4): ...1, time, id, SYMBOL dbl (5): ENTREZID, logFC, AveExpr, P.Value,
adj.P.Val
ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
Specify the column types or set `show_col_types = FALSE` to quiet this message.
• `` -&gt; `...1`</code></pre>
<pre class="r"><code>RNA_Seq_Top_Table %&gt;% dim()</code></pre>
<pre><code>[1] 28168     9</code></pre>
<pre class="r"><code># Group the 24hr data together 
RNA_Seq_Top_Table_24hr &lt;- RNA_Seq_Top_Table %&gt;% dplyr::filter(time == &quot;24_hours&quot;)



toptable_summary_DoxvNorm &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/Toptable_summary_RUVIII.csv&quot;, header = TRUE, row.names = 1)
toptable_summary_DoxvNorm %&gt;% head()</code></pre>
<pre><code>           logFC  AveExpr         t      P.Value    adj.P.Val        B Protein
Q15061  2.436903 18.60721  26.00939 1.499311e-07 1.499311e-07 7.414993  Q15061
O76021 -2.224860 21.98800 -18.86036 1.191879e-06 1.191879e-06 6.029757  O76021
P43694 -1.821642 18.81578 -15.98230 3.129665e-06 3.129665e-06 5.276514  P43694
P01130 -1.819689 20.01194 -15.77595 3.483331e-06 3.483331e-06 5.186033  P01130
Q96P11 -1.843458 17.28018 -14.33937 5.735316e-06 5.735316e-06 4.764600  Q96P11
Q6EMK4  1.838032 19.30419  14.22737 6.455771e-06 6.455771e-06 4.660093  Q6EMK4
       threshold_P
Q15061        TRUE
O76021        TRUE
P43694        TRUE
P01130        TRUE
Q96P11        TRUE
Q6EMK4        TRUE</code></pre>
<pre class="r"><code>toptable_summary_DoxvNorm %&gt;% dim()</code></pre>
<pre><code>[1] 4178    8</code></pre>
<pre class="r"><code>#### 24 hr Plots ####
# Basic workflow for visualizing. and interpreting genes for 24 hr timepoint. 
# Multiply by -1 according to Renee. 
RNA_Seq_Top_Table_24hr$logFC &lt;- RNA_Seq_Top_Table_24hr$logFC * -1 

# 1. Create a column to threshold adjusted P-values/q-values 
RNA_24hr_Dox_toptable_Threshold_0.05 &lt;- RNA_Seq_Top_Table_24hr %&gt;%  mutate(threshold_P = adj.P.Val &lt; 0.05)


# 2. Do the volcano plot 
ggplot(RNA_24hr_Dox_toptable_Threshold_0.05)+
  geom_point(mapping = aes(x = logFC, y = -log10(adj.P.Val), color = threshold_P ))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 adjusted p-value&quot;)+
  ylim(0, 25)+
  xlim(-8, 8)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/RNA-Seq%20plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-RNA-Seq-plots-1">
Past versions of "RNA-Seq plots-1.png"
</button>
</p>
<div id="fig-no-spaces-RNA-Seq-plots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/RNA-Seq plots-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#### Volcano plot with labels  #### 
ggplot(RNA_24hr_Dox_toptable_Threshold_0.05)+
  geom_point(mapping = aes(x = logFC, y = -log10(adj.P.Val), color = threshold_P))+
  geom_text(mapping = aes(x = logFC, y = -log10(adj.P.Val), label = SYMBOL), 
            hjust = -0.1, vjust = -0.1, check_overlap = TRUE, size = 3)+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 adjusted p-value&quot;)+
  ylim(0, 30)+
  xlim(-10, 10)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_bw()</code></pre>
<pre><code>Warning: Removed 70 rows containing missing values (`geom_text()`).</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/RNA-Seq%20plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-RNA-Seq-plots-2">
Past versions of "RNA-Seq plots-2.png"
</button>
</p>
<div id="fig-no-spaces-RNA-Seq-plots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/RNA-Seq plots-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Use the ensembl dataset
mart &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)

# vector of Uniprot IDs
uniprot_ids &lt;- toptable_summary_DoxvNorm %&gt;% rownames()

uniprot_ids %&gt;% length()</code></pre>
<pre><code>[1] 4178</code></pre>
<pre class="r"><code># Get the corresponding ENTREZ IDs----&quot;ensembl_gene_id&quot;
results_Pro &lt;- getBM(filters = &quot;uniprotswissprot&quot;, 
                     attributes = c(&quot;uniprotswissprot&quot;, &quot;hgnc_symbol&quot;), 
                     values = uniprot_ids, 
                     mart = mart)


# Get the corresponding SYMBOLs
results_RNA &lt;- getBM(filters = &quot;hgnc_symbol&quot;, 
                     attributes = c(&quot;uniprotswissprot&quot;, &quot;hgnc_symbol&quot;), 
                     values = RNA_24hr_Dox_toptable_Threshold_0.05$SYMBOL, 
                     mart = mart)

results_Pro %&gt;% colnames()</code></pre>
<pre><code>[1] &quot;uniprotswissprot&quot; &quot;hgnc_symbol&quot;     </code></pre>
<pre class="r"><code>results_RNA %&gt;% colnames()</code></pre>
<pre><code>[1] &quot;uniprotswissprot&quot; &quot;hgnc_symbol&quot;     </code></pre>
<pre class="r"><code>RNA_Key_Merged &lt;- merge(RNA_24hr_Dox_toptable_Threshold_0.05, results_RNA, by.x = &quot;SYMBOL&quot; , by.y = &quot;hgnc_symbol&quot;)

Pro_Key_Merged &lt;- merge(toptable_summary_DoxvNorm, results_Pro, by.x = &quot;Protein&quot; , by.y = &quot;uniprotswissprot&quot;)

Pro_RNA_Key_Merge &lt;-  merge(RNA_Key_Merged, Pro_Key_Merged, by.x = &quot;uniprotswissprot&quot; , by.y = &quot;Protein&quot;)
 

RNA_for_merge &lt;- RNA_24hr_Dox_toptable_Threshold_0.05[, colnames(RNA_24hr_Dox_toptable_Threshold_0.05) %in% c(&quot;logFC&quot;,&quot;adj.P.Val&quot;,&quot;SYMBOL&quot; )]
colnames(RNA_for_merge) &lt;- c(&quot;SYMBOL&quot;, &quot;RNA_LogFC&quot;, &quot;RNA_adj_P&quot;)
RNA_for_merge_2 &lt;- merge(results_RNA, RNA_for_merge, by.x = &quot;hgnc_symbol&quot; , by.y = &quot;SYMBOL&quot;)
RNA_for_merge_2 %&gt;% head()</code></pre>
<pre><code>  hgnc_symbol uniprotswissprot  RNA_LogFC    RNA_adj_P
1        A1BG                  -0.5180219 0.0893552102
2        A1BG           P04217 -0.5180219 0.0893552102
3         A2M           P01023  0.7826268 0.1230949055
4         A2M                   0.7826268 0.1230949055
5      A4GALT           Q9NPC4 -0.8227026 0.0060779524
6        AAAS           Q9NRG9 -0.6726711 0.0000207135</code></pre>
<pre class="r"><code>Pro_Key_Merged_2 &lt;- merge(Pro_Key_Merged, RNA_for_merge_2, by.x = &quot;Protein&quot; , by.y = &quot;uniprotswissprot&quot;)
Pro_Key_Merged_2 %&gt;% head()</code></pre>
<pre><code>     Protein       logFC  AveExpr          t   P.Value adj.P.Val         B
1 A0A0B4J2A2  0.02740191 18.36074  0.1652536 0.8740733 0.8740733 -6.909831
2     A0AVT1  0.28958132 13.44419  1.1568528 0.2903888 0.2903888 -6.228915
3     A0FGR8  0.09891217 20.14964  1.1875531 0.2786322 0.2786322 -6.195353
4     A0JLT2  0.14029123 17.75159  0.6135087 0.5616554 0.5616554 -6.715102
5     A0PJW6 -0.04745381 19.76917 -0.3727514 0.7219144 0.7219144 -6.846368
6     A1A4S6  0.14066972 16.85673  1.3533815 0.2233311 0.2233311 -6.003703
  threshold_P hgnc_symbol.x hgnc_symbol.y  RNA_LogFC    RNA_adj_P
1       FALSE       PPIAL4C       PPIAL4C  0.8434727 1.713779e-03
2       FALSE          UBA6          UBA6  0.0741531 7.259133e-01
3       FALSE         ESYT2         ESYT2  1.3269198 1.313129e-09
4       FALSE         MED19         MED19  0.3759240 4.391037e-03
5       FALSE       TMEM223       TMEM223 -1.4889825 5.703893e-04
6       FALSE      ARHGAP10      ARHGAP10  0.8139165 9.576317e-05</code></pre>
<pre class="r"><code># Joining protein and RNA data frames together 
Results_RNA_DF_Pro_DF_Merged_NAomit &lt;- data.frame(uniprotswissprot = Pro_Key_Merged_2$Protein, 
                                                  hgnc_symbol =Pro_Key_Merged_2$hgnc_symbol.x , 
                                                  RNA_LogFC = Pro_Key_Merged_2$RNA_LogFC, 
                                                  RNA_Pval = Pro_Key_Merged_2$RNA_adj_P, 
                                                  Pro_LogFC = Pro_Key_Merged_2$logFC, 
                                                  Pro_Pval =  Pro_Key_Merged_2$P.Value)

Results_RNA_DF_Pro_DF_Merged_NAomit %&gt;% head()</code></pre>
<pre><code>  uniprotswissprot hgnc_symbol  RNA_LogFC     RNA_Pval   Pro_LogFC  Pro_Pval
1       A0A0B4J2A2     PPIAL4C  0.8434727 1.713779e-03  0.02740191 0.8740733
2           A0AVT1        UBA6  0.0741531 7.259133e-01  0.28958132 0.2903888
3           A0FGR8       ESYT2  1.3269198 1.313129e-09  0.09891217 0.2786322
4           A0JLT2       MED19  0.3759240 4.391037e-03  0.14029123 0.5616554
5           A0PJW6     TMEM223 -1.4889825 5.703893e-04 -0.04745381 0.7219144
6           A1A4S6    ARHGAP10  0.8139165 9.576317e-05  0.14066972 0.2233311</code></pre>
<pre class="r"><code>Results_RNA_DF_Pro_DF_Merged_NAomit %&gt;% dim()</code></pre>
<pre><code>[1] 4155    6</code></pre>
<pre class="r"><code>#### Sign Testing #### 
# 24hr--- DE protein &amp; Not DE RNA (Up and down regulated), ~ DE protein &amp; DE RNA (up and down regulated), DE protein &amp; DE RNA with same and opposite sign (up and down regulated)


# Add a new column to the data frame indicating whether the signs of Pro_LogFC and RNA_LogFC match
Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match &lt;- sign(Results_RNA_DF_Pro_DF_Merged_NAomit$RNA_LogFC) == sign(Results_RNA_DF_Pro_DF_Merged_NAomit$Pro_LogFC)


# Calculate the proportion of matches for all members 
proportion_sign_matches &lt;- mean(Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match)
print(proportion_sign_matches)</code></pre>
<pre><code>[1] 0.5412756</code></pre>
<pre class="r"><code># Calculate the proportion of matches for DE proteins 
proportion_sign_matches &lt;- mean(Results_RNA_DF_Pro_DF_Merged_NAomit[Results_RNA_DF_Pro_DF_Merged_NAomit$Pro_Pval &lt;0.05, ]$sign_match)
print(proportion_sign_matches)</code></pre>
<pre><code>[1] 0.5876652</code></pre>
<pre class="r"><code>Results_RNA_DF_Pro_DF_Merged_NAomit %&gt;% head()</code></pre>
<pre><code>  uniprotswissprot hgnc_symbol  RNA_LogFC     RNA_Pval   Pro_LogFC  Pro_Pval
1       A0A0B4J2A2     PPIAL4C  0.8434727 1.713779e-03  0.02740191 0.8740733
2           A0AVT1        UBA6  0.0741531 7.259133e-01  0.28958132 0.2903888
3           A0FGR8       ESYT2  1.3269198 1.313129e-09  0.09891217 0.2786322
4           A0JLT2       MED19  0.3759240 4.391037e-03  0.14029123 0.5616554
5           A0PJW6     TMEM223 -1.4889825 5.703893e-04 -0.04745381 0.7219144
6           A1A4S6    ARHGAP10  0.8139165 9.576317e-05  0.14066972 0.2233311
  sign_match
1       TRUE
2       TRUE
3       TRUE
4       TRUE
5       TRUE
6       TRUE</code></pre>
<pre class="r"><code>write.csv(x = Results_RNA_DF_Pro_DF_Merged_NAomit, file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/New_RNA_PRO_DF.csv&quot;)</code></pre>
<pre class="r"><code># Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = Results_RNA_DF_Pro_DF_Merged_NAomit)



# Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.01915728</code></pre>
<pre class="r"><code>#------ Plot for both same sign and not same sign
ggplot2::ggplot(Results_RNA_DF_Pro_DF_Merged_NAomit, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-1">
Past versions of "plots for SIGN tests-1.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#------ Plot for same sign 
SAME_SIGN &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit[Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match == TRUE, ]

# Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = SAME_SIGN)

nrow(SAME_SIGN)/nrow(Results_RNA_DF_Pro_DF_Merged_NAomit)</code></pre>
<pre><code>[1] 0.5412756</code></pre>
<pre class="r"><code># Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.2535798</code></pre>
<pre class="r"><code>ggplot(SAME_SIGN, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-2">
Past versions of "plots for SIGN tests-2.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#------ Plot for opposite signs
OPPOSITE_SIGN &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit[Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match == FALSE, ]

# Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = OPPOSITE_SIGN)



# Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.2279661</code></pre>
<pre class="r"><code>ggplot(OPPOSITE_SIGN, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-3">
Past versions of "plots for SIGN tests-3.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot for Same sign &amp; DE in both cases 
SAME_Sign_DE &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit[(Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match == TRUE),  ]

SAME_Sign_DE_2 &lt;- SAME_Sign_DE[ (SAME_Sign_DE$RNA_Pval &lt; 0.05) &amp; (SAME_Sign_DE$Pro_Pval &lt; 0.05),  ]

# Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = SAME_Sign_DE_2)



# Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.5666013</code></pre>
<pre class="r"><code>ggplot(SAME_Sign_DE_2, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-4">
Past versions of "plots for SIGN tests-4.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot for opposite sign &amp; DE 
SAME_Sign_DE &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit[(Results_RNA_DF_Pro_DF_Merged_NAomit$sign_match == FALSE),  ]

SAME_Sign_DE_2 &lt;- SAME_Sign_DE[ (SAME_Sign_DE$RNA_Pval &lt; 0.05) &amp; (SAME_Sign_DE$Pro_Pval &lt; 0.05),  ]

SAME_Sign_DE_2 %&gt;% dim()</code></pre>
<pre><code>[1] 188   7</code></pre>
<pre class="r"><code># Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = SAME_Sign_DE_2)



# Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.5868337</code></pre>
<pre class="r"><code>ggplot(SAME_Sign_DE_2, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-5">
Past versions of "plots for SIGN tests-5.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#---------- Plot for DE 
RNA_Pro_DE &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit[(Results_RNA_DF_Pro_DF_Merged_NAomit$RNA_Pval &lt; 0.05) &amp; (Results_RNA_DF_Pro_DF_Merged_NAomit$Pro_Pval &lt; 0.05),  ]




# Fit linear model
model_LogFC &lt;- lm(RNA_LogFC ~ Pro_LogFC, data = RNA_Pro_DE)



# Get R-squared
r2 &lt;- summary(model_LogFC)$r.squared
r2</code></pre>
<pre><code>[1] 0.1018965</code></pre>
<pre class="r"><code>ggplot(RNA_Pro_DE, aes(x = Pro_LogFC, y = RNA_LogFC, color = sign_match)) +
  geom_point(size = 2, alpha = 0.5) +
  geom_smooth(method = &quot;lm&quot;, se = FALSE, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  annotate(&quot;text&quot;, x = Inf, y = Inf, vjust = 2, hjust = 2,
           label = paste(&quot;R^2 = &quot;, round(r2, digits = 2), sep = &quot;&quot;),
           color = &quot;black&quot;) +
  scale_color_manual(values = c(&quot;red&quot;, &quot;blue&quot;)) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme_bw()</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/plots%20for%20SIGN%20tests-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-SIGN-tests-6">
Past versions of "plots for SIGN tests-6.png"
</button>
</p>
<div id="fig-no-spaces-plots-for-SIGN-tests-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/plots for SIGN tests-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#### Read in GWAS Sumstats #### 
# Make an object for each GWAS sumstat of interest that 
# is just a vecctor of unique rsIDs. 
AC_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/eQTLs/NIHMS827139-ACGWAS.csv&quot;, header = TRUE)
AC_GWAS &lt;- AC_GWAS$RSID

gluc2hr_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/2hr_gluc_GWAS.csv&quot;, header = TRUE)
gluc2hr_GWAS &lt;- gluc2hr_GWAS$SNP %&gt;% unique()

Fast_gluc_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Fast_gluc_GWAS.csv&quot;, header = TRUE)
Fast_gluc_GWAS &lt;- Fast_gluc_GWAS$X.Uploaded_variation %&gt;% unique()

Fast_ins_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Fasting_ins_GWAS.csv&quot;, header = TRUE)
Fast_ins_GWAS &lt;- Fast_ins_GWAS$SNP %&gt;% unique()

Long_gluc_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Long_term_gluc_GWAS.csv&quot;, header = TRUE)
Long_gluc_GWAS &lt;- Long_gluc_GWAS$SNP %&gt;% unique()

HF_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/HF_GWAS_GCST90162626.csv&quot;, header = TRUE)
HF_GWAS&lt;- HF_GWAS$riskAllele

Arrythmia_GWAS &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Arrythmia_GWAS_pubmedId_36653681.csv&quot;, header = TRUE)
Arrythmia_GWAS &lt;- Arrythmia_GWAS$SNPS





# Getting close genes to SNPs using multiple DFs 
snpmart = useEnsembl(biomart = &quot;snp&quot;, dataset=&quot;hsapiens_snp&quot;, version=110)

# Example list of SNP sets
snp_sets &lt;- list(
  AC_GWAS = AC_GWAS,
  gluc2hr_GWAS = gluc2hr_GWAS,
  Fast_gluc_GWAS = Fast_gluc_GWAS, 
  Fast_ins_GWAS=Fast_ins_GWAS,
  Long_gluc_GWAS=Long_gluc_GWAS, 
  HF_GWAS = HF_GWAS, 
  Arrythmia_GWAS=Arrythmia_GWAS
)

# Initialize a list to store the results
snp_locations_list &lt;- list()

# Loop over the SNP sets
for (set_name in names(snp_sets)) {
  snps &lt;- snp_sets[[set_name]]
  
  # Retrieve SNP locations for the current set
  snp_locations &lt;- getBM(attributes = c(&#39;refsnp_id&#39;, &#39;chr_name&#39;, &#39;chrom_start&#39;, &#39;chrom_end&#39;),
                         filters = &#39;snp_filter&#39;,
                         values = snps,
                         mart = snpmart)
  
  # Store the results in the list with a dynamic name
  snp_locations_list[[paste(set_name, &quot;snp_locations&quot;, sep = &quot;_&quot;)]] &lt;- snp_locations
}




# List loop 
# Initialize a list to store BED data frames
bed_df_list &lt;- list()


# Loop over each item in snp_locations_list
for (name in names(snp_locations_list)) {
  # Extract the SNP location data
  snp_locations &lt;- snp_locations_list[[name]]
  
  # Create the BED format data frame
  bed_df &lt;- data.frame(
    chr = snp_locations$chr_name,
    start = snp_locations$chrom_start,
    end = snp_locations$chrom_end + 1,
    name = snp_locations$refsnp_id
  )
  
  # Store the BED data frame in the list with a dynamic name
  bed_df_list[[paste(name, &quot;bed_df&quot;, sep = &quot;_&quot;)]] &lt;- bed_df
}




#### Get the mapped genes #### 
# Connect to the Ensembl database
ensembl &lt;- useEnsembl(biomart = &quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)


# Assuming you have a reasonable range to search for nearby genes
search_range &lt;- 100000  # 100kb assumed to be cis-reg on the chromosome


# Initialize an empty list to store closest genes for each BED data frame
closest_genes_list &lt;- list()


# Loop over each BED data frame in the list
for (bed_name in names(bed_df_list)) {
  bed_df &lt;- bed_df_list[[bed_name]]
  closest_genes &lt;- data.frame(SNP = character(), Gene = character(), stringsAsFactors = FALSE)
  
  # Loop through each SNP location in the current BED data frame
  for (i in 1:nrow(bed_df)) {
    snp &lt;- bed_df[i, ]
    genes &lt;- getBM(
      attributes = c(&#39;external_gene_name&#39;, &#39;chromosome_name&#39;, &#39;start_position&#39;, &#39;end_position&#39;),
      filters = c(&#39;chromosome_name&#39;, &#39;start&#39;, &#39;end&#39;),
      values = list(snp$chr, max(1, snp$start - search_range), snp$end + search_range),
      mart = ensembl
    )
    
    # If genes are found, select the closest one
    if (nrow(genes) &gt; 0) {
      genes$distance &lt;- abs(genes$start_position - snp$start)
      closest_gene &lt;- genes[which.min(genes$distance), &#39;external_gene_name&#39;]
      closest_genes &lt;- rbind(closest_genes, data.frame(SNP = snp$name, Gene = closest_gene))
    } else {
      closest_genes &lt;- rbind(closest_genes, data.frame(SNP = snp$name, Gene = NA))
    }
  }
  
  # Store the closest genes data frame in the list with a dynamic name
  closest_genes_list[[bed_name]] &lt;- closest_genes
}</code></pre>
<pre class="r"><code># Read in data frame curated from (https://msgp.pt/)
MSG &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/SG_RBP_AUtoph/MSG.csv&quot;, header = TRUE)

New_RNA_PRO_DF &lt;- Results_RNA_DF_Pro_DF_Merged_NAomit


# 1. Generate lists of gene-protein pairs with interesting functional genomic properties
#*DE proteins
MSG_DE &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$Pro_Pval &lt; 0.05, ]

#*DE proteins same sign 
MSG_DE_SAME &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$sign_match == TRUE) &amp; (New_RNA_PRO_DF$Pro_Pval &lt; 0.05), ]

#*DE proteins opposite sign 
MSG_DE_OPPOSITE &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$sign_match == FALSE) &amp; (New_RNA_PRO_DF$Pro_Pval &lt; 0.05), ]




# 2. Generate sets from MSG data frame 
#* MSG (All)
MSG %&gt;% head()</code></pre>
<pre><code>  SYMBOL   RPB Autophagy Heat_shock Uniprot
1    LC3 FALSE      TRUE      FALSE  Q9H492
2 ZNF638  TRUE     FALSE      FALSE  Q14966
3 ZC3H7B  TRUE     FALSE      FALSE  Q9UGR2
4 ZC3H7A  TRUE     FALSE      FALSE  Q8IWR0
5 ZC3H14  TRUE     FALSE      FALSE  Q6PJT7
6  YWHAQ FALSE     FALSE      FALSE  P27348
                                                  NAME
1 Microtubule associated protein 1 light chain 3 alpha
2                              Zinc finger protein 638
3        Zinc finger CCCH domain-containing protein 7B
4        Zinc finger CCCH domain-containing protein 7A
5        Zinc finger CCCH domain-containing protein 14
6                                 14-3-3 protein theta</code></pre>
<pre class="r"><code>#* RPB
MSG_RPB &lt;- MSG[MSG$RPB == TRUE, ]

#* Autophagy
MSG_Auto &lt;- MSG[MSG$Autophagy == TRUE, ]

#* Heat shock 
MSG_HSP &lt;- MSG[MSG$Heat_shock == TRUE, ]


#### DE Proteins #### 
SAME_Sign_DE &lt;- MSG_DE

Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# 1. MSG (All)
#**************************
MSG_Exp &lt;- intersect(Expressed_protein, MSG$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00151&quot; &quot;O00303&quot; &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O14744&quot; &quot;O14787&quot;
  [9] &quot;O15143&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60216&quot; &quot;O60506&quot; &quot;O60684&quot;
 [17] &quot;O60716&quot; &quot;O60828&quot; &quot;O60832&quot; &quot;O75369&quot; &quot;O75531&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot;
 [25] &quot;O76021&quot; &quot;O94901&quot; &quot;O95373&quot; &quot;O95758&quot; &quot;O95793&quot; &quot;O95817&quot; &quot;O95819&quot; &quot;O96013&quot;
 [33] &quot;P02511&quot; &quot;P02545&quot; &quot;P04083&quot; &quot;P04792&quot; &quot;P05556&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot;
 [41] &quot;P07900&quot; &quot;P07947&quot; &quot;P07951&quot; &quot;P08133&quot; &quot;P09651&quot; &quot;P09874&quot; &quot;P10599&quot; &quot;P10809&quot;
 [49] &quot;P11586&quot; &quot;P11940&quot; &quot;P12004&quot; &quot;P13489&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot;
 [57] &quot;P16989&quot; &quot;P17096&quot; &quot;P17252&quot; &quot;P17987&quot; &quot;P18754&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P20073&quot;
 [65] &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23193&quot; &quot;P23246&quot; &quot;P23396&quot; &quot;P23528&quot; &quot;P23588&quot;
 [73] &quot;P24941&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27348&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P28370&quot; &quot;P29558&quot;
 [81] &quot;P30041&quot; &quot;P30153&quot; &quot;P30626&quot; &quot;P30876&quot; &quot;P31483&quot; &quot;P31946&quot; &quot;P31948&quot; &quot;P32322&quot;
 [89] &quot;P33176&quot; &quot;P33991&quot; &quot;P33992&quot; &quot;P33993&quot; &quot;P34932&quot; &quot;P35080&quot; &quot;P35637&quot; &quot;P37108&quot;
 [97] &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P40938&quot; &quot;P42025&quot; &quot;P42224&quot; &quot;P42345&quot; &quot;P43487&quot;
[105] &quot;P45974&quot; &quot;P46013&quot; &quot;P47755&quot; &quot;P48634&quot; &quot;P49411&quot; &quot;P49458&quot; &quot;P50479&quot; &quot;P50552&quot;
[113] &quot;P51114&quot; &quot;P51116&quot; &quot;P51812&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52294&quot; &quot;P52701&quot; &quot;P52948&quot;
[121] &quot;P53582&quot; &quot;P53992&quot; &quot;P54577&quot; &quot;P54886&quot; &quot;P55060&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot;
[129] &quot;P55795&quot; &quot;P55884&quot; &quot;P56192&quot; &quot;P58107&quot; &quot;P60228&quot; &quot;P60981&quot; &quot;P61163&quot; &quot;P61247&quot;
[137] &quot;P61586&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62820&quot; &quot;P62879&quot; &quot;P63244&quot;
[145] &quot;P67809&quot; &quot;P78344&quot; &quot;P83916&quot; &quot;P84103&quot; &quot;P98194&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot;
[153] &quot;Q01518&quot; &quot;Q01844&quot; &quot;Q03001&quot; &quot;Q04637&quot; &quot;Q04917&quot; &quot;Q05048&quot; &quot;Q06787&quot; &quot;Q06830&quot;
[161] &quot;Q07666&quot; &quot;Q07866&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q09028&quot; &quot;Q0ZGT2&quot; &quot;Q13123&quot; &quot;Q13148&quot;
[169] &quot;Q13200&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13347&quot; &quot;Q13409&quot; &quot;Q13425&quot; &quot;Q13464&quot; &quot;Q13501&quot;
[177] &quot;Q13642&quot; &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14195&quot; &quot;Q14203&quot;
[185] &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot; &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot;
[193] &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot; &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16555&quot; &quot;Q16630&quot;
[201] &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q562R1&quot; &quot;Q5T8P6&quot; &quot;Q63ZY3&quot; &quot;Q69YQ0&quot; &quot;Q6NYC8&quot;
[209] &quot;Q6P1J9&quot; &quot;Q6P2E9&quot; &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot;
[217] &quot;Q7L2H7&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot; &quot;Q86SQ0&quot; &quot;Q86V48&quot; &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot;
[225] &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N122&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot; &quot;Q8NF91&quot; &quot;Q8WUM4&quot;
[233] &quot;Q8WWM7&quot; &quot;Q8WX93&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92621&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot;
[241] &quot;Q92945&quot; &quot;Q92973&quot; &quot;Q93052&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96FJ2&quot; &quot;Q96HC4&quot;
[249] &quot;Q96HS1&quot; &quot;Q96I24&quot; &quot;Q96IZ0&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot; &quot;Q99459&quot; &quot;Q99623&quot;
[257] &quot;Q99653&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q99959&quot; &quot;Q99996&quot; &quot;Q9BQ39&quot; &quot;Q9BR76&quot;
[265] &quot;Q9BTD8&quot; &quot;Q9BTT0&quot; &quot;Q9BUJ2&quot; &quot;Q9BV86&quot; &quot;Q9BX40&quot; &quot;Q9BX66&quot; &quot;Q9BXP5&quot; &quot;Q9C0C2&quot;
[273] &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9H361&quot; &quot;Q9H492&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NTI5&quot;
[281] &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot; &quot;Q9NUU7&quot; &quot;Q9NVI7&quot; &quot;Q9NX63&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot;
[289] &quot;Q9UBN7&quot; &quot;Q9UBQ5&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UHD1&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot; &quot;Q9UN86&quot;
[297] &quot;Q9UNF1&quot; &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y266&quot; &quot;Q9Y2U8&quot; &quot;Q9Y2V2&quot;
[305] &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y520&quot; &quot;Q9Y570&quot; &quot;Q9Y5A9&quot; &quot;Q9Y5V3&quot; &quot;Q9Y696&quot;
[313] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 123</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 1012</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 190</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 2830</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 23.823, df = 1, p-value = 1.056e-06</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1.78e-06
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.414457 2.310042
sample estimates:
odds ratio 
  1.809954 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  MSG      123    190
  Non-MSG 1012   2830</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)


# Table to percentages 
table_percent &lt;- as.table(matrix(c(A/(A+B), B/(A+B), C/(C+D), D/(C+D)), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))

# Convert the table to a data frame with percentages 
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table_percent))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)



# Create stacked bar chart
ggplot(df_mat, aes(x=Group, y=value, fill=key)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme_minimal() +
  ylab(&quot;Frequency&quot;) +
  xlab(&quot;Category&quot;) +
  ggtitle(&quot;DE_PRO_Stress_grranule_proteome&quot;) +
  scale_fill_discrete(name=&quot;Key&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG%20enrichment%20tests-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MSG-enrichment-tests-1">
Past versions of "MSG enrichment tests-1.png"
</button>
</p>
<div id="fig-no-spaces-MSG-enrichment-tests-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG enrichment tests-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 2. RNA Binding Proteins RPBs 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_RPB$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00425&quot; &quot;O00571&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60506&quot;
  [9] &quot;O60832&quot; &quot;O75369&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot; &quot;O76021&quot; &quot;O95758&quot; &quot;O95793&quot;
 [17] &quot;P04083&quot; &quot;P04792&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot; &quot;P07900&quot; &quot;P09651&quot; &quot;P09874&quot;
 [25] &quot;P10599&quot; &quot;P10809&quot; &quot;P11940&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot; &quot;P16989&quot;
 [33] &quot;P17987&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23246&quot; &quot;P23396&quot;
 [41] &quot;P23588&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P29558&quot; &quot;P30876&quot; &quot;P31483&quot;
 [49] &quot;P31948&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P46013&quot; &quot;P48634&quot;
 [57] &quot;P49411&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52948&quot; &quot;P54577&quot;
 [65] &quot;P54886&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot; &quot;P55795&quot; &quot;P55884&quot; &quot;P58107&quot; &quot;P60228&quot;
 [73] &quot;P61247&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62879&quot; &quot;P63244&quot; &quot;P67809&quot;
 [81] &quot;P78344&quot; &quot;P84103&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q04637&quot; &quot;Q05048&quot;
 [89] &quot;Q06787&quot; &quot;Q07666&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q13148&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13425&quot;
 [97] &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot;
[105] &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot; &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot;
[113] &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16630&quot; &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q5T8P6&quot;
[121] &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot;
[129] &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot; &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot;
[137] &quot;Q8NF91&quot; &quot;Q8WWM7&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot; &quot;Q92945&quot;
[145] &quot;Q92973&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96I24&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot;
[153] &quot;Q99459&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q9BQ39&quot; &quot;Q9BTD8&quot; &quot;Q9BUJ2&quot; &quot;Q9BX40&quot;
[161] &quot;Q9BXP5&quot; &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot;
[169] &quot;Q9NUU7&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot;
[177] &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y5A9&quot;
[185] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 76</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 1059</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 109</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 2911</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 17.758, df = 1, p-value = 2.509e-05</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 4.39e-05
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.398233 2.615460
sample estimates:
odds ratio 
  1.916364 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;RPB&quot;, &quot;Non-RPB&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  RPB       76    109
  Non-RPB 1059   2911</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + RPB&quot;, &quot;DE ~ RPB&quot;, &quot;~ DE + RPB&quot;, &quot;~ DE ~ RPB&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)


# Table to percentages 
table_percent &lt;- as.table(matrix(c(A/(A+B), B/(A+B), C/(C+D), D/(C+D)), nrow = 2,
                                 dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                                 Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))

# Convert the table to a data frame with percentages 
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table_percent))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)



# Create stacked bar chart
ggplot(df_mat, aes(x=Group, y=value, fill=key)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme_minimal() +
  ylab(&quot;Frequency&quot;) +
  xlab(&quot;Category&quot;) +
  ggtitle(&quot;Contingency Table Stacked Bar Chart&quot;) +
  scale_fill_discrete(name=&quot;Key&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG%20enrichment%20tests-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MSG-enrichment-tests-2">
Past versions of "MSG enrichment tests-2.png"
</button>
</p>
<div id="fig-no-spaces-MSG-enrichment-tests-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG enrichment tests-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 3. Autophagy proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_Auto$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O60684&quot; &quot;O95817&quot; &quot;O96013&quot; &quot;P06730&quot; &quot;P07947&quot; &quot;P15170&quot; &quot;P19525&quot; &quot;P24941&quot;
 [9] &quot;P30153&quot; &quot;P42345&quot; &quot;P51812&quot; &quot;P53992&quot; &quot;P55072&quot; &quot;P62820&quot; &quot;Q01518&quot; &quot;Q13501&quot;
[17] &quot;Q562R1&quot; &quot;Q8N122&quot; &quot;Q92879&quot; &quot;Q96HS1&quot; &quot;Q99873&quot; &quot;Q9H492&quot; &quot;Q9UBN7&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 6</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 1129</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 18</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3002</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.00066092, df = 1, p-value = 0.9795</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2872762 2.3383032
sample estimates:
odds ratio 
  0.886357 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;Autophagy&quot;, &quot;Non-Autophagy&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>               Protein
Gene              DE Non-DE
  Autophagy        6     18
  Non-Autophagy 1129   3002</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + Autophagy&quot;, &quot;DE ~ Autophagy&quot;, &quot;~ DE + Autophagy&quot;, &quot;~ DE ~ Autophagy&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)


# Table to percentages 
table_percent &lt;- as.table(matrix(c(A/(A+B), B/(A+B), C/(C+D), D/(C+D)), nrow = 2,
                                 dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                                 Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))

# Convert the table to a data frame with percentages 
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table_percent))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)



# Create stacked bar chart
ggplot(df_mat, aes(x=Group, y=value, fill=key)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme_minimal() +
  ylab(&quot;Frequency&quot;) +
  xlab(&quot;Category&quot;) +
  ggtitle(&quot;Contingency Table Stacked Bar Chart&quot;) +
  scale_fill_discrete(name=&quot;Key&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG%20enrichment%20tests-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MSG-enrichment-tests-3">
Past versions of "MSG enrichment tests-3.png"
</button>
</p>
<div id="fig-no-spaces-MSG-enrichment-tests-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG enrichment tests-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 4. Heat Shock Proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_HSP$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O60506&quot; &quot;P04792&quot; &quot;P07737&quot; &quot;P09651&quot; &quot;P11940&quot;
 [9] &quot;P17252&quot; &quot;P31483&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P52292&quot;
[17] &quot;P52294&quot; &quot;P55072&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q13464&quot; &quot;Q13501&quot; &quot;Q14011&quot; &quot;Q14694&quot;
[25] &quot;Q14974&quot; &quot;Q15366&quot; &quot;Q15717&quot; &quot;Q16637&quot; &quot;Q8IX12&quot; &quot;Q8IZH2&quot; &quot;Q92499&quot; &quot;Q92804&quot;
[33] &quot;Q92879&quot; &quot;Q99700&quot; &quot;Q99996&quot; &quot;Q9H2U1&quot; &quot;Q9NR56&quot; &quot;Q9UBN7&quot; &quot;Q9UN86&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 1118</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 23</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 2997</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 3.9493, df = 1, p-value = 0.04689</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.04766
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.9894692 3.8903963
sample estimates:
odds ratio 
  1.981089 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;HSP&quot;, &quot;Non-HSP&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  HSP       17     23
  Non-HSP 1118   2997</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + HSP&quot;, &quot;DE ~ HSP&quot;, &quot;~ DE + HSP&quot;, &quot;~ DE ~ HSP&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)


# Table to percentages 
table_percent &lt;- as.table(matrix(c(A/(A+B), B/(A+B), C/(C+D), D/(C+D)), nrow = 2,
                                 dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                                 Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))

# Convert the table to a data frame with percentages 
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table_percent))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)




#### DE proteins SAME SIGN ####
SAME_Sign_DE &lt;- MSG_DE_SAME
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot


# 1. MSG (All)
#**************************
MSG_Exp &lt;- intersect(Expressed_protein, MSG$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00151&quot; &quot;O00303&quot; &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O14744&quot; &quot;O14787&quot;
  [9] &quot;O15143&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60216&quot; &quot;O60506&quot; &quot;O60684&quot;
 [17] &quot;O60716&quot; &quot;O60828&quot; &quot;O60832&quot; &quot;O75369&quot; &quot;O75531&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot;
 [25] &quot;O76021&quot; &quot;O94901&quot; &quot;O95373&quot; &quot;O95758&quot; &quot;O95793&quot; &quot;O95817&quot; &quot;O95819&quot; &quot;O96013&quot;
 [33] &quot;P02511&quot; &quot;P02545&quot; &quot;P04083&quot; &quot;P04792&quot; &quot;P05556&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot;
 [41] &quot;P07900&quot; &quot;P07947&quot; &quot;P07951&quot; &quot;P08133&quot; &quot;P09651&quot; &quot;P09874&quot; &quot;P10599&quot; &quot;P10809&quot;
 [49] &quot;P11586&quot; &quot;P11940&quot; &quot;P12004&quot; &quot;P13489&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot;
 [57] &quot;P16989&quot; &quot;P17096&quot; &quot;P17252&quot; &quot;P17987&quot; &quot;P18754&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P20073&quot;
 [65] &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23193&quot; &quot;P23246&quot; &quot;P23396&quot; &quot;P23528&quot; &quot;P23588&quot;
 [73] &quot;P24941&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27348&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P28370&quot; &quot;P29558&quot;
 [81] &quot;P30041&quot; &quot;P30153&quot; &quot;P30626&quot; &quot;P30876&quot; &quot;P31483&quot; &quot;P31946&quot; &quot;P31948&quot; &quot;P32322&quot;
 [89] &quot;P33176&quot; &quot;P33991&quot; &quot;P33992&quot; &quot;P33993&quot; &quot;P34932&quot; &quot;P35080&quot; &quot;P35637&quot; &quot;P37108&quot;
 [97] &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P40938&quot; &quot;P42025&quot; &quot;P42224&quot; &quot;P42345&quot; &quot;P43487&quot;
[105] &quot;P45974&quot; &quot;P46013&quot; &quot;P47755&quot; &quot;P48634&quot; &quot;P49411&quot; &quot;P49458&quot; &quot;P50479&quot; &quot;P50552&quot;
[113] &quot;P51114&quot; &quot;P51116&quot; &quot;P51812&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52294&quot; &quot;P52701&quot; &quot;P52948&quot;
[121] &quot;P53582&quot; &quot;P53992&quot; &quot;P54577&quot; &quot;P54886&quot; &quot;P55060&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot;
[129] &quot;P55795&quot; &quot;P55884&quot; &quot;P56192&quot; &quot;P58107&quot; &quot;P60228&quot; &quot;P60981&quot; &quot;P61163&quot; &quot;P61247&quot;
[137] &quot;P61586&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62820&quot; &quot;P62879&quot; &quot;P63244&quot;
[145] &quot;P67809&quot; &quot;P78344&quot; &quot;P83916&quot; &quot;P84103&quot; &quot;P98194&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot;
[153] &quot;Q01518&quot; &quot;Q01844&quot; &quot;Q03001&quot; &quot;Q04637&quot; &quot;Q04917&quot; &quot;Q05048&quot; &quot;Q06787&quot; &quot;Q06830&quot;
[161] &quot;Q07666&quot; &quot;Q07866&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q09028&quot; &quot;Q0ZGT2&quot; &quot;Q13123&quot; &quot;Q13148&quot;
[169] &quot;Q13200&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13347&quot; &quot;Q13409&quot; &quot;Q13425&quot; &quot;Q13464&quot; &quot;Q13501&quot;
[177] &quot;Q13642&quot; &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14195&quot; &quot;Q14203&quot;
[185] &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot; &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot;
[193] &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot; &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16555&quot; &quot;Q16630&quot;
[201] &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q562R1&quot; &quot;Q5T8P6&quot; &quot;Q63ZY3&quot; &quot;Q69YQ0&quot; &quot;Q6NYC8&quot;
[209] &quot;Q6P1J9&quot; &quot;Q6P2E9&quot; &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot;
[217] &quot;Q7L2H7&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot; &quot;Q86SQ0&quot; &quot;Q86V48&quot; &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot;
[225] &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N122&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot; &quot;Q8NF91&quot; &quot;Q8WUM4&quot;
[233] &quot;Q8WWM7&quot; &quot;Q8WX93&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92621&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot;
[241] &quot;Q92945&quot; &quot;Q92973&quot; &quot;Q93052&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96FJ2&quot; &quot;Q96HC4&quot;
[249] &quot;Q96HS1&quot; &quot;Q96I24&quot; &quot;Q96IZ0&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot; &quot;Q99459&quot; &quot;Q99623&quot;
[257] &quot;Q99653&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q99959&quot; &quot;Q99996&quot; &quot;Q9BQ39&quot; &quot;Q9BR76&quot;
[265] &quot;Q9BTD8&quot; &quot;Q9BTT0&quot; &quot;Q9BUJ2&quot; &quot;Q9BV86&quot; &quot;Q9BX40&quot; &quot;Q9BX66&quot; &quot;Q9BXP5&quot; &quot;Q9C0C2&quot;
[273] &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9H361&quot; &quot;Q9H492&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NTI5&quot;
[281] &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot; &quot;Q9NUU7&quot; &quot;Q9NVI7&quot; &quot;Q9NX63&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot;
[289] &quot;Q9UBN7&quot; &quot;Q9UBQ5&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UHD1&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot; &quot;Q9UN86&quot;
[297] &quot;Q9UNF1&quot; &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y266&quot; &quot;Q9Y2U8&quot; &quot;Q9Y2V2&quot;
[305] &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y520&quot; &quot;Q9Y570&quot; &quot;Q9Y5A9&quot; &quot;Q9Y5V3&quot; &quot;Q9Y696&quot;
[313] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 67</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 600</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 246</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3242</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 6.774, df = 1, p-value = 0.00925</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.01017
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.090538 1.964589
sample estimates:
odds ratio 
  1.471485 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  MSG       67    246
  Non-MSG  600   3242</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + MSG&quot;, &quot;DE ~ MSG&quot;, &quot;~ DE + MSG&quot;, &quot;~ DE ~ MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)




# 2. RNA Binding Proteins RPBs 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_SAME

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_RPB$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00425&quot; &quot;O00571&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60506&quot;
  [9] &quot;O60832&quot; &quot;O75369&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot; &quot;O76021&quot; &quot;O95758&quot; &quot;O95793&quot;
 [17] &quot;P04083&quot; &quot;P04792&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot; &quot;P07900&quot; &quot;P09651&quot; &quot;P09874&quot;
 [25] &quot;P10599&quot; &quot;P10809&quot; &quot;P11940&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot; &quot;P16989&quot;
 [33] &quot;P17987&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23246&quot; &quot;P23396&quot;
 [41] &quot;P23588&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P29558&quot; &quot;P30876&quot; &quot;P31483&quot;
 [49] &quot;P31948&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P46013&quot; &quot;P48634&quot;
 [57] &quot;P49411&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52948&quot; &quot;P54577&quot;
 [65] &quot;P54886&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot; &quot;P55795&quot; &quot;P55884&quot; &quot;P58107&quot; &quot;P60228&quot;
 [73] &quot;P61247&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62879&quot; &quot;P63244&quot; &quot;P67809&quot;
 [81] &quot;P78344&quot; &quot;P84103&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q04637&quot; &quot;Q05048&quot;
 [89] &quot;Q06787&quot; &quot;Q07666&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q13148&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13425&quot;
 [97] &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot;
[105] &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot; &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot;
[113] &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16630&quot; &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q5T8P6&quot;
[121] &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot;
[129] &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot; &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot;
[137] &quot;Q8NF91&quot; &quot;Q8WWM7&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot; &quot;Q92945&quot;
[145] &quot;Q92973&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96I24&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot;
[153] &quot;Q99459&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q9BQ39&quot; &quot;Q9BTD8&quot; &quot;Q9BUJ2&quot; &quot;Q9BX40&quot;
[161] &quot;Q9BXP5&quot; &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot;
[169] &quot;Q9NUU7&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot;
[177] &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y5A9&quot;
[185] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 40</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 627</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 145</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3343</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 4.0335, df = 1, p-value = 0.04461</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.04014
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.9987632 2.1242376
sample estimates:
odds ratio 
  1.470669 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;RPB&quot;, &quot;Non-RPB&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  RPB       40    145
  Non-RPB  627   3343</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + RPB&quot;, &quot;DE ~ RPB&quot;, &quot;~ DE + RPB&quot;, &quot;~ DE ~ RPB&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)




table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  MSG       40    145
  Non-MSG  627   3343</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + MSG&quot;, &quot;DE ~ MSG&quot;, &quot;~ DE + MSG&quot;, &quot;~ DE ~ MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)


# 3. Autophagy proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_SAME

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_Auto$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O60684&quot; &quot;O95817&quot; &quot;O96013&quot; &quot;P06730&quot; &quot;P07947&quot; &quot;P15170&quot; &quot;P19525&quot; &quot;P24941&quot;
 [9] &quot;P30153&quot; &quot;P42345&quot; &quot;P51812&quot; &quot;P53992&quot; &quot;P55072&quot; &quot;P62820&quot; &quot;Q01518&quot; &quot;Q13501&quot;
[17] &quot;Q562R1&quot; &quot;Q8N122&quot; &quot;Q92879&quot; &quot;Q96HS1&quot; &quot;Q99873&quot; &quot;Q9H492&quot; &quot;Q9UBN7&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 663</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3468</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 8.7562e-27, df = 1, p-value = 1</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2591628 3.1367502
sample estimates:
odds ratio 
   1.04615 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;Autophagy&quot;, &quot;Non-Autophagy&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>               Protein
Gene              DE Non-DE
  Autophagy        4     20
  Non-Autophagy  663   3468</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + Autophagy&quot;, &quot;DE ~ Autophagy&quot;, &quot;~ DE + Autophagy&quot;, &quot;~ DE ~ Autophagy&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)





# 4. Heat Shock Proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_SAME

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_HSP$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O60506&quot; &quot;P04792&quot; &quot;P07737&quot; &quot;P09651&quot; &quot;P11940&quot;
 [9] &quot;P17252&quot; &quot;P31483&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P52292&quot;
[17] &quot;P52294&quot; &quot;P55072&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q13464&quot; &quot;Q13501&quot; &quot;Q14011&quot; &quot;Q14694&quot;
[25] &quot;Q14974&quot; &quot;Q15366&quot; &quot;Q15717&quot; &quot;Q16637&quot; &quot;Q8IX12&quot; &quot;Q8IZH2&quot; &quot;Q92499&quot; &quot;Q92804&quot;
[33] &quot;Q92879&quot; &quot;Q99700&quot; &quot;Q99996&quot; &quot;Q9H2U1&quot; &quot;Q9NR56&quot; &quot;Q9UBN7&quot; &quot;Q9UN86&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 6</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 661</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 34</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3454</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 5.1699e-30, df = 1, p-value = 1</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.3151875 2.2337800
sample estimates:
odds ratio 
 0.9221255 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;HSP&quot;, &quot;Non-HSP&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  HSP        6     34
  Non-HSP  661   3454</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + HSP&quot;, &quot;DE ~ HSP&quot;, &quot;~ DE + HSP&quot;, &quot;~ DE ~ HSP&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)



# Create stacked bar chart
ggplot(df_mat, aes(x=Group, y=value, fill=key)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme_minimal() +
  ylab(&quot;Frequency&quot;) +
  xlab(&quot;Category&quot;) +
  ggtitle(&quot;&quot;) +
  scale_fill_discrete(name=&quot;Key&quot;) +
  theme(legend.position=&quot;bottom&quot;)</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG%20enrichment%20tests-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MSG-enrichment-tests-4">
Past versions of "MSG enrichment tests-4.png"
</button>
</p>
<div id="fig-no-spaces-MSG-enrichment-tests-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/MSG enrichment tests-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#### DE_OPPOSSITE Proteins ####
SAME_Sign_DE &lt;- MSG_DE_OPPOSITE
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot


# 1. MSG (All)
#**************************
MSG_Exp &lt;- intersect(Expressed_protein, MSG$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00151&quot; &quot;O00303&quot; &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O14744&quot; &quot;O14787&quot;
  [9] &quot;O15143&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60216&quot; &quot;O60506&quot; &quot;O60684&quot;
 [17] &quot;O60716&quot; &quot;O60828&quot; &quot;O60832&quot; &quot;O75369&quot; &quot;O75531&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot;
 [25] &quot;O76021&quot; &quot;O94901&quot; &quot;O95373&quot; &quot;O95758&quot; &quot;O95793&quot; &quot;O95817&quot; &quot;O95819&quot; &quot;O96013&quot;
 [33] &quot;P02511&quot; &quot;P02545&quot; &quot;P04083&quot; &quot;P04792&quot; &quot;P05556&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot;
 [41] &quot;P07900&quot; &quot;P07947&quot; &quot;P07951&quot; &quot;P08133&quot; &quot;P09651&quot; &quot;P09874&quot; &quot;P10599&quot; &quot;P10809&quot;
 [49] &quot;P11586&quot; &quot;P11940&quot; &quot;P12004&quot; &quot;P13489&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot;
 [57] &quot;P16989&quot; &quot;P17096&quot; &quot;P17252&quot; &quot;P17987&quot; &quot;P18754&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P20073&quot;
 [65] &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23193&quot; &quot;P23246&quot; &quot;P23396&quot; &quot;P23528&quot; &quot;P23588&quot;
 [73] &quot;P24941&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27348&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P28370&quot; &quot;P29558&quot;
 [81] &quot;P30041&quot; &quot;P30153&quot; &quot;P30626&quot; &quot;P30876&quot; &quot;P31483&quot; &quot;P31946&quot; &quot;P31948&quot; &quot;P32322&quot;
 [89] &quot;P33176&quot; &quot;P33991&quot; &quot;P33992&quot; &quot;P33993&quot; &quot;P34932&quot; &quot;P35080&quot; &quot;P35637&quot; &quot;P37108&quot;
 [97] &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P40938&quot; &quot;P42025&quot; &quot;P42224&quot; &quot;P42345&quot; &quot;P43487&quot;
[105] &quot;P45974&quot; &quot;P46013&quot; &quot;P47755&quot; &quot;P48634&quot; &quot;P49411&quot; &quot;P49458&quot; &quot;P50479&quot; &quot;P50552&quot;
[113] &quot;P51114&quot; &quot;P51116&quot; &quot;P51812&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52294&quot; &quot;P52701&quot; &quot;P52948&quot;
[121] &quot;P53582&quot; &quot;P53992&quot; &quot;P54577&quot; &quot;P54886&quot; &quot;P55060&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot;
[129] &quot;P55795&quot; &quot;P55884&quot; &quot;P56192&quot; &quot;P58107&quot; &quot;P60228&quot; &quot;P60981&quot; &quot;P61163&quot; &quot;P61247&quot;
[137] &quot;P61586&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62820&quot; &quot;P62879&quot; &quot;P63244&quot;
[145] &quot;P67809&quot; &quot;P78344&quot; &quot;P83916&quot; &quot;P84103&quot; &quot;P98194&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot;
[153] &quot;Q01518&quot; &quot;Q01844&quot; &quot;Q03001&quot; &quot;Q04637&quot; &quot;Q04917&quot; &quot;Q05048&quot; &quot;Q06787&quot; &quot;Q06830&quot;
[161] &quot;Q07666&quot; &quot;Q07866&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q09028&quot; &quot;Q0ZGT2&quot; &quot;Q13123&quot; &quot;Q13148&quot;
[169] &quot;Q13200&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13347&quot; &quot;Q13409&quot; &quot;Q13425&quot; &quot;Q13464&quot; &quot;Q13501&quot;
[177] &quot;Q13642&quot; &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14195&quot; &quot;Q14203&quot;
[185] &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot; &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot;
[193] &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot; &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16555&quot; &quot;Q16630&quot;
[201] &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q562R1&quot; &quot;Q5T8P6&quot; &quot;Q63ZY3&quot; &quot;Q69YQ0&quot; &quot;Q6NYC8&quot;
[209] &quot;Q6P1J9&quot; &quot;Q6P2E9&quot; &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot;
[217] &quot;Q7L2H7&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot; &quot;Q86SQ0&quot; &quot;Q86V48&quot; &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot;
[225] &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N122&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot; &quot;Q8NF91&quot; &quot;Q8WUM4&quot;
[233] &quot;Q8WWM7&quot; &quot;Q8WX93&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92621&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot;
[241] &quot;Q92945&quot; &quot;Q92973&quot; &quot;Q93052&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96FJ2&quot; &quot;Q96HC4&quot;
[249] &quot;Q96HS1&quot; &quot;Q96I24&quot; &quot;Q96IZ0&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot; &quot;Q99459&quot; &quot;Q99623&quot;
[257] &quot;Q99653&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q99959&quot; &quot;Q99996&quot; &quot;Q9BQ39&quot; &quot;Q9BR76&quot;
[265] &quot;Q9BTD8&quot; &quot;Q9BTT0&quot; &quot;Q9BUJ2&quot; &quot;Q9BV86&quot; &quot;Q9BX40&quot; &quot;Q9BX66&quot; &quot;Q9BXP5&quot; &quot;Q9C0C2&quot;
[273] &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9H361&quot; &quot;Q9H492&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NTI5&quot;
[281] &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot; &quot;Q9NUU7&quot; &quot;Q9NVI7&quot; &quot;Q9NX63&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot;
[289] &quot;Q9UBN7&quot; &quot;Q9UBQ5&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UHD1&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot; &quot;Q9UN86&quot;
[297] &quot;Q9UNF1&quot; &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y266&quot; &quot;Q9Y2U8&quot; &quot;Q9Y2V2&quot;
[305] &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y520&quot; &quot;Q9Y570&quot; &quot;Q9Y5A9&quot; &quot;Q9Y5V3&quot; &quot;Q9Y696&quot;
[313] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 56</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 412</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 257</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3430</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 14.169, df = 1, p-value = 0.0001671</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.000266
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.309751 2.478096
sample estimates:
odds ratio 
  1.813673 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  MSG       56    257
  Non-MSG  412   3430</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + MSG&quot;, &quot;DE ~ MSG&quot;, &quot;~ DE + MSG&quot;, &quot;~ DE ~ MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)





# 2. RNA Binding Proteins RPBs 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_OPPOSITE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_RPB$Uniprot)
MSG_Exp</code></pre>
<pre><code>  [1] &quot;A5YKK6&quot; &quot;O00425&quot; &quot;O00571&quot; &quot;O15347&quot; &quot;O15371&quot; &quot;O15372&quot; &quot;O43251&quot; &quot;O60506&quot;
  [9] &quot;O60832&quot; &quot;O75369&quot; &quot;O75534&quot; &quot;O75569&quot; &quot;O75821&quot; &quot;O76021&quot; &quot;O95758&quot; &quot;O95793&quot;
 [17] &quot;P04083&quot; &quot;P04792&quot; &quot;P06730&quot; &quot;P07305&quot; &quot;P07737&quot; &quot;P07900&quot; &quot;P09651&quot; &quot;P09874&quot;
 [25] &quot;P10599&quot; &quot;P10809&quot; &quot;P11940&quot; &quot;P13693&quot; &quot;P15170&quot; &quot;P15880&quot; &quot;P15924&quot; &quot;P16989&quot;
 [33] &quot;P17987&quot; &quot;P19525&quot; &quot;P20042&quot; &quot;P22087&quot; &quot;P22314&quot; &quot;P22626&quot; &quot;P23246&quot; &quot;P23396&quot;
 [41] &quot;P23588&quot; &quot;P25705&quot; &quot;P26196&quot; &quot;P27695&quot; &quot;P27816&quot; &quot;P29558&quot; &quot;P30876&quot; &quot;P31483&quot;
 [49] &quot;P31948&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P38646&quot; &quot;P39019&quot; &quot;P40227&quot; &quot;P46013&quot; &quot;P48634&quot;
 [57] &quot;P49411&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P51991&quot; &quot;P52292&quot; &quot;P52948&quot; &quot;P54577&quot;
 [65] &quot;P54886&quot; &quot;P55072&quot; &quot;P55081&quot; &quot;P55265&quot; &quot;P55795&quot; &quot;P55884&quot; &quot;P58107&quot; &quot;P60228&quot;
 [73] &quot;P61247&quot; &quot;P61978&quot; &quot;P62306&quot; &quot;P62633&quot; &quot;P62753&quot; &quot;P62879&quot; &quot;P63244&quot; &quot;P67809&quot;
 [81] &quot;P78344&quot; &quot;P84103&quot; &quot;Q00577&quot; &quot;Q01081&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q04637&quot; &quot;Q05048&quot;
 [89] &quot;Q06787&quot; &quot;Q07666&quot; &quot;Q07955&quot; &quot;Q08J23&quot; &quot;Q13148&quot; &quot;Q13283&quot; &quot;Q13310&quot; &quot;Q13425&quot;
 [97] &quot;Q14011&quot; &quot;Q14103&quot; &quot;Q14151&quot; &quot;Q14152&quot; &quot;Q14157&quot; &quot;Q14204&quot; &quot;Q14258&quot; &quot;Q14444&quot;
[105] &quot;Q14671&quot; &quot;Q14694&quot; &quot;Q14739&quot; &quot;Q14966&quot; &quot;Q14974&quot; &quot;Q15056&quot; &quot;Q15233&quot; &quot;Q15366&quot;
[113] &quot;Q15654&quot; &quot;Q15691&quot; &quot;Q15717&quot; &quot;Q16630&quot; &quot;Q16637&quot; &quot;Q16658&quot; &quot;Q53EP0&quot; &quot;Q5T8P6&quot;
[121] &quot;Q6PJT7&quot; &quot;Q6PKG0&quot; &quot;Q6ZUT6&quot; &quot;Q71RC2&quot; &quot;Q7KZF4&quot; &quot;Q7L2E3&quot; &quot;Q7Z2W4&quot; &quot;Q7Z417&quot;
[129] &quot;Q8IX01&quot; &quot;Q8IX12&quot; &quot;Q8IXT5&quot; &quot;Q8IY81&quot; &quot;Q8IZH2&quot; &quot;Q8N684&quot; &quot;Q8NC51&quot; &quot;Q8NCA5&quot;
[137] &quot;Q8NF91&quot; &quot;Q8WWM7&quot; &quot;Q92499&quot; &quot;Q92522&quot; &quot;Q92804&quot; &quot;Q92879&quot; &quot;Q92900&quot; &quot;Q92945&quot;
[145] &quot;Q92973&quot; &quot;Q93062&quot; &quot;Q96DH6&quot; &quot;Q96EP5&quot; &quot;Q96I24&quot; &quot;Q96PU8&quot; &quot;Q96QC0&quot; &quot;Q96QR8&quot;
[153] &quot;Q99459&quot; &quot;Q99700&quot; &quot;Q99729&quot; &quot;Q99873&quot; &quot;Q9BQ39&quot; &quot;Q9BTD8&quot; &quot;Q9BUJ2&quot; &quot;Q9BX40&quot;
[161] &quot;Q9BXP5&quot; &quot;Q9H0S4&quot; &quot;Q9H2U1&quot; &quot;Q9HCE1&quot; &quot;Q9NR30&quot; &quot;Q9NR56&quot; &quot;Q9NUL3&quot; &quot;Q9NUQ6&quot;
[169] &quot;Q9NUU7&quot; &quot;Q9NZB2&quot; &quot;Q9NZI8&quot; &quot;Q9P258&quot; &quot;Q9UBU9&quot; &quot;Q9UGR2&quot; &quot;Q9UKF6&quot; &quot;Q9UKV8&quot;
[177] &quot;Q9UPN3&quot; &quot;Q9UQ80&quot; &quot;Q9Y224&quot; &quot;Q9Y262&quot; &quot;Q9Y2X3&quot; &quot;Q9Y3F4&quot; &quot;Q9Y3I0&quot; &quot;Q9Y5A9&quot;
[185] &quot;Q9Y6M1&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 36</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 432</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 149</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3538</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 12.169, df = 1, p-value = 0.000486</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.0007713
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.316540 2.907693
sample estimates:
odds ratio 
  1.978372 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;RPB&quot;, &quot;Non-RPB&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  RPB       36    149
  Non-RPB  432   3538</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + RPB&quot;, &quot;DE ~ RPB&quot;, &quot;~ DE + RPB&quot;, &quot;~ DE ~ RPB&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)






# 3. Autophagy proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_OPPOSITE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_Auto$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O60684&quot; &quot;O95817&quot; &quot;O96013&quot; &quot;P06730&quot; &quot;P07947&quot; &quot;P15170&quot; &quot;P19525&quot; &quot;P24941&quot;
 [9] &quot;P30153&quot; &quot;P42345&quot; &quot;P51812&quot; &quot;P53992&quot; &quot;P55072&quot; &quot;P62820&quot; &quot;Q01518&quot; &quot;Q13501&quot;
[17] &quot;Q562R1&quot; &quot;Q8N122&quot; &quot;Q92879&quot; &quot;Q96HS1&quot; &quot;Q99873&quot; &quot;Q9H492&quot; &quot;Q9UBN7&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 466</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3665</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.017321, df = 1, p-value = 0.8953</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.08124933 2.92414046
sample estimates:
odds ratio 
 0.7150334 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;Autophagy&quot;, &quot;Non-Autophagy&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>               Protein
Gene              DE Non-DE
  Autophagy        2     22
  Non-Autophagy  466   3665</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + Autophagy&quot;, &quot;DE ~ Autophagy&quot;, &quot;~ DE + Autophagy&quot;, &quot;~ DE ~ Autophagy&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)






# 4. Heat Shock Proteins 
#**************************
# All expressed proteins 
Expressed_protein &lt;- New_RNA_PRO_DF$uniprotswissprot

# Set I am testing 
SAME_Sign_DE &lt;- MSG_DE_OPPOSITE

# Intersection between MSG proteins and my expressed/detected proteins 
MSG_Exp &lt;- intersect(Expressed_protein, MSG_HSP$Uniprot)
MSG_Exp</code></pre>
<pre><code> [1] &quot;O00425&quot; &quot;O00505&quot; &quot;O00571&quot; &quot;O60506&quot; &quot;P04792&quot; &quot;P07737&quot; &quot;P09651&quot; &quot;P11940&quot;
 [9] &quot;P17252&quot; &quot;P31483&quot; &quot;P35637&quot; &quot;P37108&quot; &quot;P49458&quot; &quot;P51114&quot; &quot;P51116&quot; &quot;P52292&quot;
[17] &quot;P52294&quot; &quot;P55072&quot; &quot;Q01085&quot; &quot;Q01844&quot; &quot;Q13464&quot; &quot;Q13501&quot; &quot;Q14011&quot; &quot;Q14694&quot;
[25] &quot;Q14974&quot; &quot;Q15366&quot; &quot;Q15717&quot; &quot;Q16637&quot; &quot;Q8IX12&quot; &quot;Q8IZH2&quot; &quot;Q92499&quot; &quot;Q92804&quot;
[33] &quot;Q92879&quot; &quot;Q99700&quot; &quot;Q99996&quot; &quot;Q9H2U1&quot; &quot;Q9NR56&quot; &quot;Q9UBN7&quot; &quot;Q9UN86&quot; &quot;Q9UQ80&quot;</code></pre>
<pre class="r"><code># A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(SAME_Sign_DE$uniprotswissprot, MSG_Exp) %&gt;% length()
A</code></pre>
<pre><code>[1] 11</code></pre>
<pre class="r"><code># B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(SAME_Sign_DE$uniprotswissprot) - A
B</code></pre>
<pre><code>[1] 457</code></pre>
<pre class="r"><code># C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(MSG_Exp) - A
C</code></pre>
<pre><code>[1] 29</code></pre>
<pre class="r"><code># D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (nrow(New_RNA_PRO_DF) - length(SAME_Sign_DE$uniprotswissprot)) - C
D</code></pre>
<pre><code>[1] 3658</code></pre>
<pre class="r"><code># Chi-Squared Test
mat &lt;- matrix(c(A, B, C, D), nrow = 2)
chisq.test(mat)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 9.0757, df = 1, p-value = 0.00259</code></pre>
<pre class="r"><code>fisher.test(mat)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.003615
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.358382 6.309658
sample estimates:
odds ratio 
  3.034893 </code></pre>
<pre class="r"><code>table &lt;- as.table(matrix(c(A, B, C, D), nrow = 2,
                         dimnames = list(Gene = c(&quot;HSP&quot;, &quot;Non-HSP&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))
table</code></pre>
<pre><code>         Protein
Gene        DE Non-DE
  HSP       11     29
  Non-HSP  457   3658</code></pre>
<pre class="r"><code># Convert the table to a data frame
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table))
df_mat$key &lt;- c(&quot;DE + HSP&quot;, &quot;DE ~ HSP&quot;, &quot;~ DE + HSP&quot;, &quot;~ DE ~ HSP&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)</code></pre>
<pre class="r"><code>enrichment_test &lt;- function(Test_set, Background_exp, Test_set_2, test_type = &quot;chisq&quot;, plot = FALSE, custom_title = &quot;Enrichment test&quot;) {
  
  # 1. pQTL test
#**************************
# pQTLs that are also expressed proteins 
EXP &lt;- intersect(Background_exp,Test_set_2)


# A = number of Test Set proteins that are in the MSG_Exp list
A &lt;- intersect(Test_set, EXP) %&gt;% length()


# B = number of Test Set proteins that are not in the MSG_Exp list
B &lt;- length(Test_set) - A

# C = number of expressed proteins that are in the MSG_Exp list but are not in Test Set proteins
C &lt;- length(EXP) - A

# D = number of expressed proteins that are not in the MSG_Exp list and are not in Test Set proteins
D &lt;- (length(Background_exp) - length(Test_set)) - C



  # Create the contingency matrix
  mat &lt;- matrix(c(A, B, C, D), nrow = 2, byrow = TRUE,
                dimnames = list(c(&quot;DE&quot;, &quot;Not DE&quot;),
                                c(&quot;In Background&quot;, &quot;Not in Background&quot;)))

  # Perform the selected test
  test_result &lt;- NULL
  if (test_type == &quot;chisq&quot;) {
    test_result &lt;- chisq.test(mat)
  } else if (test_type == &quot;fisher&quot;) {
    test_result &lt;- fisher.test(mat)
  }

  # Plotting (if requested)
  if (plot == TRUE) {
    table_percent &lt;- as.table(matrix(c(A/(A+B), B/(A+B), C/(C+D), D/(C+D)), nrow = 2,
                         dimnames = list(Gene = c(&quot;MSG&quot;, &quot;Non-MSG&quot;),
                                         Protein = c(&quot;DE&quot;, &quot;Non-DE&quot;))))

    # Convert the table to a data frame with percentages 
df_mat &lt;- tidyr::gather(as.data.frame.matrix(table_percent))
df_mat$key &lt;- c(&quot;MSG&quot;, &quot;Non-MSG&quot;, &quot;MSG&quot;, &quot;Non-MSG&quot;)
df_mat$Group &lt;- c(&quot;DE&quot;, &quot;DE&quot;, &quot;~DE&quot;, &quot;~DE&quot;)



# Create stacked bar chart
PLOT &lt;- ggplot(df_mat, aes(x=Group, y=value, fill=key)) +
  geom_bar(stat=&quot;identity&quot;) +
  theme_minimal() +
  ylab(&quot;Frequency&quot;) +
  xlab(&quot;Category&quot;) +
  ggtitle(custom_title) +
  scale_fill_discrete(name=&quot;Key&quot;) +
  theme(legend.position=&quot;bottom&quot;)
  }

  # Return the test result
  return(list(&quot;Test_Result&quot; = test_result, &quot;Plot&quot; = PLOT, &quot;Mat&quot; = mat))
}</code></pre>
<pre class="r"><code># Read in pQTL data 
Olink &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Olink.csv&quot;, header = TRUE)

DE_PRO &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$Pro_Pval &lt; 0.05, ]$uniprotswissprot
BACK_PRO &lt;- New_RNA_PRO_DF$uniprotswissprot

Olink$Protein.panel %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Inflammation_II&quot;    &quot;Cardiometabolic_II&quot; &quot;Oncology&quot;          
[4] &quot;Neurology&quot;          &quot;Oncology_II&quot;        &quot;Cardiometabolic&quot;   
[7] &quot;Inflammation&quot;       &quot;Neurology_II&quot;      </code></pre>
<pre class="r"><code>Cardiometabolic &lt;- Olink[Olink$Protein.panel == &quot;Cardiometabolic&quot;,  ]$UniProt
Cardiometabolic_II &lt;- Olink[Olink$Protein.panel == &quot;Cardiometabolic_II&quot;,  ]$UniProt
Inflammation &lt;- Olink[Olink$Protein.panel == &quot;Inflammation&quot;,  ]$UniProt
Inflammation_II &lt;- Olink[Olink$Protein.panel == &quot;Inflammation_II&quot;,  ]$UniProt

ALL_cardiometabolic &lt;- c(Cardiometabolic_II,Cardiometabolic) %&gt;% unique()
ALL_Inflammation &lt;- c(Inflammation, Inflammation_II) %&gt;% unique()

enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Cardiometabolic, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K Cardiometabolic proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.31454, df = 1, p-value = 0.5749


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-1">
Past versions of "Olinkplatform protein tests-1.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                30              1105
Not DE            69              2951</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Cardiometabolic_II, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K Cardiometabolic_II proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.8255, df = 1, p-value = 0.1767


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-2">
Past versions of "Olinkplatform protein tests-2.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                29              1106
Not DE           104              2916</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Inflammation, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K Inflammation proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.7319, df = 1, p-value = 0.09836


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-3">
Past versions of "Olinkplatform protein tests-3.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                10              1125
Not DE            49              2971</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Inflammation_II, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K Inflammation_II proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.30344, df = 1, p-value = 0.5817


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-4">
Past versions of "Olinkplatform protein tests-4.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                21              1114
Not DE            66              2954</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =ALL_cardiometabolic, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K ALL_cardiometabolic proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.34513, df = 1, p-value = 0.5569


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-5">
Past versions of "Olinkplatform protein tests-5.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                59              1076
Not DE           173              2847</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =ALL_Inflammation, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;OLINK 3 K ALL_Inflammation proteins&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.512, df = 1, p-value = 0.113


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform%20protein%20tests-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Olinkplatform-protein-tests-6">
Past versions of "Olinkplatform protein tests-6.png"
</button>
</p>
<div id="fig-no-spaces-Olinkplatform-protein-tests-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Olinkplatform protein tests-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                31              1104
Not DE           115              2905</code></pre>
<pre class="r"><code># pQTL data from https://www.nature.com/articles/s41586-023-06592-6
# Generate a series of test sets  
DE_PRO &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$Pro_Pval &lt; 0.05, ]$uniprotswissprot

BACK_PRO &lt;- New_RNA_PRO_DF$uniprotswissprot

DE_SAME &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == TRUE), ]$uniprotswissprot 

DE_OPP &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == FALSE), ]$uniprotswissprot 

DE_PRO_DE_RNA &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05), ]$uniprotswissprot 

DE_PRO_DE_RNA_SAME &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == TRUE), ]$uniprotswissprot

DE_PRO_DE_RNA_OPP &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == FALSE), ]$uniprotswissprot

DE_PRO_ONLY &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; !(New_RNA_PRO_DF$RNA_Pval &lt; 0.05), ]$uniprotswissprot


# Sanity check for test sets 
intersect(DE_PRO_DE_RNA, DE_PRO_ONLY )</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># All Unique pQTLs 
pQTL_data &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/pQTL_Genetic_reg_plasmaproteome.csv&quot;, header = TRUE)


pQTL_pro_unique &lt;- pQTL_data$Target.UniProt %&gt;% unique()
pQTL_pro_unique %&gt;% length()</code></pre>
<pre><code>[1] 2526</code></pre>
<pre class="r"><code>intersect(pQTL_pro_unique, BACK_PRO) %&gt;% length()</code></pre>
<pre><code>[1] 688</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.3705, df = 1, p-value = 0.1236


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-1">
Past versions of "pQTL and SIGN Results-1.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE               171               964
Not DE           517              2503</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_SAME in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.054625, df = 1, p-value = 0.8152


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-2">
Past versions of "pQTL and SIGN Results-2.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE               113               554
Not DE           575              2913</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_OPP in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 6.287, df = 1, p-value = 0.01216


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-3">
Past versions of "pQTL and SIGN Results-3.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                58               410
Not DE           630              3057</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.2354, df = 1, p-value = 0.2664


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-4">
Past versions of "pQTL and SIGN Results-4.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                76               439
Not DE           612              3028</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.13314, df = 1, p-value = 0.7152


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-5">
Past versions of "pQTL and SIGN Results-5.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                57               270
Not DE           631              3197</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 5.4537, df = 1, p-value = 0.01953


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-6">
Past versions of "pQTL and SIGN Results-6.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                19               169
Not DE           669              3298</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =pQTL_pro_unique, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in pQTL_pro_unique&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.70379, df = 1, p-value = 0.4015


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL%20and%20SIGN%20Results-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pQTL-and-SIGN-Results-7">
Past versions of "pQTL and SIGN Results-7.png"
</button>
</p>
<div id="fig-no-spaces-pQTL-and-SIGN-Results-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/pQTL and SIGN Results-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                95               525
Not DE           593              2942</code></pre>
<pre class="r"><code># HF GWAS from 
# Enrichment test for HF mapped genes that are detected protein in our set 
AC_TOX &lt;- closest_genes_list$AC_GWAS_snp_locations_bed_df$Gene %&gt;% unique()
AC_TOX</code></pre>
<pre><code> [1] &quot;SPATA42&quot;    &quot;HTR6&quot;       &quot;&quot;           &quot;PRDX6&quot;      &quot;GPC5&quot;      
 [6] &quot;LGMNP1&quot;     &quot;ARF4P3&quot;     &quot;ESD&quot;        NA           &quot;SGO1-AS1&quot;  
[11] &quot;CCDC54-AS1&quot; &quot;LINC02510&quot;  &quot;ATP13A4&quot;    &quot;GOLGA6L2&quot;   &quot;RPL7P48&quot;   
[16] &quot;DENND3-AS1&quot; &quot;PAM&quot;        &quot;NACAP10&quot;    &quot;ADCY2&quot;      &quot;EIF3KP1&quot;   
[21] &quot;ROBO3&quot;      &quot;C6orf15&quot;    &quot;RPL32P19&quot;   &quot;CPA6&quot;       &quot;SPATA19&quot;   
[26] &quot;RN7SL518P&quot;  &quot;RNU4-66P&quot;   &quot;SULF1&quot;      &quot;RPL30P9&quot;    &quot;WDHD1&quot;     
[31] &quot;VN1R12P&quot;    &quot;CDPF1&quot;      &quot;HOXD13&quot;     &quot;PRODH&quot;      &quot;MIR3125&quot;   
[36] &quot;LRRTM4&quot;     &quot;C11orf98P1&quot; &quot;RN7SL459P&quot;  &quot;RNU6-1088P&quot; &quot;CAV1&quot;      
[41] &quot;U3&quot;         &quot;NFE2L3&quot;     &quot;RN7SKP258&quot;  &quot;PLA2G1B&quot;   </code></pre>
<pre class="r"><code># Redefine test sets in terms of gene names 
DE_PRO &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$Pro_Pval &lt; 0.05, ]$hgnc_symbol

BACK_PRO &lt;- New_RNA_PRO_DF$hgnc_symbol

DE_SAME &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == TRUE), ]$hgnc_symbol 

DE_OPP &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == FALSE), ]$hgnc_symbol 

DE_PRO_DE_RNA &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05), ]$hgnc_symbol 

DE_PRO_DE_RNA_SAME &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == TRUE), ]$hgnc_symbol

DE_PRO_DE_RNA_OPP &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$sign_match == FALSE), ]$hgnc_symbol

DE_PRO_ONLY &lt;- New_RNA_PRO_DF[(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; !(New_RNA_PRO_DF$RNA_Pval &lt; 0.05), ]$hgnc_symbol

DE_RNA_ONLY &lt;- New_RNA_PRO_DF[!(New_RNA_PRO_DF$Pro_Pval &lt; 0.05) &amp; (New_RNA_PRO_DF$RNA_Pval &lt; 0.05), ]$hgnc_symbol




# Do enrchment tests for AC_TOX GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 7.9854e-30, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-1">
Past versions of "Enrichment GWAS AC HF TOX-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1              1134
Not DE             4              3016</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_SAME in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 3.9861e-28, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-2">
Past versions of "Enrichment GWAS AC HF TOX-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               666
Not DE             4              3484</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_OPP in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.0079964, df = 1, p-value = 0.9287


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-3">
Past versions of "Enrichment GWAS AC HF TOX-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               468
Not DE             5              3682</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.026438, df = 1, p-value = 0.8708


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-4">
Past versions of "Enrichment GWAS AC HF TOX-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               515
Not DE             5              3635</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.0864e-28, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-5">
Past versions of "Enrichment GWAS AC HF TOX-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               327
Not DE             5              3823</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 3.2913e-30, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-6">
Past versions of "Enrichment GWAS AC HF TOX-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               188
Not DE             5              3962</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 4.3545e-30, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-7">
Past versions of "Enrichment GWAS AC HF TOX-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               619
Not DE             4              3531</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =AC_TOX, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in AC_TOX&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 4.6719e-26, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20AC%20HF%20TOX-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-8">
Past versions of "Enrichment GWAS AC HF TOX-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-AC-HF-TOX-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS AC HF TOX-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2              1367
Not DE             3              2783</code></pre>
<pre class="r"><code># Enrichment test for HF mapped genes that are detected protein in our set 
HF_GWAS &lt;- closest_genes_list$HF_GWAS_snp_locations_bed_df$Gene %&gt;% unique()
HF_GWAS</code></pre>
<pre><code> [1] &quot;UBA52P6&quot;     &quot;SH2B3&quot;       NA            &quot;MIR6861&quot;     &quot;FAF1-AS1&quot;   
 [6] &quot;MIR874&quot;      &quot;OR2A15P&quot;     &quot;&quot;            &quot;BAZ1A-AS1&quot;   &quot;LYPLA1P2&quot;   
[11] &quot;TXNP1&quot;       &quot;Y_RNA&quot;       &quot;FTO&quot;         &quot;LINC01438&quot;   &quot;SRR&quot;        
[16] &quot;USP36&quot;       &quot;CLCNKA&quot;      &quot;ASXL3&quot;       &quot;PHB1P15&quot;     &quot;ZNF280A&quot;    
[21] &quot;HDGFL1&quot;      &quot;SYNPO2L-AS1&quot; &quot;RNU6-640P&quot;   &quot;LPA&quot;         &quot;POM121C&quot;    
[26] &quot;TMEM18&quot;      &quot;HEATR5B&quot;     &quot;MIR297&quot;      &quot;MIR129-2&quot;    &quot;TTC39A-AS1&quot; </code></pre>
<pre class="r"><code># Do enrchment tests for HF_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.5803
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 4.030772
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-1">
Past versions of "Enrichment GWAS HF TOX-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0              1135
Not DE             4              3016</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_SAME in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 7.927869
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-2">
Past versions of "Enrichment GWAS HF TOX-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               667
Not DE             4              3484</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_OPP in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 11.95174
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-3">
Past versions of "Enrichment GWAS HF TOX-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               468
Not DE             4              3683</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 10.71977
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-4">
Past versions of "Enrichment GWAS HF TOX-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               515
Not DE             4              3636</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 17.77345
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-5">
Past versions of "Enrichment GWAS HF TOX-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               327
Not DE             4              3824</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 32.13016
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-6">
Past versions of "Enrichment GWAS HF TOX-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               188
Not DE             4              3963</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 8.644748
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-7">
Past versions of "Enrichment GWAS HF TOX-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               620
Not DE             4              3531</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.6026
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.1474393 28.1200072
sample estimates:
odds ratio 
  2.036255 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20HF%20TOX-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-HF-TOX-8">
Past versions of "Enrichment GWAS HF TOX-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-HF-TOX-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS HF TOX-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2              1367
Not DE             2              2784</code></pre>
<pre class="r"><code># Enrichment test for Arr mapped genes that are detected protein in our set 
Arr_GWAS &lt;- closest_genes_list$Arrythmia_GWAS_snp_locations_bed_df$Gene %&gt;% unique()
Arr_GWAS</code></pre>
<pre><code>  [1] &quot;RNU6-1281P&quot;   &quot;PRRX1&quot;        &quot;MYOG&quot;         &quot;SNORA72&quot;      &quot;PERM1&quot;       
  [6] &quot;RNU6-605P&quot;    &quot;RPL23AP17&quot;    &quot;CASQ2&quot;        &quot;SPEN&quot;         &quot;KCND3-IT1&quot;   
 [11] &quot;&quot;             &quot;PMVK&quot;         &quot;HMGB3P10&quot;     &quot;CUL4A&quot;        &quot;FNTAP2&quot;      
 [16] &quot;SNORD36&quot;      &quot;KLF12&quot;        &quot;NFATC1&quot;       &quot;UBE2D3&quot;       &quot;HAND2-AS1&quot;   
 [21] &quot;EPHA3&quot;        &quot;RPL29P34&quot;     &quot;LINC01438&quot;    &quot;SCN10A&quot;       &quot;CAND2&quot;       
 [26] &quot;FRMD4B&quot;       &quot;NR3C2&quot;        &quot;RNU6-615P&quot;    &quot;RNU1-138P&quot;    &quot;XXYLT1&quot;      
 [31] &quot;HIKESHIP1&quot;    &quot;FGF5&quot;         &quot;LRIG1&quot;        &quot;MOB1B&quot;        &quot;RNA5SP155&quot;   
 [36] &quot;Metazoa_SRP&quot;  NA             &quot;RNU6-745P&quot;    &quot;Y_RNA&quot;        &quot;MYOCD-AS1&quot;   
 [41] &quot;CRK&quot;          &quot;RPL31P20&quot;     &quot;ARNT2&quot;        &quot;MIR4714&quot;      &quot;GSDMB&quot;       
 [46] &quot;RN7SL270P&quot;    &quot;FGF13-AS1&quot;    &quot;ASAH1&quot;        &quot;NME5&quot;         &quot;DINOL&quot;       
 [51] &quot;NAV2-AS3&quot;     &quot;ZMIZ1&quot;        &quot;MIR4687&quot;      &quot;PDCD4-AS1&quot;    &quot;EXT1&quot;        
 [56] &quot;SPON1&quot;        &quot;LINC01950&quot;    &quot;MIR151A&quot;      &quot;OR7E161P&quot;     &quot;XPO7&quot;        
 [61] &quot;RN7SL518P&quot;    &quot;EIF4BP2&quot;      &quot;KCNJ5&quot;        &quot;SYNPO2L-AS1&quot;  &quot;CYCSP55&quot;     
 [66] &quot;SSXP10&quot;       &quot;DEK&quot;          &quot;MTCYBP4&quot;      &quot;RN7SKP155&quot;    &quot;C1DP1&quot;       
 [71] &quot;REEP3&quot;        &quot;RNU6-523P&quot;    &quot;KIAA1755&quot;     &quot;RN7SL660P&quot;    &quot;ESR2&quot;        
 [76] &quot;MTCO1P2&quot;      &quot;MIR1-1&quot;       &quot;HBEGF&quot;        &quot;C10orf71-AS1&quot; &quot;LINC01629&quot;   
 [81] &quot;CEP68&quot;        &quot;USP34-DT&quot;     &quot;SPATS2L&quot;      &quot;CCDC134&quot;      &quot;KIF3C&quot;       
 [86] &quot;RPL23AP92&quot;    &quot;SNORA74&quot;      &quot;LINC01966&quot;    &quot;SDF2L1&quot;       &quot;CFL2&quot;        
 [91] &quot;MIR548F2&quot;     &quot;EYA4&quot;         &quot;LINC01426&quot;    &quot;CDK6&quot;         &quot;QSOX2&quot;       
 [96] &quot;METRN&quot;        &quot;COMETT&quot;       &quot;NDUFB10&quot;      &quot;NOS3&quot;         &quot;CUX2&quot;        
[101] &quot;APOLD1&quot;       &quot;CREB5&quot;        &quot;GET4&quot;         &quot;ASTN2&quot;        &quot;LRRC10&quot;      
[106] &quot;TBX5&quot;         &quot;NFYB&quot;         &quot;DGKB&quot;         &quot;NACA&quot;         &quot;FBRSL1&quot;      
[111] &quot;RNA5SP354&quot;    &quot;NCOR2&quot;        &quot;HIP1R&quot;        &quot;ITPR2-AS1&quot;   </code></pre>
<pre class="r"><code>intersect(DE_PRO, Arr_GWAS)</code></pre>
<pre><code>[1] &quot;NACA&quot;    &quot;NDUFB10&quot; &quot;LRRC10&quot; </code></pre>
<pre class="r"><code># Do enrchment tests for Arr_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.76071, df = 1, p-value = 0.3831


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-1">
Past versions of "Enrichment GWAS Arr-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 3              1132
Not DE            16              3004</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_SAME in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.2152e-27, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-2">
Past versions of "Enrichment GWAS Arr-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 3               664
Not DE            16              3472</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_OPP in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.4229, df = 1, p-value = 0.2329


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-3">
Past versions of "Enrichment GWAS Arr-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               468
Not DE            19              3668</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.0691e-27, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-4">
Past versions of "Enrichment GWAS Arr-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2               513
Not DE            17              3623</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.6062e-05, df = 1, p-value = 0.9968


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-5">
Past versions of "Enrichment GWAS Arr-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2               325
Not DE            17              3811</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.15835, df = 1, p-value = 0.6907


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-6">
Past versions of "Enrichment GWAS Arr-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               188
Not DE            19              3948</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in Arr_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.74242, df = 1, p-value = 0.3889


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-7">
Past versions of "Enrichment GWAS Arr-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               619
Not DE            18              3517</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.9016e-29, df = 1, p-value = 1


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20Arr-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-Arr-8">
Past versions of "Enrichment GWAS Arr-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-Arr-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS Arr-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 6              1363
Not DE            13              2773</code></pre>
<pre class="r"><code># Enrichment test for Arr mapped genes that are detected protein in our set 
Gluc_Fast_GWAS &lt;- closest_genes_list$Fast_gluc_GWAS_snp_locations_bed_df$Gene %&gt;% unique()
Gluc_Fast_GWAS</code></pre>
<pre><code> [1] &quot;&quot;          &quot;TOMM22P3&quot;  &quot;PDX1&quot;      &quot;RNU1-70P&quot;  &quot;MIR548AQ&quot;  &quot;TSEN15P1&quot; 
 [7] &quot;NFE2L1-DT&quot; &quot;MAEA&quot;      &quot;MIR4714&quot;   &quot;NPM1P47&quot;   &quot;NUFIP2&quot;    &quot;MIR1908&quot;  
[13] &quot;INS&quot;       &quot;C5orf67&quot;   &quot;TP53INP1&quot;  &quot;KCNQ1-AS1&quot; NA          &quot;MIR6848&quot;  
[19] &quot;MAPK8IP1&quot;  &quot;MADD-AS1&quot;  &quot;GAD2&quot;      &quot;NOX4P1&quot;    &quot;MTNR1B&quot;    &quot;CRY2&quot;     
[25] &quot;PCSK1&quot;     &quot;RPS15AP30&quot; &quot;FBLL1&quot;     &quot;RFX6&quot;      &quot;RN7SL826P&quot; &quot;PAM&quot;      
[31] &quot;PDE6C&quot;     &quot;RNA5SP224&quot; &quot;LNCNEF&quot;    &quot;FOXA2&quot;     &quot;BTBD7P2&quot;   &quot;FOXN3-AS2&quot;
[37] &quot;STARD10&quot;   &quot;ABCB11&quot;    &quot;G6PC2&quot;     &quot;C2orf16&quot;   &quot;EML6&quot;      &quot;GCKR&quot;     
[43] &quot;RPL36AP25&quot; &quot;RAPGEF4&quot;   &quot;FAM133EP&quot;  &quot;GTF3AP5&quot;   &quot;Y_RNA&quot;     &quot;YKT6&quot;     
[49] &quot;MIR7107&quot;   &quot;FAM234A&quot;   &quot;GLS2&quot;      &quot;FBRSL1&quot;    &quot;SRRM3&quot;     &quot;RMST&quot;     
[55] &quot;GLIS3-AS2&quot; &quot;ADCY9&quot;    </code></pre>
<pre class="r"><code>intersect(BACK_PRO, Gluc_Fast_GWAS)</code></pre>
<pre><code>[1] &quot;YKT6&quot;    &quot;ADCY9&quot;   &quot;PAM&quot;     &quot;NUFIP2&quot;  &quot;&quot;        &quot;FAM234A&quot;</code></pre>
<pre class="r"><code>intersect(DE_PRO, Gluc_Fast_GWAS)</code></pre>
<pre><code>[1] &quot;NUFIP2&quot;  &quot;FAM234A&quot;</code></pre>
<pre class="r"><code># Do enrchment tests for Arr_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.6674
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1202074 9.3012525
sample estimates:
odds ratio 
  1.330882 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-1">
Past versions of "Enrichment GWAS gluc fast-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2              1133
Not DE             4              3016</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_SAME in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.5982
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 4.444189
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-2">
Past versions of "Enrichment GWAS gluc fast-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               667
Not DE             6              3482</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_OPP in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.1398
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.3563615 27.6604957
sample estimates:
odds ratio 
  3.949197 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-3">
Past versions of "Enrichment GWAS gluc fast-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2               466
Not DE             4              3683</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.5482
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.02984684 12.67479279
sample estimates:
odds ratio 
  1.414258 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-4">
Past versions of "Enrichment GWAS gluc fast-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               514
Not DE             5              3635</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 9.967917
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-5">
Past versions of "Enrichment GWAS gluc fast-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               327
Not DE             6              3822</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2427
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.08911966 38.13726044
sample estimates:
odds ratio 
  4.234602 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-6">
Past versions of "Enrichment GWAS gluc fast-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               187
Not DE             5              3962</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.02407589 10.21712712
sample estimates:
odds ratio 
   1.14048 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-7">
Past versions of "Enrichment GWAS gluc fast-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               619
Not DE             5              3530</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =Gluc_Fast_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in Gluc_Fast_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.1869
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 1.727434
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20gluc%20fast-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-gluc-fast-8">
Past versions of "Enrichment GWAS gluc fast-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-gluc-fast-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS gluc fast-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0              1369
Not DE             6              2780</code></pre>
<pre class="r"><code># Enrichment test for Arr mapped genes that are detected protein in our set 
Fast_ins_GWAS &lt;- closest_genes_list$Fast_ins_GWAS_snp_locations_bed_df$Gene %&gt;% unique()
Fast_ins_GWAS</code></pre>
<pre><code> [1] &quot;TET2-AS1&quot;    NA            &quot;PEPD&quot;        &quot;PRELID1P3&quot;   &quot;&quot;           
 [6] &quot;RPS15AP29&quot;   &quot;PPARG&quot;       &quot;NUDT3&quot;       &quot;UBE2L2&quot;      &quot;LINC00208&quot;  
[11] &quot;GCKR&quot;        &quot;KLF14&quot;       &quot;ITIH4&quot;       &quot;GAPDHP54&quot;    &quot;SAE1&quot;       
[16] &quot;RAB1A&quot;       &quot;RSPO3&quot;       &quot;GTSF1L&quot;      &quot;MAP3K10&quot;     &quot;PDE2A-AS1&quot;  
[21] &quot;GLIS3-AS2&quot;   &quot;C5orf67&quot;     &quot;MAP2K7&quot;      &quot;MIR581&quot;      &quot;MIR4641&quot;    
[26] &quot;LINC01625&quot;   &quot;LYPLAL1-AS1&quot; &quot;PDGFC&quot;       &quot;RNU6-1151P&quot;  &quot;LINC01153&quot;  
[31] &quot;RPS15AP30&quot;   &quot;ABO&quot;         &quot;Y_RNA&quot;       &quot;MIR8073&quot;     &quot;LINC01713&quot;  
[36] &quot;CAMKV&quot;      </code></pre>
<pre class="r"><code>intersect(BACK_PRO, Fast_ins_GWAS)</code></pre>
<pre><code>[1] &quot;PEPD&quot;  &quot;RAB1A&quot; &quot;&quot;      &quot;SAE1&quot; </code></pre>
<pre class="r"><code>intersect(DE_PRO, Fast_ins_GWAS)</code></pre>
<pre><code>[1] &quot;PEPD&quot;</code></pre>
<pre class="r"><code># Do enrchment tests for Arr_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.0168826 11.0587150
sample estimates:
odds ratio 
  0.886858 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-1">
Past versions of "Enrichment GWAS ins fast-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1              1134
Not DE             3              3017</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_SAME in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.5035
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.03317608 21.75649279
sample estimates:
odds ratio 
  1.743962 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-2">
Past versions of "Enrichment GWAS ins fast-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               666
Not DE             3              3485</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_OPP in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 11.95174
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-3">
Past versions of "Enrichment GWAS ins fast-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               468
Not DE             4              3683</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 10.71977
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-4">
Past versions of "Enrichment GWAS ins fast-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               515
Not DE             4              3636</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 17.77345
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-5">
Past versions of "Enrichment GWAS ins fast-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               327
Not DE             4              3824</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.00000 32.13016
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-6">
Past versions of "Enrichment GWAS ins fast-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               188
Not DE             4              3963</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.4762
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.03617243 23.71494698
sample estimates:
odds ratio 
  1.901549 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-7">
Past versions of "Enrichment GWAS ins fast-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               619
Not DE             3              3532</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =Fast_ins_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in Fast_ins_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.6026
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.1474393 28.1200072
sample estimates:
odds ratio 
  2.036255 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20GWAS%20ins%20fast-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-GWAS-ins-fast-8">
Past versions of "Enrichment GWAS ins fast-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-GWAS-ins-fast-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment GWAS ins fast-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2              1367
Not DE             2              2784</code></pre>
<pre class="r"><code>HF_GWAS &lt;- readRDS(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/HF_geneset.RDS&quot;)
HF_GWAS &lt;- HF_GWAS$hgnc_symbol %&gt;% unique()
HF_GWAS</code></pre>
<pre><code> [1] &quot;ALDH2&quot;    &quot;ASXL3&quot;    &quot;BACH1&quot;    &quot;BAG3&quot;     &quot;CACNB2&quot;   &quot;CDKN1A&quot;  
 [7] &quot;CDKN2B&quot;   &quot;CHMP3&quot;    &quot;COX7C&quot;    &quot;DMRTA2&quot;   &quot;FTO&quot;      &quot;GNPDA2&quot;  
[13] &quot;GTF2I&quot;    &quot;HECTD4&quot;   &quot;HLA-DQA1&quot; &quot;HSD17B12&quot; &quot;KCNB1&quot;    &quot;KCNIP4&quot;  
[19] &quot;LPA&quot;      &quot;NEURL1&quot;   &quot;NFIA&quot;     &quot;NPC1&quot;     &quot;OR2A2&quot;    &quot;ORC5&quot;    
[25] &quot;PITX2&quot;    &quot;POM121C&quot;  &quot;PSRC1&quot;    &quot;PTPN11&quot;   &quot;SH2B3&quot;    &quot;SPATS2L&quot; 
[31] &quot;SPEN&quot;     &quot;SRR&quot;      &quot;SURF6&quot;    &quot;SYNPO2L&quot;  &quot;TTC39A&quot;   &quot;TUBA3C&quot;  
[37] &quot;USP36&quot;    &quot;YWHAE&quot;    &quot;ZFHX3&quot;    &quot;ZNF280A&quot; </code></pre>
<pre class="r"><code>intersect(BACK_PRO, HF_GWAS)</code></pre>
<pre><code> [1] &quot;POM121C&quot;  &quot;NPC1&quot;     &quot;SURF6&quot;    &quot;BAG3&quot;     &quot;ALDH2&quot;    &quot;YWHAE&quot;   
 [7] &quot;GTF2I&quot;    &quot;PTPN11&quot;   &quot;CACNB2&quot;   &quot;HSD17B12&quot; &quot;FTO&quot;      &quot;SRR&quot;     
[13] &quot;SYNPO2L&quot;  &quot;SPATS2L&quot; </code></pre>
<pre class="r"><code>intersect(DE_PRO, HF_GWAS)</code></pre>
<pre><code>[1] &quot;GTF2I&quot;  &quot;PTPN11&quot;</code></pre>
<pre class="r"><code># Do enrchment tests for HF_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.3755
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.04803599 1.99249797
sample estimates:
odds ratio 
 0.4425468 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-1">
Past versions of "Enrichment HF GWAS author reported-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2              1133
Not DE            12              3008</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_SAME in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.09449034 3.92697505
sample estimates:
odds ratio 
 0.8712093 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-2">
Past versions of "Enrichment HF GWAS author reported-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2               665
Not DE            12              3476</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_OPP in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.3905
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 2.375241
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-3">
Past versions of "Enrichment HF GWAS author reported-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               468
Not DE            14              3673</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2412
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 2.130336
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-4">
Past versions of "Enrichment HF GWAS author reported-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               515
Not DE            14              3626</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.6198
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 3.534825
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-5">
Past versions of "Enrichment HF GWAS author reported-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               327
Not DE            14              3814</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.000000 6.394922
sample estimates:
odds ratio 
         0 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-6">
Past versions of "Enrichment HF GWAS author reported-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 0               188
Not DE            14              3953</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in HF_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1030354 4.2839626
sample estimates:
odds ratio 
 0.9501115 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-7">
Past versions of "Enrichment HF GWAS author reported-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-7"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 2               618
Not DE            12              3523</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =HF_GWAS, test_type = &quot;chisq&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in HF_GWAS&quot;)</code></pre>
<pre><code>Warning in chisq.test(mat): Chi-squared approximation may be incorrect</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.7043, df = 1, p-value = 0.1001


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20HF%20GWAS%20author%20reported-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-HF-GWAS-author-reported-8">
Past versions of "Enrichment HF GWAS author reported-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-HF-GWAS-author-reported-8"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment HF GWAS author reported-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 8              1361
Not DE             6              2780</code></pre>
<pre class="r"><code>Arr_GWAS &lt;- readRDS(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/Arr_geneset.RDS&quot;)
Arr_GWAS &lt;- Arr_GWAS$hgnc_symbol %&gt;% unique()
Arr_GWAS</code></pre>
<pre><code>  [1] &quot;AGAP5&quot;           &quot;AGBL4&quot;           &quot;AKAP6&quot;           &quot;ANXA4&quot;          
  [5] &quot;ARHGAP10&quot;        &quot;ARNT2&quot;           &quot;ASAH1&quot;           &quot;ATXN1&quot;          
  [9] &quot;CALU&quot;            &quot;CAMK2D&quot;          &quot;CAND2&quot;           &quot;CASQ2&quot;          
 [13] &quot;CASZ1&quot;           &quot;CAV1&quot;            &quot;CAV2&quot;            &quot;CDK6&quot;           
 [17] &quot;CDKN1A&quot;          &quot;CEP68&quot;           &quot;CFL2&quot;            &quot;CGA&quot;            
 [21] &quot;CREB5&quot;           &quot;CRK&quot;             &quot;CUL4A&quot;           &quot;CYTH1&quot;          
 [25] &quot;DEK&quot;             &quot;DGKB&quot;            &quot;DPF3&quot;            &quot;EPHA3&quot;          
 [29] &quot;ERBB4&quot;           &quot;ESR2&quot;            &quot;FAM13B&quot;          &quot;FBRSL1&quot;         
 [33] &quot;FBXO32&quot;          &quot;FGF5&quot;            &quot;FKBP7&quot;           &quot;FRMD4B&quot;         
 [37] &quot;FXR2&quot;            &quot;GATA4&quot;           &quot;GJA5&quot;            &quot;GMCL1&quot;          
 [41] &quot;GNB4&quot;            &quot;GSDMB&quot;           &quot;GTF2I&quot;           &quot;GTF2IRD2&quot;       
 [45] &quot;GYPC&quot;            &quot;HAND2&quot;           &quot;HAND2-AS1&quot;       &quot;HCN4&quot;           
 [49] &quot;HERC1&quot;           &quot;HIP1R&quot;           &quot;HSF2&quot;            &quot;HSPG2&quot;          
 [53] &quot;IGF1R&quot;           &quot;IRF2BPL&quot;         &quot;KCND3&quot;           &quot;KCNH2&quot;          
 [57] &quot;KCNJ5&quot;           &quot;KCNN2&quot;           &quot;KCNN3&quot;           &quot;KDM1B&quot;          
 [61] &quot;KIF3C&quot;           &quot;LHX3&quot;            &quot;LINC00477&quot;       &quot;LINC00621&quot;      
 [65] &quot;LINC01142&quot;       &quot;LINC01426&quot;       &quot;LRIG1&quot;           &quot;LRRC10&quot;         
 [69] &quot;MEX3C&quot;           &quot;MIR548AZ&quot;        &quot;MIR548N&quot;         &quot;MIR6500&quot;        
 [73] &quot;MTHFD1&quot;          &quot;MYH6&quot;            &quot;MYH7&quot;            &quot;MYO18B&quot;         
 [77] &quot;MYO1C&quot;           &quot;MYOCD&quot;           &quot;MYOT&quot;            &quot;MYOZ1&quot;          
 [81] &quot;MYPN&quot;            &quot;NACA&quot;            &quot;NAV2&quot;            &quot;NEURL1&quot;         
 [85] &quot;NKX2-5&quot;          &quot;NPY6R&quot;           &quot;NR3C1&quot;           &quot;NRBF2&quot;          
 [89] &quot;NUCKS1&quot;          &quot;NUDT13&quot;          &quot;OPN1SW&quot;          &quot;ORMDL3&quot;         
 [93] &quot;PANDAR&quot;          &quot;PHLDA1&quot;          &quot;PHLDB2&quot;          &quot;PI16&quot;           
 [97] &quot;PITX2&quot;           &quot;PKP2&quot;            &quot;PLCXD2&quot;          &quot;PLN&quot;            
[101] &quot;PPFIA4&quot;          &quot;PPP2R3A&quot;         &quot;PTK2&quot;            &quot;RBM20&quot;          
[105] &quot;REEP1&quot;           &quot;REEP3&quot;           &quot;RPL3L&quot;           &quot;SCMH1&quot;          
[109] &quot;SCN10A&quot;          &quot;SCN5A&quot;           &quot;SGCG&quot;            &quot;SIRT1&quot;          
[113] &quot;SLC25A26&quot;        &quot;SLC27A6&quot;         &quot;SLC35F1&quot;         &quot;SLC41A1&quot;        
[117] &quot;SLC9B1&quot;          &quot;SLIT3&quot;           &quot;SMAD7&quot;           &quot;SORL1&quot;          
[121] &quot;SOX15&quot;           &quot;SPATS2L&quot;         &quot;SSPN&quot;            &quot;SYNE2&quot;          
[125] &quot;SYNPO2L&quot;         &quot;TBX5&quot;            &quot;TEX41&quot;           &quot;THRB&quot;           
[129] &quot;TNFSF12&quot;         &quot;TNFSF12-TNFSF13&quot; &quot;TTN&quot;             &quot;TTN-AS1&quot;        
[133] &quot;TUBA8&quot;           &quot;USP34&quot;           &quot;USP36&quot;           &quot;UST&quot;            
[137] &quot;WIPF1&quot;           &quot;WNT3&quot;            &quot;WNT8A&quot;           &quot;XPO7&quot;           
[141] &quot;XXYLT1&quot;          &quot;YWHAE&quot;           &quot;ZFHX3&quot;           &quot;ZPBP2&quot;          </code></pre>
<pre class="r"><code>intersect(BACK_PRO, Arr_GWAS)</code></pre>
<pre><code> [1] &quot;ARHGAP10&quot; &quot;NACA&quot;     &quot;MYO1C&quot;    &quot;CASQ2&quot;    &quot;CALU&quot;     &quot;SLIT3&quot;   
 [7] &quot;CAND2&quot;    &quot;PPFIA4&quot;   &quot;GYPC&quot;     &quot;IGF1R&quot;    &quot;ANXA4&quot;    &quot;MTHFD1&quot;  
[13] &quot;MYH7&quot;     &quot;MYH6&quot;     &quot;PLN&quot;      &quot;DEK&quot;      &quot;GATA4&quot;    &quot;FXR2&quot;    
[19] &quot;CAV2&quot;     &quot;YWHAE&quot;    &quot;GTF2I&quot;    &quot;HSPG2&quot;    &quot;CAV1&quot;     &quot;ASAH1&quot;   
[25] &quot;CAMK2D&quot;   &quot;CUL4A&quot;    &quot;SCN5A&quot;    &quot;LRRC10&quot;   &quot;RBM20&quot;    &quot;REEP3&quot;   
[31] &quot;SLC25A26&quot; &quot;PHLDB2&quot;   &quot;MYPN&quot;     &quot;CASZ1&quot;    &quot;MYO18B&quot;   &quot;XXYLT1&quot;  
[37] &quot;SYNE2&quot;    &quot;TTN&quot;      &quot;DPF3&quot;     &quot;PKP2&quot;     &quot;IRF2BPL&quot;  &quot;NUCKS1&quot;  
[43] &quot;REEP1&quot;    &quot;SYNPO2L&quot;  &quot;GNB4&quot;     &quot;SPATS2L&quot;  &quot;XPO7&quot;     &quot;CFL2&quot;    
[49] &quot;SLC27A6&quot;  &quot;HCN4&quot;     &quot;FKBP7&quot;   </code></pre>
<pre class="r"><code>intersect(DE_PRO, Arr_GWAS)</code></pre>
<pre><code> [1] &quot;NACA&quot;     &quot;SLIT3&quot;    &quot;PLN&quot;      &quot;GATA4&quot;    &quot;FXR2&quot;     &quot;GTF2I&quot;   
 [7] &quot;CAV1&quot;     &quot;LRRC10&quot;   &quot;RBM20&quot;    &quot;SLC25A26&quot; &quot;CASZ1&quot;    &quot;DPF3&quot;    
[13] &quot;IRF2BPL&quot;  &quot;NUCKS1&quot;  </code></pre>
<pre class="r"><code># Do enrchment tests for Arr_GWAS GWAS 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.5008334 1.9155540
sample estimates:
odds ratio 
   1.00687 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-1">
Past versions of "Enrichment Arr GWAS author reported-1.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-1.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                14              1121
Not DE            37              2983</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_SAME, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_SAME in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2549
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.6652355 2.8886165
sample estimates:
odds ratio 
   1.44528 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-2">
Past versions of "Enrichment Arr GWAS author reported-2.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-2.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                11               656
Not DE            40              3448</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_OPP, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_OPP in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.2708
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.09707574 1.52744142
sample estimates:
odds ratio 
 0.4891777 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-3">
Past versions of "Enrichment Arr GWAS author reported-3.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-3.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 3               465
Not DE            48              3639</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.3266298 2.2296633
sample estimates:
odds ratio 
 0.9417222 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-4">
Past versions of "Enrichment Arr GWAS author reported-4.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-4.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 6               509
Not DE            45              3595</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_SAME, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO_DE_RNA_SAME in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.5962
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.3930936 3.2323745
sample estimates:
odds ratio 
  1.276583 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-5">
Past versions of "Enrichment Arr GWAS author reported-5.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-5.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 5               322
Not DE            46              3782</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_DE_RNA_OPP, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_DE_RNA_OPP in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.7286
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.0103493 2.4735772
sample estimates:
odds ratio 
 0.4189911 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-6">
Past versions of "Enrichment Arr GWAS author reported-6.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-6.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 1               187
Not DE            50              3917</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_PRO_ONLY, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_PRO_ONLY in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.8434
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.4289302 2.2984975
sample estimates:
odds ratio 
  1.061557 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-7">
Past versions of "Enrichment Arr GWAS author reported-7.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-7"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-7.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                 8               612
Not DE            43              3492</code></pre>
<pre class="r"><code>enrichment_test(Test_set = DE_RNA_ONLY, Background_exp = BACK_PRO, Test_set_2 =Arr_GWAS, test_type = &quot;fisher&quot;, plot = TRUE, custom_title =&quot;DE_RNA_ONLY in Arr_GWAS&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.1339
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.8465428 2.8098371
sample estimates:
odds ratio 
  1.552581 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment%20Arr%20GWAS%20author%20reported-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Enrichment-Arr-GWAS-author-reported-8">
Past versions of "Enrichment Arr GWAS author reported-8.png"
</button>
</p>
<div id="fig-no-spaces-Enrichment-Arr-GWAS-author-reported-8"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/663fe995b8bb06fa0b3ec3bfb609b9e2e6573ccd/docs/figure/Sign_and_genetic_summary_stat_testing.Rmd/Enrichment Arr GWAS author reported-8.png" target="_blank">663fe99</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2024-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Mat
       In Background Not in Background
DE                22              1347
Not DE            29              2757</code></pre>
<div id="de-protein-enrichment-across-different-protein-attributes"
class="section level1">
<h1>DE protein enrichment across different protein attributes</h1>
<pre class="r"><code>#### Druggable Protein #### 

Drugabble &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/DIA_Framework/Druggable_proteins.csv&quot;, header = TRUE)
Drugabble &lt;- Drugabble[,1:5]
Drugabble %&gt;% dim()</code></pre>
<pre><code>[1] 186   5</code></pre>
<pre class="r"><code>DE_PRO &lt;- toptable_summary_DoxvNorm[toptable_summary_DoxvNorm$P.Value &lt; 0.05, ] %&gt;% rownames()
BACK_PRO &lt;- toptable_summary_DoxvNorm %&gt;% rownames()

# Druggable protein
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Drugabble$Uniprot, test_type = &quot;fisher&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Druggable prteins&quot;)</code></pre>
<pre><code>$Test_Result

    Fisher&#39;s Exact Test for Count Data

data:  mat
p-value = 0.6746
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.7619581 1.4999066
sample estimates:
odds ratio 
  1.075201 


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                54              1113
Not DE           130              2881</code></pre>
<pre class="r"><code>#### Transcription Factors #### 
TFs &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/DIA_Framework/Transcription_factors.csv&quot;, header = TRUE)
TFs &lt;- TFs[,1:5]
TFs %&gt;% head()</code></pre>
<pre><code>    Gene                                              Gene.synonym
1   ADNP                                           ADNP1, KIAA0784
2 ARID1A      B120, BAF250, BAF250a, C10rf4, C1orf4, P270, SMARCF1
3 ARID1B 6A3-5, BAF250b, DAN15, ELD/OSA1, KIAA1235, p250R, SMARCF2
4  ARID2         BAF200, DKFZp686G052, FLJ30619, KIAA1557, SMARCF3
5 ARID3A                                             BRIGHT, DRIL1
6   ATRX                               JMS, MRX52, RAD54, XH2, XNP
          Ensembl                           Gene.description Uniprot
1 ENSG00000101126 Activity dependent neuroprotector homeobox  Q9H2P0
2 ENSG00000117713              AT-rich interaction domain 1A  O14497
3 ENSG00000049618              AT-rich interaction domain 1B  Q8NFD5
4 ENSG00000189079               AT-rich interaction domain 2  Q68CP9
5 ENSG00000116017              AT-rich interaction domain 3A  Q99856
6 ENSG00000085224                   ATRX chromatin remodeler  P46100</code></pre>
<pre class="r"><code>TFs %&gt;% dim()</code></pre>
<pre><code>[1] 75  5</code></pre>
<pre class="r"><code># Test for enrichment
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = TFs$Uniprot, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in transcription factors&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 13.072, df = 1, p-value = 0.0002997


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                35              1132
Not DE            39              2972</code></pre>
<pre class="r"><code>#### Signaling &amp; Transmembrane Peptides #### 
SP_TM &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/DIA_Framework/Signaling_Transmembrane_peptides.csv&quot;, header = TRUE)

SP_TM %&gt;% colnames()</code></pre>
<pre><code>[1] &quot;Ensembl&quot;                                        
[2] &quot;Gene&quot;                                           
[3] &quot;Gene.description&quot;                               
[4] &quot;Uniprot&quot;                                        
[5] &quot;SP..number.of.positive.prediction.methods.&quot;     
[6] &quot;TM..number.of.predicted.transmembrane.segments.&quot;</code></pre>
<pre class="r"><code>SP_TM %&gt;% dim()</code></pre>
<pre><code>[1] 4136    6</code></pre>
<pre class="r"><code>SP_TM %&gt;% head()</code></pre>
<pre><code>          Ensembl    Gene
1 ENSG00000135387 CAPRIN1
2 ENSG00000122692    SMU1
3 ENSG00000137801   THBS1
4 ENSG00000101126    ADNP
5 ENSG00000136383   ALPK3
6 ENSG00000148677  ANKRD1
                                        Gene.description Uniprot
1                        Cell cycle associated protein 1  Q14444
2 SMU1 DNA replication regulator and spliceosomal factor  Q2TAY7
3                                       Thrombospondin 1  P07996
4             Activity dependent neuroprotector homeobox  Q9H2P0
5                                         Alpha kinase 3  Q96L96
6                                Ankyrin repeat domain 1  Q15327
  SP..number.of.positive.prediction.methods.
1                                          0
2                                          0
3                                          3
4                                          0
5                                          0
6                                          0
  TM..number.of.predicted.transmembrane.segments.
1                                               0
2                                               0
3                                               0
4                                               0
5                                               0
6                                               0</code></pre>
<pre class="r"><code>SP &lt;- SP_TM[SP_TM$SP..number.of.positive.prediction.methods. &gt; 0, ]$Uniprot
TM &lt;- SP_TM[SP_TM$TM..number.of.predicted.transmembrane.segments. &gt; 0, ]$Uniprot


# Test for enrichment of Signaling Peptides 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = SP, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in signaling peptides&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 1.469, df = 1, p-value = 0.2255


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                95              1072
Not DE           283              2728</code></pre>
<pre class="r"><code># Test for enrichment of Transmembrane Peptides 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = TM, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in transmembrane peptides&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 18.907, df = 1, p-value = 1.372e-05


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE               212               955
Not DE           738              2273</code></pre>
<pre class="r"><code>Non_DE_TM &lt;- intersect(toptable_summary_DoxvNorm[toptable_summary_DoxvNorm$P.Value &gt; 0.05, ] %&gt;% rownames(), 
        TM  )


DE_TM &lt;- intersect(toptable_summary_DoxvNorm[toptable_summary_DoxvNorm$P.Value &lt; 0.05, ] %&gt;% rownames(), 
        TM  )


#### Tissue specificity #### 
Tissue_spec &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/DIA_Framework/Tissue_specific.csv&quot;, header = TRUE)

Tissue_spec %&gt;% head()</code></pre>
<pre><code>     Gene                    Gene.synonym         Ensembl
1  ANKRD1 ALRP, C-193, CARP, CVARP, MCARP ENSG00000148677
2   CASQ2                           PDIB2 ENSG00000118729
3 CCDC141                 CAMDI, FLJ39502 ENSG00000163492
4   CORIN ATC2, CRN, Lrp4, PRSC, TMPRSS10 ENSG00000145244
5   CRIP2                      CRP2, ESP1 ENSG00000182809
6   FITM2     C20orf142, dJ881L22.2, FIT2 ENSG00000197296
                              Gene.description Uniprot       HPA_Class
1                      Ankyrin repeat domain 1  Q15327 Tissue_enriched
2                              Calsequestrin 2  O14958 Tissue_enriched
3            Coiled-coil domain containing 141  Q6ZP82 Tissue_enriched
4                      Corin, serine peptidase  Q9Y5Q5 Tissue_enriched
5                      Cysteine rich protein 2  P52943 Tissue_enriched
6 Fat storage inducing transmembrane protein 2  Q8N6M3 Tissue_enriched
  Specificity
1           1
2           1
3           1
4           1
5           1
6           1</code></pre>
<pre class="r"><code>Tissue_spec %&gt;% dim() # Specificity ranges 0-4, where 0 is most specific and 4 is most specific</code></pre>
<pre><code>[1] 277   7</code></pre>
<pre class="r"><code>Tissue_spec$HPA_Class %&gt;% unique()</code></pre>
<pre><code>[1] &quot;Tissue_enriched&quot; &quot;Group_enriched&quot;  &quot;Tissue_enhanced&quot; &quot;Heart_elevated &quot;</code></pre>
<pre class="r"><code>Tissue_spec$Specificity %&gt;% unique()# Specificity ranges 0-4, where 0 is most specific and 4 is most specific, a way I notated the classes for easier subsetting. </code></pre>
<pre><code>[1] 1 2 3 0</code></pre>
<pre class="r"><code>Tissue_spec_0 &lt;- Tissue_spec[Tissue_spec$Specificity == 0, ]$Uniprot
Tissue_spec_1 &lt;- Tissue_spec[Tissue_spec$Specificity &lt;= 1 , ]$Uniprot
Tissue_spec_11 &lt;- Tissue_spec[Tissue_spec$Specificity == 1 , ]$Uniprot
Tissue_spec_2 &lt;- Tissue_spec[Tissue_spec$Specificity &lt;= 2 , ]$Uniprot
Tissue_spec_3 &lt;- Tissue_spec[Tissue_spec$Specificity &lt;= 3 , ]$Uniprot


# Test for enrichment of Tissue_spec_0 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Tissue_spec_0, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Tissue_spec_0&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 2.2798, df = 1, p-value = 0.1311


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                20              1147
Not DE            77              2934</code></pre>
<pre class="r"><code># Test for enrichment of Tissue_spec_1 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Tissue_spec_1, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Tissue_spec_1&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 3.2816, df = 1, p-value = 0.07006


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                24              1143
Not DE            95              2916</code></pre>
<pre class="r"><code># Test for enrichment of Tissue_spec_11 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Tissue_spec_11, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Tissue_spec_11&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 0.6143, df = 1, p-value = 0.4332


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                 4              1163
Not DE            18              2993</code></pre>
<pre class="r"><code># Test for enrichment of Tissue_spec_2 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Tissue_spec_2, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Tissue_spec_2&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 7.1795, df = 1, p-value = 0.007374


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                34              1133
Not DE           146              2865</code></pre>
<pre class="r"><code># Test for enrichment of Tissue_spec_3 
enrichment_test(Test_set = DE_PRO, Background_exp = BACK_PRO, Test_set_2 = Tissue_spec_3, test_type = &quot;chisq&quot;, plot = TRUE, custom_title = &quot;DE_PRO in Tissue_spec_3&quot;)</code></pre>
<pre><code>$Test_Result

    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  mat
X-squared = 7.1795, df = 1, p-value = 0.007374


$Plot</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$Mat
       In Background Not in Background
DE                34              1133
Not DE           146              2865</code></pre>
<pre class="r"><code># DE proteins are not really specific to heart tissue, and are generally depleted. 
intersect(DE_PRO,Tissue_spec_0 )</code></pre>
<pre><code> [1] &quot;P43694&quot; &quot;Q9BPW9&quot; &quot;O14531&quot; &quot;Q9Y2D4&quot; &quot;Q2V2M9&quot; &quot;Q5T481&quot; &quot;Q13061&quot; &quot;Q8TDX7&quot;
 [9] &quot;O95167&quot; &quot;Q92784&quot; &quot;Q9NPC6&quot; &quot;P50461&quot; &quot;P06396&quot; &quot;Q96AN5&quot; &quot;Q9BZQ8&quot; &quot;P48681&quot;
[17] &quot;P54646&quot; &quot;P03905&quot; &quot;Q5SVS4&quot; &quot;P51970&quot;</code></pre>
<pre class="r"><code>Tox_set_1 &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$uniprotswissprot %in% intersect(DE_PRO,Tissue_spec_0 ), ]$hgnc_symbol


# Possible targets of AC induced toxicity: GATA4, NES, CSRP3, NDUFA8, TRDN, FHOD3, TMEM143, DHRS9, MYOZ2

Tox_set_2 &lt;- New_RNA_PRO_DF[New_RNA_PRO_DF$uniprotswissprot %in% intersect(DE_PRO,Tissue_spec_1 ), ]$hgnc_symbol

setdiff(Tox_set_2, Tox_set_1)</code></pre>
<pre><code>[1] &quot;MYL4&quot;   &quot;ANKRD1&quot; &quot;LRRC10&quot; &quot;TECRL&quot; </code></pre>
<pre class="r"><code># MYL4, ANKRD1, LRRC10, TECRL
intersect(pQTL_data$Gene.symbol, c(&quot;MYL4&quot;, &quot;ANKRD1&quot;, &quot;LRRC10&quot;, &quot;TECRL&quot;) )</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>#### PPI numbers #### 
PPIs &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/DIA_Framework/PPI_number.csv&quot;, header = TRUE)
PPIs %&gt;% head()</code></pre>
<pre><code>   Gene                Gene.synonym         Ensembl
1  AAAS                             ENSG00000094914
2  AAK1    DKFZp686K16132, KIAA1048 ENSG00000115977
3 AARS1          AARS, AlaRS, CMT2N ENSG00000090861
4 AARS2 AARSL, bA444E17.1, KIAA1270 ENSG00000124608
5  AASS              LKRSDH, LORSDH ENSG00000008311
6  ABAT               GABA-T, GABAT ENSG00000183044
                         Gene.description Uniprot Interactions
1            Aladin WD repeat nucleoporin  Q9NRG9            0
2                 AP2 associated kinase 1  Q2M2I8            2
3                Alanyl-tRNA synthetase 1  P49588            0
4 Alanyl-tRNA synthetase 2, mitochondrial  Q5JTZ9            1
5      Aminoadipate-semialdehyde synthase  Q9UDR5            0
6        4-aminobutyrate aminotransferase  P80404            0</code></pre>
<pre class="r"><code>PPIs %&gt;% dim()</code></pre>
<pre><code>[1] 4172    6</code></pre>
<pre class="r"><code># Get PPI numbered vector for Diff and non-Diff abundant proteins. 
PPI_DE &lt;- PPIs[PPIs$Uniprot%in% DE_PRO, ]$Interactions
PPI_Not_DE &lt;- PPIs[PPIs$Uniprot%in% toptable_summary_DoxvNorm[toptable_summary_DoxvNorm$P.Value &gt;0.05,]$Protein, ]$Interactions


# Wilcoxon rank sum test
PPI_test_res &lt;- wilcox.test(PPI_DE, PPI_Not_DE)
PPI_test_res</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  PPI_DE and PPI_Not_DE
W = 1792807, p-value = 0.05468
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>combined_df_PPI &lt;- rbind(
  data.frame(PPIs = PPI_DE, Group = &quot;DE&quot;),
  data.frame(PPIs = PPI_Not_DE , Group = &quot;Non-DE&quot;)
)



# Create a boxplot comparing the UTR_Length between the two groups
boxplot(PPIs ~ Group, data = combined_df_PPI, main = &quot;Number of PPIs&quot;, ylab = &quot;PPI #&quot;, xlab = &quot;&quot;, ylim = c(0,25))</code></pre>
<p><img src="figure/Sign_and_genetic_summary_stat_testing.Rmd/unnamed-chunk-1-10.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] impute_1.70.0                           
 [2] WGCNA_1.72-1                            
 [3] fastcluster_1.2.3                       
 [4] dynamicTreeCut_1.63-1                   
 [5] BioNERO_1.4.2                           
 [6] reshape2_1.4.4                          
 [7] ggridges_0.5.4                          
 [8] biomaRt_2.52.0                          
 [9] ggvenn_0.1.10                           
[10] UpSetR_1.4.0                            
[11] DOSE_3.22.1                             
[12] variancePartition_1.26.0                
[13] clusterProfiler_4.4.4                   
[14] pheatmap_1.0.12                         
[15] qvalue_2.28.0                           
[16] cluster_2.1.4                           
[17] ggfortify_0.4.16                        
[18] lubridate_1.9.2                         
[19] forcats_1.0.0                           
[20] stringr_1.5.0                           
[21] dplyr_1.1.2                             
[22] purrr_1.0.2                             
[23] readr_2.1.4                             
[24] tidyr_1.3.0                             
[25] tibble_3.2.1                            
[26] ggplot2_3.4.3                           
[27] tidyverse_2.0.0                         
[28] RColorBrewer_1.1-3                      
[29] RUVSeq_1.30.0                           
[30] edgeR_3.38.4                            
[31] limma_3.52.4                            
[32] EDASeq_2.30.0                           
[33] ShortRead_1.54.0                        
[34] GenomicAlignments_1.32.1                
[35] SummarizedExperiment_1.26.1             
[36] MatrixGenerics_1.8.1                    
[37] matrixStats_1.0.0                       
[38] Rsamtools_2.12.0                        
[39] Biostrings_2.64.1                       
[40] XVector_0.36.0                          
[41] BiocParallel_1.30.4                     
[42] Homo.sapiens_1.3.1                      
[43] org.Hs.eg.db_3.15.0                     
[44] GO.db_3.15.0                            
[45] OrganismDbi_1.38.1                      
[46] TxDb.Hsapiens.UCSC.hg38.knownGene_3.15.0
[47] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 
[48] GenomicFeatures_1.48.4                  
[49] AnnotationDbi_1.58.0                    
[50] Biobase_2.56.0                          
[51] GenomicRanges_1.48.0                    
[52] GenomeInfoDb_1.32.4                     
[53] IRanges_2.30.1                          
[54] S4Vectors_0.34.0                        
[55] BiocGenerics_0.42.0                     
[56] workflowr_1.7.1                         

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           htmlwidgets_1.6.2     
 [40] igraph_1.5.1           DBI_1.1.3              ggnewscale_0.4.9      
 [43] backports_1.4.1        annotate_1.74.0        aod_1.3.2             
 [46] deldir_1.0-9           vctrs_0.6.3            abind_1.4-5           
 [49] cachem_1.0.8           withr_2.5.0            ggforce_0.4.1         
 [52] vroom_1.6.3            checkmate_2.2.0        treeio_1.20.2         
 [55] prettyunits_1.1.1      ape_5.7-1              lazyeval_0.2.2        
 [58] crayon_1.5.2           genefilter_1.78.0      labeling_0.4.2        
 [61] pkgconfig_2.0.3        tweenr_2.0.2           nlme_3.1-163          
 [64] nnet_7.3-19            rlang_1.1.1            lifecycle_1.0.3       
 [67] downloader_0.4         filelock_1.0.2         BiocFileCache_2.4.0   
 [70] rprojroot_2.0.3        polyclip_1.10-4        graph_1.74.0          
 [73] Matrix_1.5-4.1         aplot_0.2.0            NetRep_1.2.7          
 [76] boot_1.3-28.1          base64enc_0.1-3        GlobalOptions_0.1.2   
 [79] whisker_0.4.1          processx_3.8.2         png_0.1-8             
 [82] viridisLite_0.4.2      rjson_0.2.21           bitops_1.0-7          
 [85] getPass_0.2-2          R.oo_1.25.0            ggnetwork_0.5.12      
 [88] KernSmooth_2.23-22     blob_1.2.4             shape_1.4.6           
 [91] jpeg_0.1-10            gridGraphics_0.5-1     scales_1.2.1          
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.8            
 [97] gplots_3.1.3           zlibbioc_1.42.0        compiler_4.2.0        
[100] scatterpie_0.2.1       BiocIO_1.6.0           clue_0.3-64           
[103] intergraph_2.0-3       lme4_1.1-34            cli_3.6.1             
[106] patchwork_1.1.3        ps_1.7.5               htmlTable_2.4.1       
[109] Formula_1.2-5          mgcv_1.9-0             MASS_7.3-60           
[112] tidyselect_1.2.0       stringi_1.7.12         highr_0.10            
[115] yaml_2.3.7             GOSemSim_2.22.0        locfit_1.5-9.8        
[118] latticeExtra_0.6-30    ggrepel_0.9.3          sass_0.4.7            
[121] fastmatch_1.1-4        tools_4.2.0            timechange_0.2.0      
[124] parallel_4.2.0         circlize_0.4.15        rstudioapi_0.15.0     
[127] foreign_0.8-84         foreach_1.5.2          git2r_0.32.0          
[130] gridExtra_2.3          farver_2.1.1           ggraph_2.1.0          
[133] digest_0.6.33          BiocManager_1.30.22    networkD3_0.4         
[136] Rcpp_1.0.11            broom_1.0.5            later_1.3.1           
[139] httr_1.4.7             ComplexHeatmap_2.12.1  GENIE3_1.18.0         
[142] Rdpack_2.5             colorspace_2.1-0       XML_3.99-0.14         
[145] fs_1.6.3               splines_4.2.0          statmod_1.5.0         
[148] yulab.utils_0.0.8      RBGL_1.72.0            tidytree_0.4.5        
[151] graphlayouts_1.0.0     ggplotify_0.1.2        xtable_1.8-4          
[154] jsonlite_1.8.7         nloptr_2.0.3           ggtree_3.4.4          
[157] tidygraph_1.2.3        ggfun_0.1.2            R6_2.5.1              
[160] Hmisc_5.1-0            pillar_1.9.0           htmltools_0.5.6       
[163] glue_1.6.2             fastmap_1.1.1          minqa_1.2.5           
[166] codetools_0.2-19       fgsea_1.22.0           utf8_1.2.3            
[169] sva_3.44.0             lattice_0.21-8         bslib_0.5.1           
[172] network_1.18.1         pbkrtest_0.5.2         curl_5.0.2            
[175] gtools_3.9.4           interp_1.1-4           survival_3.5-7        
[178] statnet.common_4.9.0   rmarkdown_2.24         munsell_0.5.0         
[181] GetoptLong_1.0.5       DO.db_2.9              GenomeInfoDbData_1.2.8
[184] iterators_1.0.14       gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
