<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-12-06" />

<title>WPCNA_shallow_split_2</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Omar-Johnson/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WPCNA_shallow_split_2</h1>
<h4 class="date">2023-12-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230911code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230911)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230911code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230911)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreeb1e8ba848931be2faf8efb3aebea0e81006bf1f7targetblankb1e8ba8a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Omar-Johnson/myproject/tree/b1e8ba848931be2faf8efb3aebea0e81006bf1f7" target="_blank">b1e8ba8</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomOmarJohnsonmyprojecttreeb1e8ba848931be2faf8efb3aebea0e81006bf1f7targetblankb1e8ba8a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Omar-Johnson/myproject/tree/b1e8ba848931be2faf8efb3aebea0e81006bf1f7" target="_blank">b1e8ba8</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/WPCNA_shallow_split_2.Rmd</code>)
and HTML (<code>docs/WPCNA_shallow_split_2.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Omar-Johnson/myproject/blob/b1e8ba848931be2faf8efb3aebea0e81006bf1f7/analysis/WPCNA_shallow_split_2.Rmd" target="_blank">b1e8ba8</a>
</td>
<td>
Omar-Johnson
</td>
<td>
2023-12-06
</td>
<td>
updated code
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="added-funtions" class="section level1">
<h1>Added funtions</h1>
<pre class="r"><code>remove_rows_with_nas &lt;- function(data_frame) {
  data_frame[rowSums(is.na(data_frame)) &lt; 5, ]
}</code></pre>
</div>
<div id="steps" class="section level1">
<h1>Steps</h1>
<pre class="r"><code>#* 1. Take Abundance values/counts 
#* 2. Remove NA values from Abundance values 
#* 3. Apply log2 transform to Abundance values 
#* 4. Quantile normalize abundance values 
#* 5. RUVs applied to account for technical variation 
#* 6. Use Limma to model differential protein abundaance </code></pre>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
<pre class="r"><code># Load your data
Protein_DF &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Abundance.csv&quot;, header = TRUE)

# Include metadata about samples 
Meta &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/Meta.csv&quot;, header = TRUE)</code></pre>
</div>
<div id="wrangle-data-frame" class="section level1">
<h1>Wrangle Data Frame</h1>
<pre class="r"><code># Save the original 
Full_DF &lt;- Protein_DF 


# Subset the columns referring to abundance
Protein_DF_Abundance &lt;- Protein_DF[,c(4, 5:ncol(Protein_DF))] 

# Check for duplicated rows 
# Identify duplicated values
Protein_DF_Abundance$duplicated_name &lt;- duplicated(Protein_DF_Abundance$Accession)

# This will return a logical vector where TRUE indicates the position of duplicates in the column.
# To see only rows with duplicated values, you can subset the dataframe like this:
duplicated_rows &lt;- Protein_DF_Abundance[Protein_DF_Abundance$duplicated_name == TRUE, ]
print(duplicated_rows)</code></pre>
<pre><code>    Accession Abundance..F1..Sample..Dox..n.a Abundance..F3..Sample..Dox..n.a
61     P04264                        3.87e+08                        1.45e+08
276    P35908                              NA                              NA
997    P04040                              NA                              NA
    Abundance..F5..Sample..Dox..Dox Abundance..F7..Sample..Dox..Dox
61                         2.58e+08                        7.07e+08
276                              NA                              NA
997                              NA                              NA
    Abundance..F9..Sample..Dox..Dox Abundance..F2..Control..Veh..n.a
61                         3.17e+08                         1.19e+10
276                              NA                         5.31e+06
997                              NA                               NA
    Abundance..F4..Control..Veh..n.a Abundance..F6..Control..Veh..Veh
61                          1.11e+09                         8.03e+08
276                         1.14e+06                               NA
997                               NA                               NA
    Abundance..F8..Control..Veh..Veh Abundance..F10..Control..Veh..Veh
61                          4.61e+08                          5.45e+08
276                         5.00e+05                          4.90e+05
997                               NA                                NA
    duplicated_name
61             TRUE
276            TRUE
997            TRUE</code></pre>
<pre class="r"><code># We have 3 rows with duplicates, two of which have NA values. We will remove them now since we will go ahead 
# and remove the duplicated rows later on anyways. 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[- c(61, 276, 997), ] 

# Make rownames of the data frame accession IDs
rownames(Protein_DF_Abundance) &lt;- Protein_DF_Abundance$Accession

# Change colnames 
colnames(Protein_DF_Abundance)</code></pre>
<pre><code> [1] &quot;Accession&quot;                         &quot;Abundance..F1..Sample..Dox..n.a&quot;  
 [3] &quot;Abundance..F3..Sample..Dox..n.a&quot;   &quot;Abundance..F5..Sample..Dox..Dox&quot;  
 [5] &quot;Abundance..F7..Sample..Dox..Dox&quot;   &quot;Abundance..F9..Sample..Dox..Dox&quot;  
 [7] &quot;Abundance..F2..Control..Veh..n.a&quot;  &quot;Abundance..F4..Control..Veh..n.a&quot; 
 [9] &quot;Abundance..F6..Control..Veh..Veh&quot;  &quot;Abundance..F8..Control..Veh..Veh&quot; 
[11] &quot;Abundance..F10..Control..Veh..Veh&quot; &quot;duplicated_name&quot;                  </code></pre>
<pre class="r"><code>Protein_DF_Abundance &lt;- Protein_DF_Abundance[, -c(1, 12)]

# Assuming column names of Protein_DF match with the rows in Meta
rownames(Meta) &lt;- Meta$Samples

colnames(Protein_DF_Abundance) &lt;- Meta$Samples</code></pre>
</div>
<div id="remove-na-values" class="section level1">
<h1>Remove NA values</h1>
<pre class="r"><code>#1.  Find rows with any NA values
rows_with_na &lt;- apply(Protein_DF_Abundance, 1, function(x) any(is.na(x))) %&gt;% which()
 

rows_with_na &lt;- as.numeric(rows_with_na)
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 4501   10</code></pre>
<pre class="r"><code># 2. Remove rows with NA values 
Protein_DF_Abundance &lt;- Protein_DF_Abundance[-rows_with_na, ]

# 3. Sanity check 
Protein_DF_Abundance %&gt;% dim()</code></pre>
<pre><code>[1] 3384   10</code></pre>
</div>
<div id="log2-transform" class="section level1">
<h1>Log2 transform</h1>
<pre class="r"><code>log2_norm_counts &lt;- log2(Protein_DF_Abundance)</code></pre>
</div>
<div id="quantile-normalization" class="section level1">
<h1>Quantile Normalization</h1>
<pre class="r"><code>normalizedData &lt;- limma::normalizeBetweenArrays(log2_norm_counts, method = &quot;quantile&quot;)</code></pre>
</div>
<div id="prepare-data-object-for-ruvs" class="section level1">
<h1>Prepare Data Object for RUVs</h1>
<pre class="r"><code>#  counts need to be integer values and in a numeric matrix 
counts &lt;- as.matrix(normalizedData) 

# Create a DataFrame for the phenoData
phenoData &lt;- DataFrame(Meta)

# Now create the RangedSummarizedExperiment necessary for RUVs input 
set &lt;- SummarizedExperiment(assays =  normalizedData, metadata =phenoData)</code></pre>
</div>
<div id="ruvs-matrix" class="section level1">
<h1>RUVs Matrix</h1>
<pre class="r"><code># Generate a background matrix where each set of replicates are in a row
# and null spaces where there are replicate mismatches get the value -1. 
# This will be used in the RUVs function, which is the typeII method to
# remove unwanted variation in the RUVs package. This will be used for all 
# RUVs permutations where the number of variation factors will be changed. 
scIdx &lt;- matrix(-1, nrow = 6, ncol = 3)
scIdx[1, 1:3] &lt;- c(which(colnames(counts) %in% c(&quot;S6&quot;, &quot;S8&quot;, &quot;S10&quot;))) # Control group- Technical reps Ind. 48-1
scIdx[2, 1:3] &lt;- c(which(colnames(counts) %in% c(&quot;S5&quot;, &quot;S7&quot;, &quot;S9&quot;))) # Dox group- Technical reps Ind. 48-1
scIdx[3, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S2&quot;))) # Control group- Bio reps Ind. 77-1
scIdx[4, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S1&quot;))) # Dox group- 77-1
scIdx[5, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S4&quot;))) # Control group- Bio reps Ind. 87-1
scIdx[6, 1] &lt;- c(which(colnames(counts) %in% c(&quot;S3&quot;))) # Dox group- 87-1
scIdx</code></pre>
<pre><code>     [,1] [,2] [,3]
[1,]    8    9   10
[2,]    3    4    5
[3,]    6   -1   -1
[4,]    1   -1   -1
[5,]    7   -1   -1
[6,]    2   -1   -1</code></pre>
</div>
<div id="linear-modelling" class="section level1">
<h1>Linear Modelling</h1>
<pre class="r"><code># Apply RUVs function from RUVSeq and add to set object
set &lt;- RUVSeq::RUVs(x = normalizedData, k = 1, scIdx = scIdx, isLog = TRUE)
RUV_1 &lt;- set$W[,1]

set$W</code></pre>
<pre><code>            W_1
S1   0.14326547
S3  -0.08146518
S5  -0.25579041
S7   0.54847999
S9  -0.50097929
S2   0.15639944
S4  -0.04767015
S6  -0.35626503
S8  -0.19216750
S10  0.48449144</code></pre>
<pre class="r"><code>phenoData$RUV_1 &lt;- RUV_1


# Create Design Matrix 
# phenoData$Cond &lt;- factor(phenoData$Cond , levels = c(&quot;Control&quot;, &quot;Dox&quot;))
design &lt;- model.matrix(~0 + Cond + RUV_1, phenoData)
design</code></pre>
<pre><code>    CondControl CondDox       RUV_1
S1            0       1  0.14326547
S3            0       1 -0.08146518
S5            0       1 -0.25579041
S7            0       1  0.54847999
S9            0       1 -0.50097929
S2            1       0  0.15639944
S4            1       0 -0.04767015
S6            1       0 -0.35626503
S8            1       0 -0.19216750
S10           1       0  0.48449144
attr(,&quot;assign&quot;)
[1] 1 1 2
attr(,&quot;contrasts&quot;)
attr(,&quot;contrasts&quot;)$Cond
[1] &quot;contr.treatment&quot;</code></pre>
<pre class="r"><code># rename columns 
colnames(design) &lt;- c(&#39;Control&#39;, &quot;Dox&quot;, &quot;RUV_1&quot;)


# Fit model
fit &lt;- lmFit(set$normalizedCounts, design)
fit2 &lt;- eBayes(fit)


# Make contrasts
cm &lt;- makeContrasts(
  DoxvNorm = Dox - Control,
  levels = design)


# Model with contrasts
fit2 &lt;- contrasts.fit(fit, cm)
fit2 &lt;- eBayes(fit2, robust = TRUE, trend = TRUE)</code></pre>
<pre><code>Warning: 3 very small variances detected, have been offset away from zero</code></pre>
<pre class="r"><code># Summarize
results_summary &lt;- decideTests(fit2, adjust.method = &quot;none&quot;, p.value = 0.05)
summary(results_summary)</code></pre>
<pre><code>       DoxvNorm
Down        300
NotSig     2882
Up          202</code></pre>
<pre class="r"><code># Toptable summary organized to contain results for all tested proteins   
toptable_summary &lt;- topTable(fit2, coef = &quot;DoxvNorm&quot;,number = (nrow(set$normalizedCounts )), p.value = 1, adjust.method = &quot;none&quot;)
toptable_summary$Protein &lt;- rownames(toptable_summary)</code></pre>
</div>
<div id="wpcna--shallow-cuts" class="section level1">
<h1>WPCNA- shallow cuts</h1>
<pre class="r"><code># Create summarized experiment object 
WGCNA_DF &lt;- SummarizedExperiment(assays = set$normalizedCounts)
exp_filt &lt;- filter_by_variance(WGCNA_DF, n = 3384)</code></pre>
</div>
<div id="sft-power-threshold" class="section level1">
<h1>SFT power threshold</h1>
<pre class="r"><code># Determine the soft power threshold to get scale free shape
sft &lt;- SFT_fit(exp_filt, net_type = &quot;signed&quot;, rsquared = 0.8,  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>Warning: executing %dopar% sequentially: no parallel backend registered</code></pre>
<pre><code>Warning in eval(xpr, envir = envir): Some correlations are NA in block 1 : 3384
.</code></pre>
<pre><code>   Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
1      3  0.20300  4.530          0.757   655.0     653.0    811
2      4  0.35300  4.960          0.444   459.0     455.0    633
3      5  0.40300  4.100          0.270   338.0     335.0    516
4      6  0.73400  2.230          0.734   259.0     255.0    433
5      7  0.56500  1.410          0.498   204.0     199.0    372
6      8  0.61600  1.030          0.875   164.0     159.0    326
7      9  0.55600  0.854          0.953   135.0     129.0    289
8     10  0.09660  0.501          0.934   112.0     107.0    259
9     11  0.00844 -0.121          0.918    94.8      88.8    235
10    12  0.16500 -0.490          0.951    80.9      74.9    214
11    13  0.33100 -0.720          0.950    69.8      63.6    196
12    14  0.44200 -0.869          0.950    60.8      54.5    181
13    15  0.53200 -0.975          0.955    53.3      47.0    167
14    16  0.59400 -1.060          0.967    47.0      40.7    156
15    17  0.65300 -1.140          0.972    41.7      35.5    145
16    18  0.69400 -1.220          0.973    37.3      31.3    136
17    19  0.72700 -1.270          0.975    33.4      27.6    128
18    20  0.74300 -1.330          0.972    30.1      24.5    120</code></pre>
<pre><code>No power reached R-squared cut-off, now choosing max R-squared based power</code></pre>
<pre class="r"><code>sft$power</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>power &lt;- sft$power
sft$plot </code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/SFT%20power%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>power</code></pre>
<pre><code>[1] 20</code></pre>
</div>
<div id="modify-workflow-to-have-a-simpler-network"
class="section level1">
<h1>Modify workflow to have a simpler network</h1>
<pre class="r"><code>exp2gcn2_EDIT &lt;- function (exp, net_type = &quot;signed&quot;, module_merging_threshold = 0.8, 
          SFTpower = NULL, cor_method = &quot;spearman&quot;, verbose = FALSE) 
{
  params &lt;- list(net_type = net_type, module_merging_threshold = module_merging_threshold, 
                 SFTpower = SFTpower, cor_method = cor_method)
  norm.exp &lt;- BioNERO:::handleSE(exp)
  if (is.null(SFTpower)) {
    stop(&quot;Please, specify the SFT power.&quot;)
  }
  if (verbose) {
    message(&quot;Calculating adjacency matrix...&quot;)
  }
  cor_matrix &lt;- BioNERO:::calculate_cor_adj(cor_method, norm.exp, SFTpower, 
                                  net_type)$cor
  adj_matrix &lt;- BioNERO:::calculate_cor_adj(cor_method, norm.exp, SFTpower, 
                                  net_type)$adj
  gene_ids &lt;- rownames(adj_matrix)
  adj_matrix &lt;- matrix(adj_matrix, nrow = nrow(adj_matrix))
  rownames(adj_matrix) &lt;- gene_ids
  colnames(adj_matrix) &lt;- gene_ids
  if (verbose) {
    message(&quot;Calculating topological overlap matrix (TOM)...&quot;)
  }
  tomtype &lt;- BioNERO:::get_TOMtype(net_type)
  TOM &lt;- WGCNA::TOMsimilarity(adj_matrix, TOMType = tomtype)
  dissTOM &lt;- 1 - TOM
  geneTree &lt;- hclust(as.dist(dissTOM), method = &quot;average&quot;)
  if (verbose) {
    message(&quot;Detecting coexpression modules...&quot;)
  }
  old.module_labels &lt;- dynamicTreeCut::cutreeDynamicTree(dendro = geneTree, maxTreeHeight = 4 , 
                                                         minModuleSize = 30, deepSplit = FALSE)
  nmod &lt;- length(unique(old.module_labels))
  palette &lt;- rev(WGCNA::standardColors(nmod))
  old.module_colors &lt;- WGCNA::labels2colors(old.module_labels, 
                                            colorSeq = palette)
  if (verbose) {
    message(&quot;Calculating module eigengenes (MEs)...&quot;)
  }
  old.MElist &lt;- WGCNA::moduleEigengenes(t(norm.exp), colors = old.module_colors, 
                                        softPower = SFTpower)
  old.MEs &lt;- old.MElist$eigengenes
  MEDiss1 &lt;- 1 - cor(old.MEs)
  old.METree &lt;- hclust(as.dist(MEDiss1), method = &quot;average&quot;)
  MEDissThreshold &lt;- 1 - module_merging_threshold
  if (verbose) {
    message(&quot;Merging similar modules...&quot;)
  }
  if (cor_method == &quot;pearson&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, colorSeq = palette)
  }
  else if (cor_method == &quot;spearman&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, corOptions = list(use = &quot;p&quot;, 
                                                                                                   method = &quot;spearman&quot;), colorSeq = palette)
  }
  else if (cor_method == &quot;biweight&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, corFnc = bicor, 
                                       colorSeq = palette)
  }
  else {
    stop(&quot;Please, specify a correlation method. One of &#39;spearman&#39;, &#39;pearson&#39; or &#39;biweight&#39;.&quot;)
  }
  new.module_colors &lt;- merge1$colors
  new.MEs &lt;- merge1$newMEs
  new.METree &lt;- hclust(as.dist(1 - cor(new.MEs)), method = &quot;average&quot;)
  genes_and_modules &lt;- as.data.frame(cbind(gene_ids, new.module_colors))
  colnames(genes_and_modules) &lt;- c(&quot;Genes&quot;, &quot;Modules&quot;)
  if (verbose) {
    message(&quot;Calculating intramodular connectivity...&quot;)
  }
  kwithin &lt;- WGCNA::intramodularConnectivity(adj_matrix, new.module_colors)
  result.list &lt;- list(adjacency_matrix = adj_matrix, MEs = new.MEs, 
                      genes_and_modules = genes_and_modules, kIN = kwithin, 
                      correlation_matrix = cor_matrix, params = params, dendro_plot_objects = list(tree = geneTree, 
                                                                                                   unmerged = old.module_colors))
  return(result.list)
}</code></pre>
</div>
<div id="generate-network-with-max-merge-threshold"
class="section level1">
<h1>Generate network with max merge threshold</h1>
<pre class="r"><code>net &lt;- exp2gcn2_EDIT(
  exp_filt, net_type = &quot;signed&quot;, module_merging_threshold = 0.99, SFTpower = power, 
  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>Warning in cor(t(norm.exp), method = &quot;pearson&quot;): the standard deviation is zero

Warning in cor(t(norm.exp), method = &quot;pearson&quot;): the standard deviation is zero</code></pre>
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
<pre class="r"><code># Dendogram
plot_dendro_and_colors(net )</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code># Eigengene networks
plot_eigengene_network(net)</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code># Genes per module 
plot_ngenes_per_module(net)</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="generate-network-with-desired-merge-threshold"
class="section level1">
<h1>Generate network with desired merge threshold</h1>
<pre class="r"><code>net &lt;- exp2gcn2_EDIT(
  exp_filt, net_type = &quot;signed&quot;, module_merging_threshold = 0.9, SFTpower = power, 
  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>Warning in cor(t(norm.exp), method = &quot;pearson&quot;): the standard deviation is zero

Warning in cor(t(norm.exp), method = &quot;pearson&quot;): the standard deviation is zero</code></pre>
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
<pre class="r"><code># Dendogram
plot_dendro_and_colors(net )</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code># Eigengene networks
plot_eigengene_network(net)</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code># Genes per module 
plot_ngenes_per_module(net)</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="define-trat-and-get-eigen-proteins" class="section level1">
<h1>Define trat and get eigen proteins</h1>
<pre class="r"><code>WGCNA_DF$trait &lt;- c(1,1,1,1,1,0,0,0,0,0)
names(WGCNA_DF$trait) &lt;- c(&quot;S1&quot;, &quot;S3&quot;, &quot;S5&quot;, &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot;, &quot;S8&quot;, &quot;S10&quot;)


eigenmatrix &lt;- net$MEs %&gt;% as.matrix()
eigenmatrix</code></pre>
<pre><code>    MEdarkgreen MEgreenyellow     MEmagenta     MEgreen     MEblack
S1   0.18210620  -0.005457502  0.0546504186  0.23785319  0.33106572
S3   0.09718442  -0.655438915 -0.3541181487 -0.40451119 -0.54702648
S5   0.47377047   0.268889367  0.3926743828 -0.09393906  0.19215729
S7   0.37887364   0.167144729  0.3404051718 -0.14945040  0.16136285
S9   0.28268140   0.099590513  0.3214629391 -0.16090425  0.08620987
S2  -0.14172225   0.133528742 -0.0307145452  0.46446037  0.36795956
S4  -0.45770935  -0.577675132 -0.7049856193 -0.58826295 -0.61902191
S6  -0.10370053   0.187073292 -0.0007939648  0.22622080  0.03890218
S8  -0.44529220   0.208263512 -0.0158109910  0.23026588  0.01151404
S10 -0.26619180   0.174081395 -0.0027696432  0.23826761 -0.02312313
        MEsalmon MElightgreen   MEyellow       MEblue        MEred   MEdarkred
S1   0.542569499 -0.532070908 -0.1860911 -0.200268432 -0.244062683  0.61376944
S3  -0.294753301  0.133852972  0.5204434  0.606669131  0.397994490  0.12868780
S5   0.083283562  0.392477389  0.2639844 -0.047183088  0.157919249 -0.17542230
S7   0.009538343  0.323113109  0.2405889 -0.025545978  0.106267829 -0.12726682
S9  -0.038409580  0.283818231  0.1258283  0.006566511  0.135611486 -0.22293667
S2   0.564028253 -0.598142839 -0.3915650 -0.476664129 -0.789311176  0.54523021
S4  -0.487479730  0.038774557  0.3389226  0.550788180  0.314929917  0.02152929
S6  -0.070323396 -0.004717392 -0.2792831 -0.128104270 -0.072046134 -0.19659221
S8  -0.185490529 -0.019443983 -0.2680474 -0.165087386 -0.002735437 -0.26719162
S10 -0.122963121 -0.017661136 -0.3647809 -0.121170540 -0.004567542 -0.31980710
        MEbrown MElightcyan      MEpink MElightyellow MEroyalblue        MEgrey
S1   0.20642078  0.16555633 -0.08055571  -0.154632644  -0.4139420 -0.0003025242
S3   0.33116917  0.61995623 -0.34137266   0.309740607  -0.2748380  0.0040883740
S5  -0.12029380 -0.15360435 -0.21437397  -0.367098522  -0.4156534  0.0003822208
S7  -0.18038246 -0.06386671 -0.18266541  -0.469429132  -0.2406861 -0.0036422147
S9  -0.56619499 -0.03109400 -0.24024156  -0.375568753  -0.1299847 -0.0007300618
S2   0.09492993 -0.05626903  0.81404455   0.001984603   0.1126542  0.0040695629
S4   0.59639333  0.49138129  0.23786812   0.532176407   0.2150345 -0.0034138146
S6  -0.05482123 -0.32175845  0.11219158   0.107575233   0.3172382 -0.0049835507
S8   0.02699682 -0.32301622 -0.08849206   0.169108022   0.4411122  0.0028889389
S10 -0.33421753 -0.32728509 -0.01640288   0.246144180   0.3890651  0.0016430694</code></pre>
</div>
<div id="trait-module-correlation-test" class="section level1">
<h1>Trait-module correlation test</h1>
<pre class="r"><code>correlationResults &lt;- cor(eigenmatrix, WGCNA_DF$trait, method=&quot;pearson&quot;)

pvalues &lt;- sapply(1:ncol(eigenmatrix), function(column) cor.test(eigenmatrix[,column], WGCNA_DF$trait)$p.value )

WGCNA_DF$trait &lt;- as.matrix(WGCNA_DF$trait)

Modeigen &lt;- as.matrix(net$MEs)


# Trait data frame
trait_data &lt;- data.frame(
  ME = correlationResults %&gt;% rownames(),
  Value = correlationResults[,1],
  PValue = pvalues
)
trait_data$neglogP &lt;- -log(trait_data$PValue)
trait_data</code></pre>
<pre><code>                         ME       Value       PValue    neglogP
MEdarkgreen     MEdarkgreen  0.89468179 4.731737e-04 7.65604799
MEgreenyellow MEgreenyellow -0.07922885 8.277707e-01 0.18901907
MEmagenta         MEmagenta  0.47755121 1.627645e-01 1.81545108
MEgreen             MEgreen -0.36110157 3.052819e-01 1.18651965
MEblack             MEblack  0.14152410 6.965426e-01 0.36162639
MEsalmon           MEsalmon  0.19114610 5.968131e-01 0.51615123
MElightgreen   MElightgreen  0.38022644 2.784303e-01 1.27858762
MEyellow           MEyellow  0.61016390 6.102503e-02 2.79647117
MEblue               MEblue  0.21518550 5.504793e-01 0.59696584
MEred                 MEred  0.35020984 3.211616e-01 1.13581099
MEdarkred         MEdarkred  0.13713624 7.055927e-01 0.34871708
MEbrown             MEbrown -0.20825578 5.636895e-01 0.57325176
MElightcyan     MElightcyan  0.33959542 3.370407e-01 1.08755149
MEpink               MEpink -0.66990279 3.406547e-02 3.37947099
MElightyellow MElightyellow -0.66849819 3.458275e-02 3.36440039
MEroyalblue     MEroyalblue -0.93293782 8.156486e-05 9.41411198
MEgrey               MEgrey -0.01326124 9.709961e-01 0.02943278</code></pre>
<pre class="r"><code># Plot correlation across modules
ggplot(trait_data, aes(x = ME, y = neglogP, color = Value)) + 
  geom_segment(aes(xend = ME, yend = 0), linetype = &quot;solid&quot;) +
  geom_point(size = 10) + 
  scale_color_gradient(
    low = &quot;red&quot;, 
    high = &quot;blue&quot;, 
    name = &quot;corr&quot;,
    limits = c(-1, 1),
    breaks = seq(-1, 1, by = 0.5)
  ) +
  labs(x = &quot;Module&quot;, y = &quot;-log(P-value)&quot;, title = &quot;Module correlation with DOX treatment&quot;) + 
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  geom_hline(yintercept = 2.995732, color = &quot;black&quot;, linetype = &quot;dashed&quot;) +
  geom_text(aes(x = 0, y = 3.5, label = &quot;P-Val Threshold&quot;), hjust = -0.1, vjust = 1.5)</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Trait-module%20correlation%20test-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="general-wrangling" class="section level1">
<h1>General wrangling</h1>
<pre class="r"><code># Get modules that significantly associate with DOX treatment 
Signifigant_modules &lt;-  trait_data[trait_data$PValue &lt; 0.05, ]$ME


# Get your genes/proteins and modules DF 
Gene_mod_DF &lt;- net$genes_and_modules 



# Module data gets merged with toptable
toptable_summary %&gt;% head()</code></pre>
<pre><code>            logFC  AveExpr          t      P.Value    adj.P.Val        B
Q15327 -3.5105795 23.99385 -16.206595 1.439598e-08 1.439598e-08 9.198952
O76021 -2.1010283 24.26413 -13.346701 9.464205e-08 9.464205e-08 7.858831
Q969Q1 -4.1607271 20.89793  -9.786021 1.771490e-06 1.771490e-06 5.466554
Q92841 -0.7680309 27.58532  -9.120228 3.274667e-06 3.274667e-06 4.925168
P67809 -1.0832634 26.08848  -9.068268 3.560682e-06 3.560682e-06 4.850881
Q9BQG0 -2.1975800 23.65668  -9.036430 3.676428e-06 3.676428e-06 4.822314
       Protein
Q15327  Q15327
O76021  O76021
Q969Q1  Q969Q1
Q92841  Q92841
P67809  P67809
Q9BQG0  Q9BQG0</code></pre>
<pre class="r"><code>Toptable_Modules &lt;- merge(toptable_summary, 
                          Gene_mod_DF, by.x = &quot;Protein&quot;, by.y = &quot;Genes&quot;)





# Get the corresponding Gene name IDs
mart &lt;- useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)


results_Pro_names &lt;- getBM(filters = &quot;uniprotswissprot&quot;, 
                           attributes = c(&quot;uniprotswissprot&quot;, &quot;hgnc_symbol&quot;), 
                           values = toptable_summary$Protein, 
                           mart = mart)


merged_results_Pro_names_AND_Toptable_for_ENZYME &lt;- merge(toptable_summary, results_Pro_names, by.x = &quot;Protein&quot;, by.y = &quot;uniprotswissprot&quot;)
ENZYME_DF_Toptable &lt;- merged_results_Pro_names_AND_Toptable_for_ENZYME

DE_Enzymes &lt;- ENZYME_DF_Toptable[(ENZYME_DF_Toptable$P.Value &lt; 0.05) &amp; (!is.na(ENZYME_DF_Toptable$ENZYME)), ]$hgnc_symbol

All_Proteins_ENZYMES &lt;- ENZYME_DF_Toptable[(!is.na(ENZYME_DF_Toptable$ENZYME)), ]$hgnc_symbol





# Plotting LogFC in significant modules module 

Significant_modules_NoME &lt;- stringr::str_replace(Signifigant_modules, &quot;^ME&quot;, &quot;&quot;)

Toptable_Modules_Sig &lt;- Toptable_Modules[Toptable_Modules$Modules %in% Significant_modules_NoME, ]



# n-modules total 
Toptable_Modules$Modules %&gt;% unique() %&gt;% length()</code></pre>
<pre><code>[1] 17</code></pre>
<pre class="r"><code># n-sig modules
Signifigant_modules %&gt;% length()</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code># Percent sig modules 
(Signifigant_modules %&gt;% length()) / (Toptable_Modules$Modules %&gt;% unique() %&gt;% length())</code></pre>
<pre><code>[1] 0.2352941</code></pre>
<pre class="r"><code># Get modules that significantly associate with DOX treatment 
Signifigant_modules &lt;-  trait_data[trait_data$PValue &lt; 0.05, ]$ME


# Get your genes/proteins and modules DF 
Gene_mod_DF &lt;- net$genes_and_modules 


# Subset genes/proteins and modules DF by the significantly correlated modules to DOX 
Sig_Gene_mod_DF &lt;- Gene_mod_DF[Gene_mod_DF$Modules %in% Signifigant_modules, ]


# Module data gets merged with toptable
Toptable_Modules &lt;- merge(toptable_summary, 
                          Gene_mod_DF, by.x = &quot;Protein&quot;, by.y = &quot;Genes&quot;)

Toptable_Modules %&gt;% head()</code></pre>
<pre><code>     Protein        logFC  AveExpr            t   P.Value adj.P.Val         B
1 A0A0B4J2D5  0.183555215 28.46018  2.024992979 0.0699426 0.0699426 -4.632753
2     A0FGR8  0.201889154 23.26550  0.730591393 0.4816055 0.4816055 -6.193199
3     A1A4S6  1.195526327 19.15569  1.248949692 0.2398095 0.2398095 -5.695109
4     A1L0T0 -0.193311424 24.34047 -1.611750825 0.1376069 0.1376069 -5.234984
5     A2RRP1 -0.250688318 24.19697 -1.786081518 0.1040252 0.1040252 -4.989523
6     A3KMH1  0.001585613 25.50568  0.009025006 0.9929747 0.9929747 -6.472078
    Modules
1 darkgreen
2       red
3 lightcyan
4 royalblue
5   darkred
6   darkred</code></pre>
<pre class="r"><code>Toptable_Modules %&gt;% dim()</code></pre>
<pre><code>[1] 3384    8</code></pre>
</div>
<div id="combining-diff-abundant-test-with-modules"
class="section level1">
<h1>Combining diff abundant test with modules</h1>
<pre class="r"><code># LogFC Ridge plot 
ggplot(Toptable_Modules_Sig, aes(x = logFC, y = Modules, fill = Modules)) +
  ggridges::geom_density_ridges(alpha = 0.8) +
  labs(title = &quot;Dox Associated Module LogFC distribution from DE test&quot;,
       x = &quot;Log Fold Change&quot;,
       y = &quot;Modules&quot;) +
  ggridges::theme_ridges() </code></pre>
<pre><code>Picking joint bandwidth of 0.107</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Riddge-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># LogFC Ridge plot 
ggplot(Toptable_Modules, aes(x = logFC, y = Modules, fill = Modules)) +
  ggridges::geom_density_ridges(alpha = 0.8) +
  labs(title = &quot;Dox Associated Module LogFC distribution from DE test&quot;,
       x = &quot;Log Fold Change&quot;,
       y = &quot;Modules&quot;) +
  ggridges::theme_ridges() </code></pre>
<pre><code>Picking joint bandwidth of 0.0783</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Riddge-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Pval Ridge plot 
ggplot(Toptable_Modules_Sig, aes(x = P.Value, y = Modules, fill = Modules)) +
  ggridges::geom_density_ridges(alpha = 0.8) +
  labs(title = &quot;Module pval distribution from DE test&quot;,
       x = &quot;Pval&quot;,
       y = &quot;&quot;) +
  ggridges::theme_ridges()+
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), limits = c(0, 1))</code></pre>
<pre><code>Picking joint bandwidth of 0.0615</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Riddge-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Pval Ridge plot 
ggplot(Toptable_Modules, aes(x = P.Value, y = Modules, fill = Modules)) +
  ggridges::geom_density_ridges(alpha = 0.6) +
  labs(title = &quot;Module pval distribution from DE test&quot;,
       x = &quot;Pval&quot;,
       y = &quot;Modules&quot;) +
  ggridges::theme_ridges()+
  scale_x_continuous(breaks = seq(0, 1, by = 0.05), limits = c(0, 1))</code></pre>
<pre><code>Picking joint bandwidth of 0.0752</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Riddge-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plotting LogFC between DE and non DE proteins 
Toptable_Modules$DE_or_Not &lt;- Toptable_Modules$P.Value &lt; 0.05


# Calculate percentage of DE genes for each module
DE_Mod_result &lt;- Toptable_Modules %&gt;%
  group_by(Modules) %&gt;%
  summarize(
    total_genes = n(),
    DE_genes = sum(P.Value &lt; 0.05),
    percent_DE = (DE_genes / total_genes) * 100
  )

print(DE_Mod_result)</code></pre>
<pre><code># A tibble: 17 × 4
   Modules     total_genes DE_genes percent_DE
   &lt;chr&gt;             &lt;int&gt;    &lt;int&gt;      &lt;dbl&gt;
 1 black               454       17      3.74 
 2 blue                109        0      0    
 3 brown                47        0      0    
 4 darkgreen           399       90     22.6  
 5 darkred             317        7      2.21 
 6 green                52        7     13.5  
 7 greenyellow          89        0      0    
 8 grey                173        1      0.578
 9 lightcyan           188       13      6.91 
10 lightgreen          279       55     19.7  
11 lightyellow         305       53     17.4  
12 magenta              76        1      1.32 
13 pink                 69       21     30.4  
14 red                 347       16      4.61 
15 royalblue           314      211     67.2  
16 salmon              115        0      0    
17 yellow               51       10     19.6  </code></pre>
<pre class="r"><code># Plot
ggplot(DE_Mod_result, aes(x = Modules, y = percent_DE, fill = Modules)) +
  geom_bar(stat = &quot;identity&quot;) +
  labs(
    x = &quot;Module&quot;,
    y = &quot;Percentage of DE genes&quot;,
    title = &quot;Percentage of DE genes in each module&quot;,
    fill = &quot;Module&quot;
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/DE%20across%20modules-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Compute complementary percentage for non-DE genes
DE_Mod_result$percent_nonDE &lt;- 100 - DE_Mod_result$percent_DE

# Reshape data to long format
long_data &lt;- DE_Mod_result %&gt;% 
  pivot_longer(cols = c(percent_DE, percent_nonDE), 
               names_to = &quot;type&quot;, 
               values_to = &quot;percentage&quot;)


# Plot stacked bar chart
ggplot(long_data, aes(x = Modules, y = percentage, fill = type)) +
  geom_bar(stat = &quot;identity&quot;) +
  labs(
    x = &quot;Module&quot;,
    y = &quot;Percentage&quot;,
    title = &quot;Percentage of DE and non-DE genes in each module&quot;,
    fill = &quot;Type&quot;
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_y_continuous(breaks = rev(seq(0, 100, by = 10)))</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/DE%20across%20modules-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Significant_modules_NoME</code></pre>
<pre><code>[1] &quot;darkgreen&quot;   &quot;pink&quot;        &quot;lightyellow&quot; &quot;royalblue&quot;  </code></pre>
<pre class="r"><code>metatestDF &lt;- phenoData$Cond %&gt;% as.data.frame()</code></pre>
</div>
<div id="go-enrichment-acrross-dox-correlated-modules"
class="section level1">
<h1>GO enrichment acrross dox correlated modules</h1>
<pre class="r"><code>hubs &lt;- get_hubs_gcn(WGCNA_DF, net)</code></pre>
<pre><code>Warning in WGCNA::signedKME(t(exp), MEs, outputColumnName = &quot;&quot;): Some genes are
constant. Hint: consider removing constant columns from datExpr.</code></pre>
<pre class="r"><code>ontology_categories &lt;- c(&quot;ALL&quot;)


# Loop over each module and ontology category
for (module in Significant_modules_NoME) {
  # Extract hub genes for the current module
  module_hubs &lt;- hubs[hubs$Module == module, ]$Gene
  
  # All module proteins
  module_proteins &lt;- Toptable_Modules[(Toptable_Modules$Modules == module), ]$Protein
  
  # Get gene names for the proteins in the module
  module_genes &lt;- ENZYME_DF_Toptable[ENZYME_DF_Toptable$Protein %in% module_proteins, ]$hgnc_symbol
  
  # Define background genes
  background_genes_mod &lt;- ENZYME_DF_Toptable$hgnc_symbol
  
  for (ont in ontology_categories) {
    # GO Enrichment
    go_enrichment &lt;- enrichGO(gene = module_genes,
                              OrgDb = org.Hs.eg.db,
                              keyType = &quot;SYMBOL&quot;,
                              universe = background_genes_mod,
                              ont = ont,
                              pvalueCutoff = 1,
                              qvalueCutoff = 1)
    
    # Plotting the results
    # Plotting the results
    p &lt;- barplot(go_enrichment, showCategory = 10) +
      ggtitle(paste(&quot;Module:&quot;, module, &quot;- Ontology:&quot;, ont))
    
    print(p)   
    
  }
}</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/GO%20enrichment-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/WPCNA_shallow_split_2.Rmd/GO%20enrichment-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/WPCNA_shallow_split_2.Rmd/GO%20enrichment-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/WPCNA_shallow_split_2.Rmd/GO%20enrichment-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="protein-corrrelation-plots-within-modules"
class="section level1">
<h1>Protein corrrelation plots within modules</h1>
<pre class="r"><code># Remove edges based on optimal scale-free topology fit
edges_filtered &lt;- get_edge_list(net, module = c(&quot;darkgreen&quot;, &quot;pink&quot;, &quot;lightyellow&quot;, &quot;royalblue&quot;), filter = TRUE)</code></pre>
<pre><code>The correlation threshold that best fits the scale-free topology is 0.9</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Protein%20corrrelation%20plots-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## The correlation threshold that best fits the scale-free topology is 0.7
dim(edges_filtered)</code></pre>
<pre><code>[1] 3875    3</code></pre>
<pre class="r"><code>## [1] 588   3

#### Network for all ####
# Remove edges based on minimum correlation
edges_filtered &lt;- get_edge_list(
  net, module = c(&quot;darkgreen&quot;, &quot;pink&quot;, &quot;lightyellow&quot;, &quot;royalblue&quot;), 
  filter = TRUE, method = &quot;min_cor&quot;, rcutoff = 0.93,
  check_SFT = TRUE
)</code></pre>
<pre><code>Your graph fits the scale-free topology. P-value:0.999999905427657</code></pre>
<pre class="r"><code>dim(edges_filtered)</code></pre>
<pre><code>[1] 1515    3</code></pre>
<pre class="r"><code>plot_gcn(
  edgelist_gcn = edges_filtered, 
  net = net, 
  color_by = &quot;module&quot;, 
  hubs = hubs,
  top_n_hubs = 30,
  show_labels = &quot;tophubs&quot;
)</code></pre>
<pre><code>Warning: ggrepel: 23 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Protein%20corrrelation%20plots-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="network-darkgreen" class="section level4">
<h4>Network darkgreen</h4>
<pre class="r"><code>edges_filtered &lt;- get_edge_list(
  net, module = c(&quot;darkgreen&quot; ), 
  filter = TRUE, method = &quot;min_cor&quot;, rcutoff = 0.9,
  check_SFT = TRUE
)</code></pre>
<pre><code>Your graph fits the scale-free topology. P-value:0.864905208343156</code></pre>
<pre class="r"><code>dim(edges_filtered)</code></pre>
<pre><code>[1] 908   3</code></pre>
<pre class="r"><code>plot_gcn(
  edgelist_gcn = edges_filtered, 
  net = net, 
  color_by = &quot;module&quot;, 
  hubs = hubs,
  top_n_hubs = 30,
  show_labels = &quot;tophubs&quot;
)</code></pre>
<pre><code>Warning: ggrepel: 10 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Network%20darkgreen-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="network-pink" class="section level4">
<h4>Network pink</h4>
<pre class="r"><code>edges_filtered &lt;- get_edge_list(
  net, module = c( &quot;pink&quot;), 
  filter = TRUE, method = &quot;min_cor&quot;, rcutoff = 0.8,
  check_SFT = TRUE
)</code></pre>
<pre><code>Your graph fits the scale-free topology. P-value:0.999999982872347</code></pre>
<pre class="r"><code>dim(edges_filtered)</code></pre>
<pre><code>[1] 348   3</code></pre>
<pre class="r"><code>plot_gcn(
  edgelist_gcn = edges_filtered, 
  net = net, 
  color_by = &quot;module&quot;, 
  hubs = hubs,
  top_n_hubs = 30,
  show_labels = &quot;tophubs&quot;
)</code></pre>
<pre><code>Warning: ggrepel: 5 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Network%20pink-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="network-lightyellow" class="section level4">
<h4>Network lightyellow</h4>
<pre class="r"><code>edges_filtered &lt;- get_edge_list(
  net, module = c(&quot;lightyellow&quot; ), 
  filter = TRUE, method = &quot;min_cor&quot;, rcutoff = 0.92,
  check_SFT = TRUE
)</code></pre>
<pre><code>Your graph fits the scale-free topology. P-value:0.99999999996731</code></pre>
<pre class="r"><code>dim(edges_filtered)</code></pre>
<pre><code>[1] 394   3</code></pre>
<pre class="r"><code>plot_gcn(
  edgelist_gcn = edges_filtered, 
  net = net, 
  color_by = &quot;module&quot;, 
  hubs = hubs,
  top_n_hubs = 30,
  show_labels = &quot;tophubs&quot;
)</code></pre>
<pre><code>Warning: ggrepel: 25 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Network%20lightyellow-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="network-royalblue" class="section level4">
<h4>Network royalblue</h4>
<pre class="r"><code>edges_filtered &lt;- get_edge_list(
  net, module = c(&quot;royalblue&quot;), 
  filter = TRUE, method = &quot;min_cor&quot;, rcutoff = 0.94,
  check_SFT = TRUE
)</code></pre>
<pre><code>Your graph fits the scale-free topology. P-value:0.999999990710176</code></pre>
<pre class="r"><code>dim(edges_filtered)</code></pre>
<pre><code>[1] 555   3</code></pre>
<pre class="r"><code>plot_gcn(
  edgelist_gcn = edges_filtered, 
  net = net, 
  color_by = &quot;module&quot;, 
  hubs = hubs,
  top_n_hubs = 30,
  show_labels = &quot;tophubs&quot;
)</code></pre>
<pre><code>Warning: ggrepel: 30 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/WPCNA_shallow_split_2.Rmd/Network%20royalblue-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] impute_1.70.0                          
 [2] WGCNA_1.72-1                           
 [3] fastcluster_1.2.3                      
 [4] dynamicTreeCut_1.63-1                  
 [5] BioNERO_1.4.2                          
 [6] reshape2_1.4.4                         
 [7] ggridges_0.5.4                         
 [8] biomaRt_2.52.0                         
 [9] ggvenn_0.1.10                          
[10] UpSetR_1.4.0                           
[11] DOSE_3.22.1                            
[12] variancePartition_1.26.0               
[13] clusterProfiler_4.4.4                  
[14] pheatmap_1.0.12                        
[15] qvalue_2.28.0                          
[16] Homo.sapiens_1.3.1                     
[17] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[18] org.Hs.eg.db_3.15.0                    
[19] GO.db_3.15.0                           
[20] OrganismDbi_1.38.1                     
[21] GenomicFeatures_1.48.4                 
[22] AnnotationDbi_1.58.0                   
[23] cluster_2.1.4                          
[24] ggfortify_0.4.16                       
[25] lubridate_1.9.2                        
[26] forcats_1.0.0                          
[27] stringr_1.5.0                          
[28] dplyr_1.1.2                            
[29] purrr_1.0.2                            
[30] readr_2.1.4                            
[31] tidyr_1.3.0                            
[32] tibble_3.2.1                           
[33] ggplot2_3.4.3                          
[34] tidyverse_2.0.0                        
[35] RColorBrewer_1.1-3                     
[36] RUVSeq_1.30.0                          
[37] edgeR_3.38.4                           
[38] limma_3.52.4                           
[39] EDASeq_2.30.0                          
[40] ShortRead_1.54.0                       
[41] GenomicAlignments_1.32.1               
[42] SummarizedExperiment_1.26.1            
[43] MatrixGenerics_1.8.1                   
[44] matrixStats_1.0.0                      
[45] Rsamtools_2.12.0                       
[46] GenomicRanges_1.48.0                   
[47] Biostrings_2.64.1                      
[48] GenomeInfoDb_1.32.4                    
[49] XVector_0.36.0                         
[50] IRanges_2.30.1                         
[51] S4Vectors_0.34.0                       
[52] BiocParallel_1.30.4                    
[53] Biobase_2.56.0                         
[54] BiocGenerics_0.42.0                    
[55] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           htmlwidgets_1.6.2     
 [40] igraph_1.5.1           DBI_1.1.3              ggnewscale_0.4.9      
 [43] backports_1.4.1        annotate_1.74.0        aod_1.3.2             
 [46] deldir_1.0-9           vctrs_0.6.3            abind_1.4-5           
 [49] cachem_1.0.8           withr_2.5.0            ggforce_0.4.1         
 [52] checkmate_2.2.0        treeio_1.20.2          prettyunits_1.1.1     
 [55] ape_5.7-1              lazyeval_0.2.2         crayon_1.5.2          
 [58] genefilter_1.78.0      labeling_0.4.2         pkgconfig_2.0.3       
 [61] tweenr_2.0.2           nlme_3.1-163           nnet_7.3-19           
 [64] rlang_1.1.1            lifecycle_1.0.3        downloader_0.4        
 [67] filelock_1.0.2         BiocFileCache_2.4.0    rprojroot_2.0.3       
 [70] polyclip_1.10-4        graph_1.74.0           Matrix_1.5-4.1        
 [73] aplot_0.2.0            NetRep_1.2.7           boot_1.3-28.1         
 [76] base64enc_0.1-3        GlobalOptions_0.1.2    whisker_0.4.1         
 [79] processx_3.8.2         png_0.1-8              viridisLite_0.4.2     
 [82] rjson_0.2.21           bitops_1.0-7           getPass_0.2-2         
 [85] R.oo_1.25.0            ggnetwork_0.5.12       KernSmooth_2.23-22    
 [88] blob_1.2.4             shape_1.4.6            jpeg_0.1-10           
 [91] gridGraphics_0.5-1     scales_1.2.1           memoise_2.0.1         
 [94] magrittr_2.0.3         plyr_1.8.8             gplots_3.1.3          
 [97] zlibbioc_1.42.0        compiler_4.2.0         scatterpie_0.2.1      
[100] BiocIO_1.6.0           clue_0.3-64            intergraph_2.0-3      
[103] lme4_1.1-34            cli_3.6.1              patchwork_1.1.3       
[106] ps_1.7.5               htmlTable_2.4.1        Formula_1.2-5         
[109] mgcv_1.9-0             MASS_7.3-60            tidyselect_1.2.0      
[112] stringi_1.7.12         highr_0.10             yaml_2.3.7            
[115] GOSemSim_2.22.0        locfit_1.5-9.8         latticeExtra_0.6-30   
[118] ggrepel_0.9.3          sass_0.4.7             fastmatch_1.1-4       
[121] tools_4.2.0            timechange_0.2.0       parallel_4.2.0        
[124] circlize_0.4.15        rstudioapi_0.15.0      foreign_0.8-84        
[127] foreach_1.5.2          git2r_0.32.0           gridExtra_2.3         
[130] farver_2.1.1           ggraph_2.1.0           digest_0.6.33         
[133] BiocManager_1.30.22    networkD3_0.4          Rcpp_1.0.11           
[136] broom_1.0.5            later_1.3.1            httr_1.4.7            
[139] ComplexHeatmap_2.12.1  GENIE3_1.18.0          Rdpack_2.5            
[142] colorspace_2.1-0       XML_3.99-0.14          fs_1.6.3              
[145] splines_4.2.0          statmod_1.5.0          yulab.utils_0.0.8     
[148] RBGL_1.72.0            tidytree_0.4.5         graphlayouts_1.0.0    
[151] ggplotify_0.1.2        xtable_1.8-4           jsonlite_1.8.7        
[154] nloptr_2.0.3           ggtree_3.4.4           tidygraph_1.2.3       
[157] ggfun_0.1.2            R6_2.5.1               Hmisc_5.1-0           
[160] pillar_1.9.0           htmltools_0.5.6        glue_1.6.2            
[163] fastmap_1.1.1          minqa_1.2.5            codetools_0.2-19      
[166] fgsea_1.22.0           utf8_1.2.3             sva_3.44.0            
[169] lattice_0.21-8         bslib_0.5.1            network_1.18.1        
[172] pbkrtest_0.5.2         curl_5.0.2             gtools_3.9.4          
[175] interp_1.1-4           survival_3.5-7         statnet.common_4.9.0  
[178] rmarkdown_2.24         munsell_0.5.0          GetoptLong_1.0.5      
[181] DO.db_2.9              GenomeInfoDbData_1.2.8 iterators_1.0.14      
[184] gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
